{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3020: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "data = pd.read_csv('fasta_index_file_oa.txt', sep=\"\\t\", header=None, names = [\"id\", \"filename_id\", \"filename\", \"database\",\"sub_database\",\"filename_2\",\"7\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename_id</th>\n",
       "      <th>filename</th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>filename_2</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_GCA_000208265.2_ASM20826v2_genomic.fna.gz</td>\n",
       "      <td>GCA_000208265.2_ASM20826v2_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_000208265.2_ASM20826v2_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2_GCA_900007175.1_NOCA2_genomic.fna.gz</td>\n",
       "      <td>GCA_900007175.1_NOCA2_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900007175.1_NOCA2_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3_GCA_900007185.1_NOCA1_genomic.fna.gz</td>\n",
       "      <td>GCA_900007185.1_NOCA1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900007185.1_NOCA1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4_GCA_900010595.2_55Contigs_genomic.fna.gz</td>\n",
       "      <td>GCA_900010595.2_55Contigs_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900010595.2_55Contigs_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5_GCA_900010795.1_Thauera_sp._DTG_genomic.fna...</td>\n",
       "      <td>GCA_900010795.1_Thauera_sp._DTG_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900010795.1_Thauera_sp._DTG_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6_GCA_900078555.1_M.thermophila.draft_genomic...</td>\n",
       "      <td>GCA_900078555.1_M.thermophila.draft_genomic.f...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078555.1_M.thermophila.draft_genomic.f...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7_GCA_900078565.1_Herbaspirillum.draft_genomi...</td>\n",
       "      <td>GCA_900078565.1_Herbaspirillum.draft_genomic....</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078565.1_Herbaspirillum.draft_genomic....</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8_GCA_900078575.1_Caulobacteraceae.draft_geno...</td>\n",
       "      <td>GCA_900078575.1_Caulobacteraceae.draft_genomi...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078575.1_Caulobacteraceae.draft_genomi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9_GCA_900078585.1_Arthrobacter.draft_genomic....</td>\n",
       "      <td>GCA_900078585.1_Arthrobacter.draft_genomic.fn...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078585.1_Arthrobacter.draft_genomic.fn...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10_GCA_900078595.1_Oxalobacteraceae.draft_gen...</td>\n",
       "      <td>GCA_900078595.1_Oxalobacteraceae.draft_genomi...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078595.1_Oxalobacteraceae.draft_genomi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11_GCA_900078605.1_Kitasatospora.draft_genomi...</td>\n",
       "      <td>GCA_900078605.1_Kitasatospora.draft_genomic.f...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078605.1_Kitasatospora.draft_genomic.f...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12_GCA_900078615.1_Opitutaceae.draft_genomic....</td>\n",
       "      <td>GCA_900078615.1_Opitutaceae.draft_genomic.fna...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078615.1_Opitutaceae.draft_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13_GCA_900078625.1_Candidatus.Saccharibacteri...</td>\n",
       "      <td>GCA_900078625.1_Candidatus.Saccharibacteria.d...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078625.1_Candidatus.Saccharibacteria.d...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14_GCA_900078635.1_Heterogeneous.bin_genomic....</td>\n",
       "      <td>GCA_900078635.1_Heterogeneous.bin_genomic.fna...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078635.1_Heterogeneous.bin_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15_GCA_900078645.1_Chthoniobacter.1.draft_gen...</td>\n",
       "      <td>GCA_900078645.1_Chthoniobacter.1.draft_genomi...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078645.1_Chthoniobacter.1.draft_genomi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>16_GCA_900078655.1_Cthoniobacter.2.draft_geno...</td>\n",
       "      <td>GCA_900078655.1_Cthoniobacter.2.draft_genomic...</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900078655.1_Cthoniobacter.2.draft_genomic...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>17_GCA_900091065.1_ASM90009106v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091065.1_ASM90009106v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091065.1_ASM90009106v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>18_GCA_900091075.1_ASM90009107v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091075.1_ASM90009107v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091075.1_ASM90009107v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19_GCA_900091085.1_ASM90009108v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091085.1_ASM90009108v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091085.1_ASM90009108v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20_GCA_900091095.1_ASM90009109v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091095.1_ASM90009109v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091095.1_ASM90009109v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>21_GCA_900091105.1_ASM90009110v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091105.1_ASM90009110v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091105.1_ASM90009110v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22_GCA_900091115.1_ASM90009111v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091115.1_ASM90009111v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091115.1_ASM90009111v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>23_GCA_900091125.1_ASM90009112v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091125.1_ASM90009112v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091125.1_ASM90009112v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>24_GCA_900091135.1_ASM90009113v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091135.1_ASM90009113v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091135.1_ASM90009113v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>25_GCA_900091145.1_ASM90009114v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091145.1_ASM90009114v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091145.1_ASM90009114v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>26_GCA_900091155.1_ASM90009115v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091155.1_ASM90009115v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091155.1_ASM90009115v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>27_GCA_900091165.1_ASM90009116v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091165.1_ASM90009116v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091165.1_ASM90009116v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>28_GCA_900091175.1_ASM90009117v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091175.1_ASM90009117v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091175.1_ASM90009117v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>29_GCA_900091185.1_ASM90009118v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091185.1_ASM90009118v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091185.1_ASM90009118v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>30_GCA_900091195.1_ASM90009119v1_genomic.fna.gz</td>\n",
       "      <td>GCA_900091195.1_ASM90009119v1_genomic.fna.gz</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>GCA_900091195.1_ASM90009119v1_genomic.fna.gz</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688159</th>\n",
       "      <td>688159_ZellerG_2014__CCIS27927933ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS27927933ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS27927933ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688160</th>\n",
       "      <td>688160_ZellerG_2014__CCIS21126322ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS21126322ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS21126322ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688161</th>\n",
       "      <td>688161_ZellerG_2014__CCIS88007743ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS88007743ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS88007743ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688162</th>\n",
       "      <td>688162_ZellerG_2014__CCIS46467422ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS46467422ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS46467422ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688163</th>\n",
       "      <td>688163_ZellerG_2014__CCIS07539127ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS07539127ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS07539127ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688164</th>\n",
       "      <td>688164_ZellerG_2014__CCIS29210128ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS29210128ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS29210128ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688165</th>\n",
       "      <td>688165_ZellerG_2014__CCIS07648107ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS07648107ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS07648107ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688166</th>\n",
       "      <td>688166_ZellerG_2014__CCIS12656533ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS12656533ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS12656533ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688167</th>\n",
       "      <td>688167_ZellerG_2014__CCIS94417875ST-3-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS94417875ST-3-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS94417875ST-3-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688168</th>\n",
       "      <td>688168_ZellerG_2014__MMPU99077057ST.fa.gz</td>\n",
       "      <td>ZellerG_2014__MMPU99077057ST.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__MMPU99077057ST.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688169</th>\n",
       "      <td>688169_ZellerG_2014__CCIS03473770ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS03473770ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS03473770ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688170</th>\n",
       "      <td>688170_ZellerG_2014__CCIS12370844ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS12370844ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS12370844ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688171</th>\n",
       "      <td>688171_ZellerG_2014__CCIS32105356ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS32105356ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS32105356ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688172</th>\n",
       "      <td>688172_ZellerG_2014__CCIS93040568ST-20-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS93040568ST-20-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS93040568ST-20-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688173</th>\n",
       "      <td>688173_ZellerG_2014__CCIS77100899ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS77100899ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS77100899ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688174</th>\n",
       "      <td>688174_ZellerG_2014__CCIS63448910ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS63448910ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS63448910ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688175</th>\n",
       "      <td>688175_ZellerG_2014__CCIS51667829ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS51667829ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS51667829ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688176</th>\n",
       "      <td>688176_ZellerG_2014__CCIS44757994ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS44757994ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS44757994ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688177</th>\n",
       "      <td>688177_ZellerG_2014__CCIS02856720ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS02856720ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS02856720ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688178</th>\n",
       "      <td>688178_ZellerG_2014__CCIS50471204ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS50471204ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS50471204ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688179</th>\n",
       "      <td>688179_ZellerG_2014__CCIS87290971ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS87290971ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS87290971ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688180</th>\n",
       "      <td>688180_ZellerG_2014__CCIS41548810ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS41548810ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS41548810ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688181</th>\n",
       "      <td>688181_ZellerG_2014__CCIS81735969ST-20-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS81735969ST-20-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS81735969ST-20-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688182</th>\n",
       "      <td>688182_ZellerG_2014__CCMD13934959ST-21-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCMD13934959ST-21-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCMD13934959ST-21-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688183</th>\n",
       "      <td>688183_ZellerG_2014__CCIS71301801ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS71301801ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS71301801ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688184</th>\n",
       "      <td>688184_ZellerG_2014__CCMD40701343ST-21-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCMD40701343ST-21-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCMD40701343ST-21-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688185</th>\n",
       "      <td>688185_ZellerG_2014__CCIS50369211ST-20-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS50369211ST-20-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS50369211ST-20-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688186</th>\n",
       "      <td>688186_ZellerG_2014__CCIS16561622ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS16561622ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS16561622ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688187</th>\n",
       "      <td>688187_ZellerG_2014__CCMD96553385ST-21-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCMD96553385ST-21-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCMD96553385ST-21-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688188</th>\n",
       "      <td>688188_ZellerG_2014__CCIS77252613ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014__CCIS77252613ST-4-0.fa.gz</td>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>ZellerG_2014__CCIS77252613ST-4-0.fa.gz</td>\n",
       "      <td>ZellerG_2014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>688188 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              filename_id  \\\n",
       "id                                                          \n",
       "1            1_GCA_000208265.2_ASM20826v2_genomic.fna.gz    \n",
       "2                 2_GCA_900007175.1_NOCA2_genomic.fna.gz    \n",
       "3                 3_GCA_900007185.1_NOCA1_genomic.fna.gz    \n",
       "4             4_GCA_900010595.2_55Contigs_genomic.fna.gz    \n",
       "5        5_GCA_900010795.1_Thauera_sp._DTG_genomic.fna...   \n",
       "6        6_GCA_900078555.1_M.thermophila.draft_genomic...   \n",
       "7        7_GCA_900078565.1_Herbaspirillum.draft_genomi...   \n",
       "8        8_GCA_900078575.1_Caulobacteraceae.draft_geno...   \n",
       "9        9_GCA_900078585.1_Arthrobacter.draft_genomic....   \n",
       "10       10_GCA_900078595.1_Oxalobacteraceae.draft_gen...   \n",
       "11       11_GCA_900078605.1_Kitasatospora.draft_genomi...   \n",
       "12       12_GCA_900078615.1_Opitutaceae.draft_genomic....   \n",
       "13       13_GCA_900078625.1_Candidatus.Saccharibacteri...   \n",
       "14       14_GCA_900078635.1_Heterogeneous.bin_genomic....   \n",
       "15       15_GCA_900078645.1_Chthoniobacter.1.draft_gen...   \n",
       "16       16_GCA_900078655.1_Cthoniobacter.2.draft_geno...   \n",
       "17       17_GCA_900091065.1_ASM90009106v1_genomic.fna.gz    \n",
       "18       18_GCA_900091075.1_ASM90009107v1_genomic.fna.gz    \n",
       "19       19_GCA_900091085.1_ASM90009108v1_genomic.fna.gz    \n",
       "20       20_GCA_900091095.1_ASM90009109v1_genomic.fna.gz    \n",
       "21       21_GCA_900091105.1_ASM90009110v1_genomic.fna.gz    \n",
       "22       22_GCA_900091115.1_ASM90009111v1_genomic.fna.gz    \n",
       "23       23_GCA_900091125.1_ASM90009112v1_genomic.fna.gz    \n",
       "24       24_GCA_900091135.1_ASM90009113v1_genomic.fna.gz    \n",
       "25       25_GCA_900091145.1_ASM90009114v1_genomic.fna.gz    \n",
       "26       26_GCA_900091155.1_ASM90009115v1_genomic.fna.gz    \n",
       "27       27_GCA_900091165.1_ASM90009116v1_genomic.fna.gz    \n",
       "28       28_GCA_900091175.1_ASM90009117v1_genomic.fna.gz    \n",
       "29       29_GCA_900091185.1_ASM90009118v1_genomic.fna.gz    \n",
       "30       30_GCA_900091195.1_ASM90009119v1_genomic.fna.gz    \n",
       "...                                                   ...   \n",
       "688159     688159_ZellerG_2014__CCIS27927933ST-4-0.fa.gz    \n",
       "688160     688160_ZellerG_2014__CCIS21126322ST-4-0.fa.gz    \n",
       "688161     688161_ZellerG_2014__CCIS88007743ST-4-0.fa.gz    \n",
       "688162     688162_ZellerG_2014__CCIS46467422ST-4-0.fa.gz    \n",
       "688163     688163_ZellerG_2014__CCIS07539127ST-4-0.fa.gz    \n",
       "688164     688164_ZellerG_2014__CCIS29210128ST-4-0.fa.gz    \n",
       "688165     688165_ZellerG_2014__CCIS07648107ST-4-0.fa.gz    \n",
       "688166     688166_ZellerG_2014__CCIS12656533ST-4-0.fa.gz    \n",
       "688167     688167_ZellerG_2014__CCIS94417875ST-3-0.fa.gz    \n",
       "688168         688168_ZellerG_2014__MMPU99077057ST.fa.gz    \n",
       "688169     688169_ZellerG_2014__CCIS03473770ST-4-0.fa.gz    \n",
       "688170     688170_ZellerG_2014__CCIS12370844ST-4-0.fa.gz    \n",
       "688171     688171_ZellerG_2014__CCIS32105356ST-4-0.fa.gz    \n",
       "688172    688172_ZellerG_2014__CCIS93040568ST-20-0.fa.gz    \n",
       "688173     688173_ZellerG_2014__CCIS77100899ST-4-0.fa.gz    \n",
       "688174     688174_ZellerG_2014__CCIS63448910ST-4-0.fa.gz    \n",
       "688175     688175_ZellerG_2014__CCIS51667829ST-4-0.fa.gz    \n",
       "688176     688176_ZellerG_2014__CCIS44757994ST-4-0.fa.gz    \n",
       "688177     688177_ZellerG_2014__CCIS02856720ST-4-0.fa.gz    \n",
       "688178     688178_ZellerG_2014__CCIS50471204ST-4-0.fa.gz    \n",
       "688179     688179_ZellerG_2014__CCIS87290971ST-4-0.fa.gz    \n",
       "688180     688180_ZellerG_2014__CCIS41548810ST-4-0.fa.gz    \n",
       "688181    688181_ZellerG_2014__CCIS81735969ST-20-0.fa.gz    \n",
       "688182    688182_ZellerG_2014__CCMD13934959ST-21-0.fa.gz    \n",
       "688183     688183_ZellerG_2014__CCIS71301801ST-4-0.fa.gz    \n",
       "688184    688184_ZellerG_2014__CCMD40701343ST-21-0.fa.gz    \n",
       "688185    688185_ZellerG_2014__CCIS50369211ST-20-0.fa.gz    \n",
       "688186     688186_ZellerG_2014__CCIS16561622ST-4-0.fa.gz    \n",
       "688187    688187_ZellerG_2014__CCMD96553385ST-21-0.fa.gz    \n",
       "688188     688188_ZellerG_2014__CCIS77252613ST-4-0.fa.gz    \n",
       "\n",
       "                                                 filename          database  \\\n",
       "id                                                                            \n",
       "1              GCA_000208265.2_ASM20826v2_genomic.fna.gz              NCBI    \n",
       "2                   GCA_900007175.1_NOCA2_genomic.fna.gz              NCBI    \n",
       "3                   GCA_900007185.1_NOCA1_genomic.fna.gz              NCBI    \n",
       "4               GCA_900010595.2_55Contigs_genomic.fna.gz              NCBI    \n",
       "5         GCA_900010795.1_Thauera_sp._DTG_genomic.fna.gz              NCBI    \n",
       "6        GCA_900078555.1_M.thermophila.draft_genomic.f...             NCBI    \n",
       "7        GCA_900078565.1_Herbaspirillum.draft_genomic....             NCBI    \n",
       "8        GCA_900078575.1_Caulobacteraceae.draft_genomi...             NCBI    \n",
       "9        GCA_900078585.1_Arthrobacter.draft_genomic.fn...             NCBI    \n",
       "10       GCA_900078595.1_Oxalobacteraceae.draft_genomi...             NCBI    \n",
       "11       GCA_900078605.1_Kitasatospora.draft_genomic.f...             NCBI    \n",
       "12       GCA_900078615.1_Opitutaceae.draft_genomic.fna...             NCBI    \n",
       "13       GCA_900078625.1_Candidatus.Saccharibacteria.d...             NCBI    \n",
       "14       GCA_900078635.1_Heterogeneous.bin_genomic.fna...             NCBI    \n",
       "15       GCA_900078645.1_Chthoniobacter.1.draft_genomi...             NCBI    \n",
       "16       GCA_900078655.1_Cthoniobacter.2.draft_genomic...             NCBI    \n",
       "17          GCA_900091065.1_ASM90009106v1_genomic.fna.gz              NCBI    \n",
       "18          GCA_900091075.1_ASM90009107v1_genomic.fna.gz              NCBI    \n",
       "19          GCA_900091085.1_ASM90009108v1_genomic.fna.gz              NCBI    \n",
       "20          GCA_900091095.1_ASM90009109v1_genomic.fna.gz              NCBI    \n",
       "21          GCA_900091105.1_ASM90009110v1_genomic.fna.gz              NCBI    \n",
       "22          GCA_900091115.1_ASM90009111v1_genomic.fna.gz              NCBI    \n",
       "23          GCA_900091125.1_ASM90009112v1_genomic.fna.gz              NCBI    \n",
       "24          GCA_900091135.1_ASM90009113v1_genomic.fna.gz              NCBI    \n",
       "25          GCA_900091145.1_ASM90009114v1_genomic.fna.gz              NCBI    \n",
       "26          GCA_900091155.1_ASM90009115v1_genomic.fna.gz              NCBI    \n",
       "27          GCA_900091165.1_ASM90009116v1_genomic.fna.gz              NCBI    \n",
       "28          GCA_900091175.1_ASM90009117v1_genomic.fna.gz              NCBI    \n",
       "29          GCA_900091185.1_ASM90009118v1_genomic.fna.gz              NCBI    \n",
       "30          GCA_900091195.1_ASM90009119v1_genomic.fna.gz              NCBI    \n",
       "...                                                   ...               ...   \n",
       "688159            ZellerG_2014__CCIS27927933ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688160            ZellerG_2014__CCIS21126322ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688161            ZellerG_2014__CCIS88007743ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688162            ZellerG_2014__CCIS46467422ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688163            ZellerG_2014__CCIS07539127ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688164            ZellerG_2014__CCIS29210128ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688165            ZellerG_2014__CCIS07648107ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688166            ZellerG_2014__CCIS12656533ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688167            ZellerG_2014__CCIS94417875ST-3-0.fa.gz    Segata-Pasolli    \n",
       "688168                ZellerG_2014__MMPU99077057ST.fa.gz    Segata-Pasolli    \n",
       "688169            ZellerG_2014__CCIS03473770ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688170            ZellerG_2014__CCIS12370844ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688171            ZellerG_2014__CCIS32105356ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688172           ZellerG_2014__CCIS93040568ST-20-0.fa.gz    Segata-Pasolli    \n",
       "688173            ZellerG_2014__CCIS77100899ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688174            ZellerG_2014__CCIS63448910ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688175            ZellerG_2014__CCIS51667829ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688176            ZellerG_2014__CCIS44757994ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688177            ZellerG_2014__CCIS02856720ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688178            ZellerG_2014__CCIS50471204ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688179            ZellerG_2014__CCIS87290971ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688180            ZellerG_2014__CCIS41548810ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688181           ZellerG_2014__CCIS81735969ST-20-0.fa.gz    Segata-Pasolli    \n",
       "688182           ZellerG_2014__CCMD13934959ST-21-0.fa.gz    Segata-Pasolli    \n",
       "688183            ZellerG_2014__CCIS71301801ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688184           ZellerG_2014__CCMD40701343ST-21-0.fa.gz    Segata-Pasolli    \n",
       "688185           ZellerG_2014__CCIS50369211ST-20-0.fa.gz    Segata-Pasolli    \n",
       "688186            ZellerG_2014__CCIS16561622ST-4-0.fa.gz    Segata-Pasolli    \n",
       "688187           ZellerG_2014__CCMD96553385ST-21-0.fa.gz    Segata-Pasolli    \n",
       "688188            ZellerG_2014__CCIS77252613ST-4-0.fa.gz    Segata-Pasolli    \n",
       "\n",
       "        sub_database                                         filename_2  \\\n",
       "id                                                                        \n",
       "1        metagenome           GCA_000208265.2_ASM20826v2_genomic.fna.gz   \n",
       "2        metagenome                GCA_900007175.1_NOCA2_genomic.fna.gz   \n",
       "3        metagenome                GCA_900007185.1_NOCA1_genomic.fna.gz   \n",
       "4        metagenome            GCA_900010595.2_55Contigs_genomic.fna.gz   \n",
       "5        metagenome      GCA_900010795.1_Thauera_sp._DTG_genomic.fna.gz   \n",
       "6        metagenome    GCA_900078555.1_M.thermophila.draft_genomic.f...   \n",
       "7        metagenome    GCA_900078565.1_Herbaspirillum.draft_genomic....   \n",
       "8        metagenome    GCA_900078575.1_Caulobacteraceae.draft_genomi...   \n",
       "9        metagenome    GCA_900078585.1_Arthrobacter.draft_genomic.fn...   \n",
       "10       metagenome    GCA_900078595.1_Oxalobacteraceae.draft_genomi...   \n",
       "11       metagenome    GCA_900078605.1_Kitasatospora.draft_genomic.f...   \n",
       "12       metagenome    GCA_900078615.1_Opitutaceae.draft_genomic.fna.gz   \n",
       "13       metagenome    GCA_900078625.1_Candidatus.Saccharibacteria.d...   \n",
       "14       metagenome    GCA_900078635.1_Heterogeneous.bin_genomic.fna.gz   \n",
       "15       metagenome    GCA_900078645.1_Chthoniobacter.1.draft_genomi...   \n",
       "16       metagenome    GCA_900078655.1_Cthoniobacter.2.draft_genomic...   \n",
       "17       metagenome        GCA_900091065.1_ASM90009106v1_genomic.fna.gz   \n",
       "18       metagenome        GCA_900091075.1_ASM90009107v1_genomic.fna.gz   \n",
       "19       metagenome        GCA_900091085.1_ASM90009108v1_genomic.fna.gz   \n",
       "20       metagenome        GCA_900091095.1_ASM90009109v1_genomic.fna.gz   \n",
       "21       metagenome        GCA_900091105.1_ASM90009110v1_genomic.fna.gz   \n",
       "22       metagenome        GCA_900091115.1_ASM90009111v1_genomic.fna.gz   \n",
       "23       metagenome        GCA_900091125.1_ASM90009112v1_genomic.fna.gz   \n",
       "24       metagenome        GCA_900091135.1_ASM90009113v1_genomic.fna.gz   \n",
       "25       metagenome        GCA_900091145.1_ASM90009114v1_genomic.fna.gz   \n",
       "26       metagenome        GCA_900091155.1_ASM90009115v1_genomic.fna.gz   \n",
       "27       metagenome        GCA_900091165.1_ASM90009116v1_genomic.fna.gz   \n",
       "28       metagenome        GCA_900091175.1_ASM90009117v1_genomic.fna.gz   \n",
       "29       metagenome        GCA_900091185.1_ASM90009118v1_genomic.fna.gz   \n",
       "30       metagenome        GCA_900091195.1_ASM90009119v1_genomic.fna.gz   \n",
       "...              ...                                                ...   \n",
       "688159   metagenome             ZellerG_2014__CCIS27927933ST-4-0.fa.gz    \n",
       "688160   metagenome             ZellerG_2014__CCIS21126322ST-4-0.fa.gz    \n",
       "688161   metagenome             ZellerG_2014__CCIS88007743ST-4-0.fa.gz    \n",
       "688162   metagenome             ZellerG_2014__CCIS46467422ST-4-0.fa.gz    \n",
       "688163   metagenome             ZellerG_2014__CCIS07539127ST-4-0.fa.gz    \n",
       "688164   metagenome             ZellerG_2014__CCIS29210128ST-4-0.fa.gz    \n",
       "688165   metagenome             ZellerG_2014__CCIS07648107ST-4-0.fa.gz    \n",
       "688166   metagenome             ZellerG_2014__CCIS12656533ST-4-0.fa.gz    \n",
       "688167   metagenome             ZellerG_2014__CCIS94417875ST-3-0.fa.gz    \n",
       "688168   metagenome                 ZellerG_2014__MMPU99077057ST.fa.gz    \n",
       "688169   metagenome             ZellerG_2014__CCIS03473770ST-4-0.fa.gz    \n",
       "688170   metagenome             ZellerG_2014__CCIS12370844ST-4-0.fa.gz    \n",
       "688171   metagenome             ZellerG_2014__CCIS32105356ST-4-0.fa.gz    \n",
       "688172   metagenome            ZellerG_2014__CCIS93040568ST-20-0.fa.gz    \n",
       "688173   metagenome             ZellerG_2014__CCIS77100899ST-4-0.fa.gz    \n",
       "688174   metagenome             ZellerG_2014__CCIS63448910ST-4-0.fa.gz    \n",
       "688175   metagenome             ZellerG_2014__CCIS51667829ST-4-0.fa.gz    \n",
       "688176   metagenome             ZellerG_2014__CCIS44757994ST-4-0.fa.gz    \n",
       "688177   metagenome             ZellerG_2014__CCIS02856720ST-4-0.fa.gz    \n",
       "688178   metagenome             ZellerG_2014__CCIS50471204ST-4-0.fa.gz    \n",
       "688179   metagenome             ZellerG_2014__CCIS87290971ST-4-0.fa.gz    \n",
       "688180   metagenome             ZellerG_2014__CCIS41548810ST-4-0.fa.gz    \n",
       "688181   metagenome            ZellerG_2014__CCIS81735969ST-20-0.fa.gz    \n",
       "688182   metagenome            ZellerG_2014__CCMD13934959ST-21-0.fa.gz    \n",
       "688183   metagenome             ZellerG_2014__CCIS71301801ST-4-0.fa.gz    \n",
       "688184   metagenome            ZellerG_2014__CCMD40701343ST-21-0.fa.gz    \n",
       "688185   metagenome            ZellerG_2014__CCIS50369211ST-20-0.fa.gz    \n",
       "688186   metagenome             ZellerG_2014__CCIS16561622ST-4-0.fa.gz    \n",
       "688187   metagenome            ZellerG_2014__CCMD96553385ST-21-0.fa.gz    \n",
       "688188   metagenome             ZellerG_2014__CCIS77252613ST-4-0.fa.gz    \n",
       "\n",
       "                    7  \n",
       "id                     \n",
       "1                 NaN  \n",
       "2                 NaN  \n",
       "3                 NaN  \n",
       "4                 NaN  \n",
       "5                 NaN  \n",
       "6                 NaN  \n",
       "7                 NaN  \n",
       "8                 NaN  \n",
       "9                 NaN  \n",
       "10                NaN  \n",
       "11                NaN  \n",
       "12                NaN  \n",
       "13                NaN  \n",
       "14                NaN  \n",
       "15                NaN  \n",
       "16                NaN  \n",
       "17                NaN  \n",
       "18                NaN  \n",
       "19                NaN  \n",
       "20                NaN  \n",
       "21                NaN  \n",
       "22                NaN  \n",
       "23                NaN  \n",
       "24                NaN  \n",
       "25                NaN  \n",
       "26                NaN  \n",
       "27                NaN  \n",
       "28                NaN  \n",
       "29                NaN  \n",
       "30                NaN  \n",
       "...               ...  \n",
       "688159   ZellerG_2014  \n",
       "688160   ZellerG_2014  \n",
       "688161   ZellerG_2014  \n",
       "688162   ZellerG_2014  \n",
       "688163   ZellerG_2014  \n",
       "688164   ZellerG_2014  \n",
       "688165   ZellerG_2014  \n",
       "688166   ZellerG_2014  \n",
       "688167   ZellerG_2014  \n",
       "688168   ZellerG_2014  \n",
       "688169   ZellerG_2014  \n",
       "688170   ZellerG_2014  \n",
       "688171   ZellerG_2014  \n",
       "688172   ZellerG_2014  \n",
       "688173   ZellerG_2014  \n",
       "688174   ZellerG_2014  \n",
       "688175   ZellerG_2014  \n",
       "688176   ZellerG_2014  \n",
       "688177   ZellerG_2014  \n",
       "688178   ZellerG_2014  \n",
       "688179   ZellerG_2014  \n",
       "688180   ZellerG_2014  \n",
       "688181   ZellerG_2014  \n",
       "688182   ZellerG_2014  \n",
       "688183   ZellerG_2014  \n",
       "688184   ZellerG_2014  \n",
       "688185   ZellerG_2014  \n",
       "688186   ZellerG_2014  \n",
       "688187   ZellerG_2014  \n",
       "688188   ZellerG_2014  \n",
       "\n",
       "[688188 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.set_index(\"id\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db=data.groupby(\"database\")[\"id\"].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub=data.groupby([\"database\",\"sub_database\"])[\"id\"].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "## importing graph stuff\n",
    "from math import pi\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "from bokeh.io import output_file, show, output_notebook\n",
    "from bokeh.palettes import Category20c, Category10, Blues, Greens, Oranges, Reds, RdPu\n",
    "from bokeh.plotting import figure\n",
    "from bokeh.transform import cumsum\n",
    "from bokeh.models import HoverTool, BoxZoomTool, PanTool, WheelZoomTool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "color_dict={'NCBI': Blues[3].copy(),\n",
    " 'JGI': Oranges[6].copy(),\n",
    " 'ENA': Greens[3].copy(),\n",
    " 'ensembl': [ \"#d62728\"],\n",
    " 'MGnify': [\"#9467bd\"],\n",
    " 'Metasub': [\"#8c564b\"],\n",
    " 'Segata-Pasolli': [\"#e377c2\"]}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Amount of files\n",
    "### Plot by database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db=by_db.reset_index(name='value').rename(columns={'index':'database'})\n",
    "#by_db['angle'] = by_db['value']/by_db['value'].sum() * 2*pi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db['angle'] = by_db['value']/by_db['value'].sum() * 2*pi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>198435</td>\n",
       "      <td>1.811720</td>\n",
       "      <td>#3182bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JGI</td>\n",
       "      <td>48938</td>\n",
       "      <td>0.446806</td>\n",
       "      <td>#a63603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>4644</td>\n",
       "      <td>0.042400</td>\n",
       "      <td>#31a354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>3840</td>\n",
       "      <td>0.035059</td>\n",
       "      <td>#d62728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>329715</td>\n",
       "      <td>3.010312</td>\n",
       "      <td>#9467bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>9395</td>\n",
       "      <td>0.085777</td>\n",
       "      <td>#8c564b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>93221</td>\n",
       "      <td>0.851112</td>\n",
       "      <td>#e377c2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           database   value     angle    color\n",
       "0              ENA   198435  1.811720  #3182bd\n",
       "1              JGI    48938  0.446806  #a63603\n",
       "2           MGnify     4644  0.042400  #31a354\n",
       "3          Metasub     3840  0.035059  #d62728\n",
       "4             NCBI   329715  3.010312  #9467bd\n",
       "5   Segata-Pasolli     9395  0.085777  #8c564b\n",
       "6          ensembl    93221  0.851112  #e377c2"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "colors= [i[0] for i in color_dict.values()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['#3182bd', '#a63603', '#31a354', '#d62728', '#9467bd', '#8c564b', '#e377c2']"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"1602\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"1602\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n",
       "    }\n",
       "    finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.info(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(js_urls, callback) {\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = js_urls.length;\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var s = document.createElement('script');\n",
       "      s.src = url;\n",
       "      s.async = false;\n",
       "      s.onreadystatechange = s.onload = function() {\n",
       "        root._bokeh_is_loading--;\n",
       "        if (root._bokeh_is_loading === 0) {\n",
       "          console.log(\"Bokeh: all BokehJS libraries loaded\");\n",
       "          run_callbacks()\n",
       "        }\n",
       "      };\n",
       "      s.onerror = function() {\n",
       "        console.warn(\"failed to load library \" + url);\n",
       "      };\n",
       "      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "    }\n",
       "  };var element = document.getElementById(\"1602\");\n",
       "  if (element == null) {\n",
       "    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '1602' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    \n",
       "    function(Bokeh) {\n",
       "      \n",
       "    },\n",
       "    function(Bokeh) {\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if ((root.Bokeh !== undefined) || (force === true)) {\n",
       "      for (var i = 0; i < inline_js.length; i++) {\n",
       "        inline_js[i].call(root, root.Bokeh);\n",
       "      }if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"1602\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(js_urls, function() {\n",
       "      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"1602\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n    }\n    finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.info(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(js_urls, callback) {\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = js_urls.length;\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var s = document.createElement('script');\n      s.src = url;\n      s.async = false;\n      s.onreadystatechange = s.onload = function() {\n        root._bokeh_is_loading--;\n        if (root._bokeh_is_loading === 0) {\n          console.log(\"Bokeh: all BokehJS libraries loaded\");\n          run_callbacks()\n        }\n      };\n      s.onerror = function() {\n        console.warn(\"failed to load library \" + url);\n      };\n      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.getElementsByTagName(\"head\")[0].appendChild(s);\n    }\n  };var element = document.getElementById(\"1602\");\n  if (element == null) {\n    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '1602' but no matching script tag was found. \")\n    return false;\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n    }\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"1602\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(js_urls, function() {\n      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"ccf5d5ec-d5a3-4f50-bed7-d91c267b645b\" data-root-id=\"1604\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"206cec9f-9ff2-40b2-ac53-d982f89587c0\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"1614\",\"type\":\"LinearAxis\"}],\"left\":[{\"id\":\"1619\",\"type\":\"LinearAxis\"}],\"plot_height\":500,\"plot_width\":800,\"renderers\":[{\"id\":\"1614\",\"type\":\"LinearAxis\"},{\"id\":\"1618\",\"type\":\"Grid\"},{\"id\":\"1619\",\"type\":\"LinearAxis\"},{\"id\":\"1623\",\"type\":\"Grid\"},{\"id\":\"1640\",\"type\":\"Legend\"},{\"id\":\"1633\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"1603\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"1625\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"1606\",\"type\":\"DataRange1d\"},\"x_scale\":{\"id\":\"1610\",\"type\":\"LinearScale\"},\"y_range\":{\"id\":\"1608\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"1612\",\"type\":\"LinearScale\"}},\"id\":\"1604\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"source\":{\"id\":\"1629\",\"type\":\"ColumnDataSource\"}},\"id\":\"1634\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"1612\",\"type\":\"LinearScale\"},{\"attributes\":{\"formatter\":{\"id\":\"1636\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"1604\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1615\",\"type\":\"BasicTicker\"}},\"id\":\"1614\",\"type\":\"LinearAxis\"},{\"attributes\":{},\"id\":\"1636\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"1638\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"plot\":{\"id\":\"1604\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1615\",\"type\":\"BasicTicker\"}},\"id\":\"1618\",\"type\":\"Grid\"},{\"attributes\":{\"items\":[{\"id\":\"1641\",\"type\":\"LegendItem\"}],\"plot\":{\"id\":\"1604\",\"subtype\":\"Figure\",\"type\":\"Plot\"}},\"id\":\"1640\",\"type\":\"Legend\"},{\"attributes\":{},\"id\":\"1615\",\"type\":\"BasicTicker\"},{\"attributes\":{\"formatter\":{\"id\":\"1638\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"1604\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1620\",\"type\":\"BasicTicker\"}},\"id\":\"1619\",\"type\":\"LinearAxis\"},{\"attributes\":{\"plot\":null,\"text\":\"Genomic Data on Server\"},\"id\":\"1603\",\"type\":\"Title\"},{\"attributes\":{\"label\":{\"field\":\"database\"},\"renderers\":[{\"id\":\"1633\",\"type\":\"GlyphRenderer\"}]},\"id\":\"1641\",\"type\":\"LegendItem\"},{\"attributes\":{},\"id\":\"1610\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"1620\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"1647\",\"type\":\"Selection\"},{\"attributes\":{\"dimension\":1,\"plot\":{\"id\":\"1604\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1620\",\"type\":\"BasicTicker\"}},\"id\":\"1623\",\"type\":\"Grid\"},{\"attributes\":{\"callback\":null},\"id\":\"1608\",\"type\":\"DataRange1d\"},{\"attributes\":{},\"id\":\"1648\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"callback\":null},\"id\":\"1606\",\"type\":\"DataRange1d\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"1624\",\"type\":\"HoverTool\"}]},\"id\":\"1625\",\"type\":\"Toolbar\"},{\"attributes\":{\"callback\":null,\"tooltips\":\"@database: @value\"},\"id\":\"1624\",\"type\":\"HoverTool\"},{\"attributes\":{\"field\":\"angle\"},\"id\":\"1628\",\"type\":\"CumSum\"},{\"attributes\":{\"callback\":null,\"data\":{\"angle\":{\"__ndarray__\":\"1LVv+s38/D8GyqopeJjcPyci1w5xtaU/zP948kzzoT+N/b1XHhUIQEBanvZ29bU/edSLek486z8=\",\"dtype\":\"float64\",\"shape\":[7]},\"color\":[\"#3182bd\",\"#a63603\",\"#31a354\",\"#d62728\",\"#9467bd\",\"#8c564b\",\"#e377c2\"],\"database\":[\" ENA \",\" JGI \",\" MGnify \",\" Metasub \",\" NCBI \",\" Segata-Pasolli \",\" ensembl \"],\"index\":[0,1,2,3,4,5,6],\"value\":[198435,48938,4644,3840,329715,9395,93221]},\"selected\":{\"id\":\"1647\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"1648\",\"type\":\"UnionRenderers\"}},\"id\":\"1629\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"data_source\":{\"id\":\"1629\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"1631\",\"type\":\"Wedge\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1632\",\"type\":\"Wedge\"},\"selection_glyph\":null,\"view\":{\"id\":\"1634\",\"type\":\"CDSView\"}},\"id\":\"1633\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"field\":\"angle\",\"include_zero\":true},\"id\":\"1627\",\"type\":\"CumSum\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"1628\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"1627\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"1632\",\"type\":\"Wedge\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"1628\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_color\":{\"field\":\"color\"},\"line_color\":{\"value\":\"white\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"1627\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"1631\",\"type\":\"Wedge\"}],\"root_ids\":[\"1604\"]},\"title\":\"Bokeh Application\",\"version\":\"1.0.4\"}};\n",
       "  var render_items = [{\"docid\":\"206cec9f-9ff2-40b2-ac53-d982f89587c0\",\"roots\":{\"1604\":\"ccf5d5ec-d5a3-4f50-bed7-d91c267b645b\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "1604"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "output_notebook();\n",
    "by_db['color'] = colors\n",
    "\n",
    "p = figure(plot_height=500, plot_width=800,title=\"Genomic Data on Server\", toolbar_location=None,\n",
    "        tools=\"hover\", tooltips=\"@database: @value\")\n",
    "\n",
    "p.wedge(x=0, y=1, radius=0.4,\n",
    "        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),\n",
    "        line_color=\"white\", fill_color='color', legend='database', source=by_db)\n",
    "\n",
    "show(p);"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### By subdatabase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub.reset_index(name='value').rename(columns={'index':'sub_database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot['angle'] = by_sub_plot['value']/by_sub_plot['value'].sum() * 2*pi\n",
    "by_sub_plot['complete_database'] = by_sub_plot['database']+by_sub_plot['sub_database']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "      <th>complete_database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>51876</td>\n",
       "      <td>0.473630</td>\n",
       "      <td>ENA  env</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>146559</td>\n",
       "      <td>1.338090</td>\n",
       "      <td>ENA  pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>777</td>\n",
       "      <td>0.007094</td>\n",
       "      <td>JGI  JGI_archaea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>181</td>\n",
       "      <td>0.001653</td>\n",
       "      <td>JGI  JGI_archaea_v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>42226</td>\n",
       "      <td>0.385525</td>\n",
       "      <td>JGI  JGI_bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>682</td>\n",
       "      <td>0.006227</td>\n",
       "      <td>JGI  JGI_bacteria_v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>4900</td>\n",
       "      <td>0.044737</td>\n",
       "      <td>JGI  JGI_metagenomes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>172</td>\n",
       "      <td>0.001570</td>\n",
       "      <td>JGI  JGI_metagenomes_v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>4644</td>\n",
       "      <td>0.042400</td>\n",
       "      <td>MGnify  bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>3840</td>\n",
       "      <td>0.035059</td>\n",
       "      <td>Metasub  bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>3541</td>\n",
       "      <td>0.032329</td>\n",
       "      <td>NCBI  archaea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>315772</td>\n",
       "      <td>2.883012</td>\n",
       "      <td>NCBI  bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>10402</td>\n",
       "      <td>0.094971</td>\n",
       "      <td>NCBI  metagenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>9395</td>\n",
       "      <td>0.085777</td>\n",
       "      <td>Segata-Pasolli  metagenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>93221</td>\n",
       "      <td>0.851112</td>\n",
       "      <td>ensembl  bacteria</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database   value     angle  \\\n",
       "0               ENA                   env    51876  0.473630   \n",
       "1               ENA                   pro   146559  1.338090   \n",
       "2               JGI           JGI_archaea      777  0.007094   \n",
       "3               JGI        JGI_archaea_v2      181  0.001653   \n",
       "4               JGI          JGI_bacteria    42226  0.385525   \n",
       "5               JGI       JGI_bacteria_v2      682  0.006227   \n",
       "6               JGI       JGI_metagenomes     4900  0.044737   \n",
       "7               JGI    JGI_metagenomes_v2      172  0.001570   \n",
       "8            MGnify              bacteria     4644  0.042400   \n",
       "9           Metasub              bacteria     3840  0.035059   \n",
       "10             NCBI               archaea     3541  0.032329   \n",
       "11             NCBI              bacteria   315772  2.883012   \n",
       "12             NCBI            metagenome    10402  0.094971   \n",
       "13   Segata-Pasolli            metagenome     9395  0.085777   \n",
       "14          ensembl              bacteria    93221  0.851112   \n",
       "\n",
       "               complete_database  \n",
       "0                      ENA  env   \n",
       "1                      ENA  pro   \n",
       "2              JGI  JGI_archaea   \n",
       "3           JGI  JGI_archaea_v2   \n",
       "4             JGI  JGI_bacteria   \n",
       "5          JGI  JGI_bacteria_v2   \n",
       "6          JGI  JGI_metagenomes   \n",
       "7       JGI  JGI_metagenomes_v2   \n",
       "8              MGnify  bacteria   \n",
       "9             Metasub  bacteria   \n",
       "10                NCBI  archaea   \n",
       "11               NCBI  bacteria   \n",
       "12             NCBI  metagenome   \n",
       "13   Segata-Pasolli  metagenome   \n",
       "14            ensembl  bacteria   "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\ipykernel_launcher.py:7: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  import sys\n"
     ]
    }
   ],
   "source": [
    "for index, row in by_sub_plot.iterrows():\n",
    "    #print (color_dict)\n",
    "    \n",
    "    db_name=row['database'].split()[0]\n",
    "    \n",
    "    co=color_dict[db_name].pop()\n",
    "    by_sub_plot.set_value(index,\"color\",co)\n",
    "\n",
    "#by_sub_plot = by_sub_plot.sort_values(by=\"database\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "      <th>complete_database</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>51876</td>\n",
       "      <td>0.473630</td>\n",
       "      <td>ENA  env</td>\n",
       "      <td>#e5f5e0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>146559</td>\n",
       "      <td>1.338090</td>\n",
       "      <td>ENA  pro</td>\n",
       "      <td>#a1d99b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>777</td>\n",
       "      <td>0.007094</td>\n",
       "      <td>JGI  JGI_archaea</td>\n",
       "      <td>#feedde</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>181</td>\n",
       "      <td>0.001653</td>\n",
       "      <td>JGI  JGI_archaea_v2</td>\n",
       "      <td>#fdd0a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>42226</td>\n",
       "      <td>0.385525</td>\n",
       "      <td>JGI  JGI_bacteria</td>\n",
       "      <td>#fdae6b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>682</td>\n",
       "      <td>0.006227</td>\n",
       "      <td>JGI  JGI_bacteria_v2</td>\n",
       "      <td>#fd8d3c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>4900</td>\n",
       "      <td>0.044737</td>\n",
       "      <td>JGI  JGI_metagenomes</td>\n",
       "      <td>#e6550d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>172</td>\n",
       "      <td>0.001570</td>\n",
       "      <td>JGI  JGI_metagenomes_v2</td>\n",
       "      <td>#a63603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>4644</td>\n",
       "      <td>0.042400</td>\n",
       "      <td>MGnify  bacteria</td>\n",
       "      <td>#9467bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>3840</td>\n",
       "      <td>0.035059</td>\n",
       "      <td>Metasub  bacteria</td>\n",
       "      <td>#8c564b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>3541</td>\n",
       "      <td>0.032329</td>\n",
       "      <td>NCBI  archaea</td>\n",
       "      <td>#deebf7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>315772</td>\n",
       "      <td>2.883012</td>\n",
       "      <td>NCBI  bacteria</td>\n",
       "      <td>#9ecae1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>10402</td>\n",
       "      <td>0.094971</td>\n",
       "      <td>NCBI  metagenome</td>\n",
       "      <td>#3182bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>9395</td>\n",
       "      <td>0.085777</td>\n",
       "      <td>Segata-Pasolli  metagenome</td>\n",
       "      <td>#e377c2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>93221</td>\n",
       "      <td>0.851112</td>\n",
       "      <td>ensembl  bacteria</td>\n",
       "      <td>#d62728</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database   value     angle  \\\n",
       "0               ENA                   env    51876  0.473630   \n",
       "1               ENA                   pro   146559  1.338090   \n",
       "2               JGI           JGI_archaea      777  0.007094   \n",
       "3               JGI        JGI_archaea_v2      181  0.001653   \n",
       "4               JGI          JGI_bacteria    42226  0.385525   \n",
       "5               JGI       JGI_bacteria_v2      682  0.006227   \n",
       "6               JGI       JGI_metagenomes     4900  0.044737   \n",
       "7               JGI    JGI_metagenomes_v2      172  0.001570   \n",
       "8            MGnify              bacteria     4644  0.042400   \n",
       "9           Metasub              bacteria     3840  0.035059   \n",
       "10             NCBI               archaea     3541  0.032329   \n",
       "11             NCBI              bacteria   315772  2.883012   \n",
       "12             NCBI            metagenome    10402  0.094971   \n",
       "13   Segata-Pasolli            metagenome     9395  0.085777   \n",
       "14          ensembl              bacteria    93221  0.851112   \n",
       "\n",
       "               complete_database    color  \n",
       "0                      ENA  env   #e5f5e0  \n",
       "1                      ENA  pro   #a1d99b  \n",
       "2              JGI  JGI_archaea   #feedde  \n",
       "3           JGI  JGI_archaea_v2   #fdd0a2  \n",
       "4             JGI  JGI_bacteria   #fdae6b  \n",
       "5          JGI  JGI_bacteria_v2   #fd8d3c  \n",
       "6          JGI  JGI_metagenomes   #e6550d  \n",
       "7       JGI  JGI_metagenomes_v2   #a63603  \n",
       "8              MGnify  bacteria   #9467bd  \n",
       "9             Metasub  bacteria   #8c564b  \n",
       "10                NCBI  archaea   #deebf7  \n",
       "11               NCBI  bacteria   #9ecae1  \n",
       "12             NCBI  metagenome   #3182bd  \n",
       "13   Segata-Pasolli  metagenome   #e377c2  \n",
       "14            ensembl  bacteria   #d62728  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"1890\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"1890\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n",
       "    }\n",
       "    finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.info(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(js_urls, callback) {\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = js_urls.length;\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var s = document.createElement('script');\n",
       "      s.src = url;\n",
       "      s.async = false;\n",
       "      s.onreadystatechange = s.onload = function() {\n",
       "        root._bokeh_is_loading--;\n",
       "        if (root._bokeh_is_loading === 0) {\n",
       "          console.log(\"Bokeh: all BokehJS libraries loaded\");\n",
       "          run_callbacks()\n",
       "        }\n",
       "      };\n",
       "      s.onerror = function() {\n",
       "        console.warn(\"failed to load library \" + url);\n",
       "      };\n",
       "      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "    }\n",
       "  };var element = document.getElementById(\"1890\");\n",
       "  if (element == null) {\n",
       "    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '1890' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    \n",
       "    function(Bokeh) {\n",
       "      \n",
       "    },\n",
       "    function(Bokeh) {\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if ((root.Bokeh !== undefined) || (force === true)) {\n",
       "      for (var i = 0; i < inline_js.length; i++) {\n",
       "        inline_js[i].call(root, root.Bokeh);\n",
       "      }if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"1890\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(js_urls, function() {\n",
       "      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"1890\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n    }\n    finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.info(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(js_urls, callback) {\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = js_urls.length;\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var s = document.createElement('script');\n      s.src = url;\n      s.async = false;\n      s.onreadystatechange = s.onload = function() {\n        root._bokeh_is_loading--;\n        if (root._bokeh_is_loading === 0) {\n          console.log(\"Bokeh: all BokehJS libraries loaded\");\n          run_callbacks()\n        }\n      };\n      s.onerror = function() {\n        console.warn(\"failed to load library \" + url);\n      };\n      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.getElementsByTagName(\"head\")[0].appendChild(s);\n    }\n  };var element = document.getElementById(\"1890\");\n  if (element == null) {\n    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '1890' but no matching script tag was found. \")\n    return false;\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n    }\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"1890\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(js_urls, function() {\n      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"19ca30dd-66d5-42f8-950a-c56dcb591561\" data-root-id=\"1892\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"9ee6d97c-56c4-4336-9fc3-b01f4f1927a1\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"1902\",\"type\":\"LinearAxis\"}],\"left\":[{\"id\":\"1907\",\"type\":\"LinearAxis\"}],\"plot_height\":500,\"plot_width\":800,\"renderers\":[{\"id\":\"1902\",\"type\":\"LinearAxis\"},{\"id\":\"1906\",\"type\":\"Grid\"},{\"id\":\"1907\",\"type\":\"LinearAxis\"},{\"id\":\"1911\",\"type\":\"Grid\"},{\"id\":\"1928\",\"type\":\"Legend\"},{\"id\":\"1921\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"1891\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"1913\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"1894\",\"type\":\"DataRange1d\"},\"x_scale\":{\"id\":\"1898\",\"type\":\"LinearScale\"},\"y_range\":{\"id\":\"1896\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"1900\",\"type\":\"LinearScale\"}},\"id\":\"1892\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{},\"id\":\"1926\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"1924\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"1956\",\"type\":\"Selection\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"1916\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"1915\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"1920\",\"type\":\"Wedge\"},{\"attributes\":{\"formatter\":{\"id\":\"1926\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"1892\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1908\",\"type\":\"BasicTicker\"}},\"id\":\"1907\",\"type\":\"LinearAxis\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"1912\",\"type\":\"HoverTool\"}]},\"id\":\"1913\",\"type\":\"Toolbar\"},{\"attributes\":{\"plot\":{\"id\":\"1892\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1903\",\"type\":\"BasicTicker\"}},\"id\":\"1906\",\"type\":\"Grid\"},{\"attributes\":{\"callback\":null,\"data\":{\"angle\":{\"__ndarray__\":\"W7pBcPRP3j89R1/e0Gj1P0QR546kDn0/45KHuj4TWz+r4havcazYP0/J6f4ngXk/vbtnhcvnpj8c5iT1mLpZPyci1w5xtaU/zP948kzzoT9pHP5JfY2gPxmHf2NoEAdAU8DP4v9PuD9AWp72dvW1P3nUi3pOPOs/\",\"dtype\":\"float64\",\"shape\":[15]},\"color\":[\"#e5f5e0\",\"#a1d99b\",\"#feedde\",\"#fdd0a2\",\"#fdae6b\",\"#fd8d3c\",\"#e6550d\",\"#a63603\",\"#9467bd\",\"#8c564b\",\"#deebf7\",\"#9ecae1\",\"#3182bd\",\"#e377c2\",\"#d62728\"],\"complete_database\":[\" ENA  env \",\" ENA  pro \",\" JGI  JGI_archaea \",\" JGI  JGI_archaea_v2 \",\" JGI  JGI_bacteria \",\" JGI  JGI_bacteria_v2 \",\" JGI  JGI_metagenomes \",\" JGI  JGI_metagenomes_v2 \",\" MGnify  bacteria \",\" Metasub  bacteria \",\" NCBI  archaea \",\" NCBI  bacteria \",\" NCBI  metagenome \",\" Segata-Pasolli  metagenome \",\" ensembl  bacteria \"],\"database\":[\" ENA \",\" ENA \",\" JGI \",\" JGI \",\" JGI \",\" JGI \",\" JGI \",\" JGI \",\" MGnify \",\" Metasub \",\" NCBI \",\" NCBI \",\" NCBI \",\" Segata-Pasolli \",\" ensembl \"],\"index\":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14],\"sub_database\":[\" env \",\" pro \",\" JGI_archaea \",\" JGI_archaea_v2 \",\" JGI_bacteria \",\" JGI_bacteria_v2 \",\" JGI_metagenomes \",\" JGI_metagenomes_v2 \",\" bacteria \",\" bacteria \",\" archaea \",\" bacteria \",\" metagenome \",\" metagenome \",\" bacteria \"],\"value\":[51876,146559,777,181,42226,682,4900,172,4644,3840,3541,315772,10402,9395,93221]},\"selected\":{\"id\":\"1956\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"1957\",\"type\":\"UnionRenderers\"}},\"id\":\"1917\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"1908\",\"type\":\"BasicTicker\"},{\"attributes\":{\"dimension\":1,\"plot\":{\"id\":\"1892\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1908\",\"type\":\"BasicTicker\"}},\"id\":\"1911\",\"type\":\"Grid\"},{\"attributes\":{\"source\":{\"id\":\"1917\",\"type\":\"ColumnDataSource\"}},\"id\":\"1922\",\"type\":\"CDSView\"},{\"attributes\":{\"label\":{\"field\":\"complete_database\"},\"renderers\":[{\"id\":\"1921\",\"type\":\"GlyphRenderer\"}]},\"id\":\"1929\",\"type\":\"LegendItem\"},{\"attributes\":{\"field\":\"angle\",\"include_zero\":true},\"id\":\"1915\",\"type\":\"CumSum\"},{\"attributes\":{\"data_source\":{\"id\":\"1917\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"1919\",\"type\":\"Wedge\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"1920\",\"type\":\"Wedge\"},\"selection_glyph\":null,\"view\":{\"id\":\"1922\",\"type\":\"CDSView\"}},\"id\":\"1921\",\"type\":\"GlyphRenderer\"},{\"attributes\":{},\"id\":\"1957\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"1900\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"1898\",\"type\":\"LinearScale\"},{\"attributes\":{\"callback\":null},\"id\":\"1894\",\"type\":\"DataRange1d\"},{\"attributes\":{\"callback\":null},\"id\":\"1896\",\"type\":\"DataRange1d\"},{\"attributes\":{\"items\":[{\"id\":\"1929\",\"type\":\"LegendItem\"}],\"plot\":{\"id\":\"1892\",\"subtype\":\"Figure\",\"type\":\"Plot\"}},\"id\":\"1928\",\"type\":\"Legend\"},{\"attributes\":{\"field\":\"angle\"},\"id\":\"1916\",\"type\":\"CumSum\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"1916\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_color\":{\"field\":\"color\"},\"line_color\":{\"value\":\"white\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"1915\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"1919\",\"type\":\"Wedge\"},{\"attributes\":{\"callback\":null,\"tooltips\":\"@complete_database: @value\"},\"id\":\"1912\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"1903\",\"type\":\"BasicTicker\"},{\"attributes\":{\"plot\":null,\"text\":\"Genomic Data on Server\"},\"id\":\"1891\",\"type\":\"Title\"},{\"attributes\":{\"formatter\":{\"id\":\"1924\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"1892\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"1903\",\"type\":\"BasicTicker\"}},\"id\":\"1902\",\"type\":\"LinearAxis\"}],\"root_ids\":[\"1892\"]},\"title\":\"Bokeh Application\",\"version\":\"1.0.4\"}};\n",
       "  var render_items = [{\"docid\":\"9ee6d97c-56c4-4336-9fc3-b01f4f1927a1\",\"roots\":{\"1892\":\"19ca30dd-66d5-42f8-950a-c56dcb591561\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "1892"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "output_notebook();\n",
    "\n",
    "p = figure(plot_height=500, plot_width=800,title=\"Genomic Data on Server\", toolbar_location=None,\n",
    "        tools=\"hover\", tooltips=\"@complete_database: @value\")\n",
    "\n",
    "p.wedge(x=0, y=1, radius=0.4,\n",
    "        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),\n",
    "        line_color=\"white\", fill_color='color', legend='complete_database', source=by_sub_plot)\n",
    "\n",
    "show(p);"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### By Data Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub.reset_index(name='value').rename(columns={'index':'sub_database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>51876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>146559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>42226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>4900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>4644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>3840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>3541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>315772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>10402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>9395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>93221</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database   value\n",
       "0               ENA                   env    51876\n",
       "1               ENA                   pro   146559\n",
       "2               JGI           JGI_archaea      777\n",
       "3               JGI        JGI_archaea_v2      181\n",
       "4               JGI          JGI_bacteria    42226\n",
       "5               JGI       JGI_bacteria_v2      682\n",
       "6               JGI       JGI_metagenomes     4900\n",
       "7               JGI    JGI_metagenomes_v2      172\n",
       "8            MGnify              bacteria     4644\n",
       "9           Metasub              bacteria     3840\n",
       "10             NCBI               archaea     3541\n",
       "11             NCBI              bacteria   315772\n",
       "12             NCBI            metagenome    10402\n",
       "13   Segata-Pasolli            metagenome     9395\n",
       "14          ensembl              bacteria    93221"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\ipykernel_launcher.py:16: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['env', 'pro', 'archaea', 'bacteria', 'metagenomes']"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_dbs_dir = {'metagenome': 'metagenomes',\n",
    " 'JGI_archaea':'archaea',\n",
    " 'JGI_bacteria':'bacteria',\n",
    " 'JGI_metagenomes':'metagenomes',\n",
    " 'JGI_metagenomes_v2':'metagenomes',\n",
    " 'JGI_bacteria_v2':\"bacteria\",\n",
    " 'JGI_archaea_v2':\"archaea\"}\n",
    "\n",
    "\n",
    "for index, row in by_sub_plot.iterrows():\n",
    "    #print (color_dict)\n",
    "    \n",
    "    db_name=row['sub_database']\n",
    "    \n",
    "    new_db = sub_dbs_dir.get(db_name.strip(),db_name.strip())\n",
    "    by_sub_plot.set_value(index,\"sub_database\",new_db)\n",
    "\n",
    "    \n",
    "sub_db = by_sub_plot.sub_database.unique()\n",
    "list(sub_db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub_plot.groupby(\"sub_database\")[\"value\"].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub_plot.reset_index(name='value').rename(columns={'index':'sub_database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot['angle'] = by_sub_plot['value']/by_sub_plot['value'].sum() * 2*pi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot['color'] = Blues[len(by_sub_plot)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>archaea</td>\n",
       "      <td>4499</td>\n",
       "      <td>0.041076</td>\n",
       "      <td>#08519c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bacteria</td>\n",
       "      <td>460385</td>\n",
       "      <td>4.203334</td>\n",
       "      <td>#3182bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>env</td>\n",
       "      <td>51876</td>\n",
       "      <td>0.473630</td>\n",
       "      <td>#6baed6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>metagenomes</td>\n",
       "      <td>24869</td>\n",
       "      <td>0.227055</td>\n",
       "      <td>#bdd7e7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pro</td>\n",
       "      <td>146559</td>\n",
       "      <td>1.338090</td>\n",
       "      <td>#eff3ff</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sub_database   value     angle    color\n",
       "0      archaea    4499  0.041076  #08519c\n",
       "1     bacteria  460385  4.203334  #3182bd\n",
       "2          env   51876  0.473630  #6baed6\n",
       "3  metagenomes   24869  0.227055  #bdd7e7\n",
       "4          pro  146559  1.338090  #eff3ff"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"2117\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"2117\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n",
       "    }\n",
       "    finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.info(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(js_urls, callback) {\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = js_urls.length;\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var s = document.createElement('script');\n",
       "      s.src = url;\n",
       "      s.async = false;\n",
       "      s.onreadystatechange = s.onload = function() {\n",
       "        root._bokeh_is_loading--;\n",
       "        if (root._bokeh_is_loading === 0) {\n",
       "          console.log(\"Bokeh: all BokehJS libraries loaded\");\n",
       "          run_callbacks()\n",
       "        }\n",
       "      };\n",
       "      s.onerror = function() {\n",
       "        console.warn(\"failed to load library \" + url);\n",
       "      };\n",
       "      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "    }\n",
       "  };var element = document.getElementById(\"2117\");\n",
       "  if (element == null) {\n",
       "    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '2117' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    \n",
       "    function(Bokeh) {\n",
       "      \n",
       "    },\n",
       "    function(Bokeh) {\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if ((root.Bokeh !== undefined) || (force === true)) {\n",
       "      for (var i = 0; i < inline_js.length; i++) {\n",
       "        inline_js[i].call(root, root.Bokeh);\n",
       "      }if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"2117\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(js_urls, function() {\n",
       "      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"2117\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n    }\n    finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.info(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(js_urls, callback) {\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = js_urls.length;\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var s = document.createElement('script');\n      s.src = url;\n      s.async = false;\n      s.onreadystatechange = s.onload = function() {\n        root._bokeh_is_loading--;\n        if (root._bokeh_is_loading === 0) {\n          console.log(\"Bokeh: all BokehJS libraries loaded\");\n          run_callbacks()\n        }\n      };\n      s.onerror = function() {\n        console.warn(\"failed to load library \" + url);\n      };\n      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.getElementsByTagName(\"head\")[0].appendChild(s);\n    }\n  };var element = document.getElementById(\"2117\");\n  if (element == null) {\n    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '2117' but no matching script tag was found. \")\n    return false;\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n    }\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"2117\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(js_urls, function() {\n      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"1d4d3415-0223-45ea-8b51-2c12787d729f\" data-root-id=\"2119\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"ee99cd5e-282f-415a-9914-e3f8ff13d9c2\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"2129\",\"type\":\"LinearAxis\"}],\"left\":[{\"id\":\"2134\",\"type\":\"LinearAxis\"}],\"plot_height\":500,\"plot_width\":800,\"renderers\":[{\"id\":\"2129\",\"type\":\"LinearAxis\"},{\"id\":\"2133\",\"type\":\"Grid\"},{\"id\":\"2134\",\"type\":\"LinearAxis\"},{\"id\":\"2138\",\"type\":\"Grid\"},{\"id\":\"2155\",\"type\":\"Legend\"},{\"id\":\"2148\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"2118\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"2140\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"2121\",\"type\":\"DataRange1d\"},\"x_scale\":{\"id\":\"2125\",\"type\":\"LinearScale\"},\"y_range\":{\"id\":\"2123\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"2127\",\"type\":\"LinearScale\"}},\"id\":\"2119\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"field\":\"angle\"},\"id\":\"2143\",\"type\":\"CumSum\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"2139\",\"type\":\"HoverTool\"}]},\"id\":\"2140\",\"type\":\"Toolbar\"},{\"attributes\":{\"formatter\":{\"id\":\"2151\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"2119\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"2130\",\"type\":\"BasicTicker\"}},\"id\":\"2129\",\"type\":\"LinearAxis\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"2143\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_color\":{\"field\":\"color\"},\"line_color\":{\"value\":\"white\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"2142\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"2146\",\"type\":\"Wedge\"},{\"attributes\":{\"source\":{\"id\":\"2144\",\"type\":\"ColumnDataSource\"}},\"id\":\"2149\",\"type\":\"CDSView\"},{\"attributes\":{\"callback\":null,\"tooltips\":\"@sub_database: @value\"},\"id\":\"2139\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"2135\",\"type\":\"BasicTicker\"},{\"attributes\":{\"plot\":null,\"text\":\"Genomic Data on Server\"},\"id\":\"2118\",\"type\":\"Title\"},{\"attributes\":{\"formatter\":{\"id\":\"2153\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"2119\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"2135\",\"type\":\"BasicTicker\"}},\"id\":\"2134\",\"type\":\"LinearAxis\"},{\"attributes\":{\"callback\":null},\"id\":\"2123\",\"type\":\"DataRange1d\"},{\"attributes\":{},\"id\":\"2125\",\"type\":\"LinearScale\"},{\"attributes\":{\"dimension\":1,\"plot\":{\"id\":\"2119\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"2135\",\"type\":\"BasicTicker\"}},\"id\":\"2138\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"2151\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"data_source\":{\"id\":\"2144\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"2146\",\"type\":\"Wedge\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"2147\",\"type\":\"Wedge\"},\"selection_glyph\":null,\"view\":{\"id\":\"2149\",\"type\":\"CDSView\"}},\"id\":\"2148\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"callback\":null},\"id\":\"2121\",\"type\":\"DataRange1d\"},{\"attributes\":{},\"id\":\"2127\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"2198\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"callback\":null,\"data\":{\"angle\":{\"__ndarray__\":\"KTuv0esHpT/9BgXiNtAQQFu6QXD0T94/BEb7fyMQzT89R1/e0Gj1Pw==\",\"dtype\":\"float64\",\"shape\":[5]},\"color\":[\"#08519c\",\"#3182bd\",\"#6baed6\",\"#bdd7e7\",\"#eff3ff\"],\"index\":[0,1,2,3,4],\"sub_database\":[\"archaea\",\"bacteria\",\"env\",\"metagenomes\",\"pro\"],\"value\":[4499,460385,51876,24869,146559]},\"selected\":{\"id\":\"2197\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"2198\",\"type\":\"UnionRenderers\"}},\"id\":\"2144\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"label\":{\"field\":\"sub_database\"},\"renderers\":[{\"id\":\"2148\",\"type\":\"GlyphRenderer\"}]},\"id\":\"2156\",\"type\":\"LegendItem\"},{\"attributes\":{\"plot\":{\"id\":\"2119\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"2130\",\"type\":\"BasicTicker\"}},\"id\":\"2133\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"2153\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"field\":\"angle\",\"include_zero\":true},\"id\":\"2142\",\"type\":\"CumSum\"},{\"attributes\":{},\"id\":\"2130\",\"type\":\"BasicTicker\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"2143\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"2142\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"2147\",\"type\":\"Wedge\"},{\"attributes\":{},\"id\":\"2197\",\"type\":\"Selection\"},{\"attributes\":{\"items\":[{\"id\":\"2156\",\"type\":\"LegendItem\"}],\"plot\":{\"id\":\"2119\",\"subtype\":\"Figure\",\"type\":\"Plot\"}},\"id\":\"2155\",\"type\":\"Legend\"}],\"root_ids\":[\"2119\"]},\"title\":\"Bokeh Application\",\"version\":\"1.0.4\"}};\n",
       "  var render_items = [{\"docid\":\"ee99cd5e-282f-415a-9914-e3f8ff13d9c2\",\"roots\":{\"2119\":\"1d4d3415-0223-45ea-8b51-2c12787d729f\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "2119"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "output_notebook();\n",
    "\n",
    "p = figure(plot_height=500, plot_width=800,title=\"Genomic Data on Server\", toolbar_location=None,\n",
    "        tools=\"hover\", tooltips=\"@sub_database: @value\")\n",
    "\n",
    "p.wedge(x=0, y=1, radius=0.4,\n",
    "        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),\n",
    "        line_color=\"white\", fill_color='color', legend='sub_database', source=by_sub_plot)\n",
    "\n",
    "show(p);"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Bar Graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3020: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "data = pd.read_csv('fasta_index_file_oa.txt', sep=\"\\t\", header=None, names = [\"id\", \"filename_id\", \"filename\", \"database\",\"sub_database\",\"filename_2\",\"7\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=data.groupby([\"database\",\"sub_database\"])[\"id\"].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=by_db_type.reset_index(name='value').rename(columns={'index':'database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>51876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>146559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>42226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>4900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>4644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>3840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>3541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>315772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>10402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>9395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>93221</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database   value\n",
       "0               ENA                   env    51876\n",
       "1               ENA                   pro   146559\n",
       "2               JGI           JGI_archaea      777\n",
       "3               JGI        JGI_archaea_v2      181\n",
       "4               JGI          JGI_bacteria    42226\n",
       "5               JGI       JGI_bacteria_v2      682\n",
       "6               JGI       JGI_metagenomes     4900\n",
       "7               JGI    JGI_metagenomes_v2      172\n",
       "8            MGnify              bacteria     4644\n",
       "9           Metasub              bacteria     3840\n",
       "10             NCBI               archaea     3541\n",
       "11             NCBI              bacteria   315772\n",
       "12             NCBI            metagenome    10402\n",
       "13   Segata-Pasolli            metagenome     9395\n",
       "14          ensembl              bacteria    93221"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_db_type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\ipykernel_launcher.py:16: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['env', 'pro', 'archaea', 'bacteria', 'metagenomes']"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_dbs_dir = {'metagenome': 'metagenomes',\n",
    " 'JGI_archaea':'archaea',\n",
    " 'JGI_bacteria':'bacteria',\n",
    " 'JGI_metagenomes':'metagenomes',\n",
    " 'JGI_metagenomes_v2':'metagenomes',\n",
    " 'JGI_bacteria_v2':\"bacteria\",\n",
    " 'JGI_archaea_v2':\"archaea\"}\n",
    "\n",
    "\n",
    "for index, row in by_db_type.iterrows():\n",
    "    #print (color_dict)\n",
    "    \n",
    "    db_name=row['sub_database']\n",
    "    \n",
    "    new_db = sub_dbs_dir.get(db_name.strip(),db_name.strip())\n",
    "    by_db_type.set_value(index,\"sub_database\",new_db)\n",
    "\n",
    "    \n",
    "sub_db = by_db_type.sub_database.unique()\n",
    "list(sub_db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=by_db_type.groupby([\"database\",\"sub_database\"])[\"value\"].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=by_db_type.reset_index(name='value').rename(columns={'index':'database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>51876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>146559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>42908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>metagenomes</td>\n",
       "      <td>5072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>4644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>3840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>3541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>315772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenomes</td>\n",
       "      <td>10402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenomes</td>\n",
       "      <td>9395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>93221</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database sub_database   value\n",
       "0               ENA           env   51876\n",
       "1               ENA           pro  146559\n",
       "2               JGI       archaea     958\n",
       "3               JGI      bacteria   42908\n",
       "4               JGI   metagenomes    5072\n",
       "5            MGnify      bacteria    4644\n",
       "6           Metasub      bacteria    3840\n",
       "7              NCBI       archaea    3541\n",
       "8              NCBI      bacteria  315772\n",
       "9              NCBI   metagenomes   10402\n",
       "10   Segata-Pasolli   metagenomes    9395\n",
       "11          ensembl      bacteria   93221"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_db_type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>sub_database</th>\n",
       "      <th>archaea</th>\n",
       "      <th>bacteria</th>\n",
       "      <th>env</th>\n",
       "      <th>metagenomes</th>\n",
       "      <th>pro</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>database</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENA</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>51876.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>146559.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JGI</th>\n",
       "      <td>958.0</td>\n",
       "      <td>42908.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5072.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGnify</th>\n",
       "      <td>NaN</td>\n",
       "      <td>4644.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metasub</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3840.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI</th>\n",
       "      <td>3541.0</td>\n",
       "      <td>315772.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10402.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Segata-Pasolli</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9395.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl</th>\n",
       "      <td>NaN</td>\n",
       "      <td>93221.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "sub_database      archaea  bacteria      env  metagenomes       pro\n",
       "database                                                           \n",
       " ENA                  NaN       NaN  51876.0          NaN  146559.0\n",
       " JGI                958.0   42908.0      NaN       5072.0       NaN\n",
       " MGnify               NaN    4644.0      NaN          NaN       NaN\n",
       " Metasub              NaN    3840.0      NaN          NaN       NaN\n",
       " NCBI              3541.0  315772.0      NaN      10402.0       NaN\n",
       " Segata-Pasolli       NaN       NaN      NaN       9395.0       NaN\n",
       " ensembl              NaN   93221.0      NaN          NaN       NaN"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pivot_df = by_db_type.pivot(index='database', columns='sub_database', values='value')\n",
    "pivot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmwAAAHwCAYAAAAb2TOAAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzs3Xu8VmWd///XBwTBE1riaMIENqgIbFA5iY6KjEJHtTQ8JaYzmGapv0ZLy8OYOtY4Nul4+GqalicMMy11Sk0zEpWN4hYkAhUVMUVRUhHl8Pn9cS9og5sNsoF73ZvX8/HYj32va50+9w1s3nutdV1XZCaSJEkqrzbVLkCSJEnNM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUskZ2CRJkkrOwCZJklRyBjZJkqSSM7BJkiSV3EbVLmBt23rrrbNbt27VLkOSJGmVJk6c+Hpmdl7Vdq0usHXr1o36+vpqlyFJkrRKEfHC6mznLVFJkqSSM7BJkiSVnIFNkiSp5FrdM2ySJG0oFi5cyKxZs1iwYEG1S9EqdOjQgS5dutCuXbs12t/AJklSjZo1axabb7453bp1IyKqXY5WIjN54403mDVrFt27d1+jY3hLVJKkGrVgwQI+/vGPG9ZKLiL4+Mc/3qIroQY2SZJqmGGtNrT0z8nAJkmSVHIGNkmS9JGde+65XHzxxWu07zHHHMPYsWOb3eb6669n9uzZqzzWvvvuu0EMmG9gkyRJpbO6gW1DYWCTJEkAvPvuu3z2s5+lb9++9O7dmzFjxtCtWzdef/11AOrr69l3332Xbf/UU0+x33770aNHD6655pqVHjczOemkk9hll1347Gc/y2uvvbZs3XnnnceAAQPo3bs3o0ePJjMZO3Ys9fX1HHnkkfTr14/33nuvye2WuvHGGxkyZAi9e/fm8ccfB+Dxxx9nyJAh7LrrrgwZMoRp06YBMGXKFAYOHEi/fv2oq6tj+vTpy46xtP34449n8eLFa+1zXRsMbJIkCYD/+7//4xOf+ARPPfUUkydPZsSIEc1u39DQwN1338348eM577zzVnpF7I477mDatGk8/fTTXHPNNTzyyCPL1p100klMmDCByZMn89577/Gb3/yGQw45hP79+3PTTTcxadIkOnbs2OR2S7377rs88sgjXHHFFRx77LEA7Lzzzjz88MM8+eSTnHfeeZx55pkAXHXVVZx88slMmjSJ+vp6unTpwtSpUxkzZgx/+tOfmDRpEm3btuWmm25q6ce5VhnYJEkSAH369OH+++/n29/+Nn/84x/p1KlTs9sfeOCBdOzYka233pqhQ4cuu7q1oocffpjDDz+ctm3b8olPfIL99ttv2boHH3yQQYMG0adPH37/+98zZcqUJo/R3HaHH344AHvvvTd/+9vfeOutt5g3bx6HHnoovXv35tRTT122/R577MGFF17ID37wA1544QU6duzIAw88wMSJExkwYAD9+vXjgQce4LnnnvtIn9265sC5kiQJgB133JGJEydyzz33cMYZZ3DAAQew0UYbsWTJEoAPjSO24lAVzQ1d0dS6BQsWcOKJJ1JfX0/Xrl0599xzmxyrbFXbNVXHWWedxdChQ7njjjuYOXPmslu5RxxxBIMGDeLuu+9m+PDh/OQnPyEzGTVqFP/5n//Z/AdURV5hkyRJAMyePZtNNtmEo446in//93/niSeeoFu3bkycOBGA22+/fbnt77zzThYsWMAbb7zBQw89xIABA5o87t57782tt97K4sWLeeWVV3jwwQeBvwfArbfemnfeeWe5nqObb745b7/99iq3AxgzZgwA48aNo1OnTnTq1Il58+ax/fbbA5UODEs999xz7LDDDnzzm9/kC1/4Ag0NDQwbNoyxY8cue7Zu7ty5vPDCCx/9A1yHvMImSZIAePrppznttNNo06YN7dq148orr+S9997juOOO48ILL2TQoEHLbT9w4EA++9nP8uKLL3LWWWfxiU98osnjHnzwwfz+97+nT58+7Ljjjuyzzz4AbLnllvzbv/0bffr0oVu3bssFvmOOOYavfe1rdOzYkfHjx690O4CtttqKIUOG8Le//Y3rrrsOgNNPP51Ro0ZxySWXLHcLdsyYMdx44420a9eObbfdlrPPPpuPfexjnH/++RxwwAEsWbKEdu3acfnll/PJT35yrXyua0M07mXRGvTv3z83hPFYJGldWvLXHtUuoVlttp1e7RJKYerUqfTs2bPaZWg1NfXnFRETM7P/qvb1lqgkSVLJeUtUkiStFU8//TRf+cpXlmvbeOONeeyxx6pUUethYJMkSWtFnz59mDRpUrXLaJW8JSpJklRyBjZJkqSSM7BJkiSVnM+wSZLUSuzf5tC1erz7lvxirR5Pa84rbJIkSSVnYJMkSS1y4403MnDgQPr168fxxx/P4sWL2Wyzzfjud79L3759GTx4MK+++irz5s2jW7duy+YmnT9/Pl27dmXhwoVVfgflZ2CTJElrbOrUqYwZM4Y//elPTJo0ibZt23LTTTfx7rvvMnjwYJ566in23ntvrrnmGjp16kTfvn35wx/+AMCvf/1rhg8fTrt27ar8LsrPZ9gkSR9y9At7V7uEZt24bbUr0FIPPPAAEydOXDa/53vvvcc222xD+/bt+dznPgfA7rvvzn333QfAyJEjGTNmDEOHDuXWW2/lxBNPrFrttcQrbJIkaY1lJqNGjWLSpElMmjSJadOmce6559KuXTsiAoC2bduyaNEiAL7whS9w7733MnfuXCZOnLjcxOxaOQObJElaY8OGDWPs2LG89tprAMydO5cXXnhhpdtvttlmDBw4kJNPPpnPfe5ztG3bdn2VWtO8JSpJUitRjWE4dtllF84//3wOOOAAlixZQrt27bj88sub3WfkyJEceuihPPTQQ+unyFbAwCZJklpk5MiRjBw5crm2d955Z9nrQw45hEMOOWS55cxcb/W1Bt4SlSRJKjkDmyRJUskZ2CRJkkrOwCZJklRyBjZJkqSSM7BJkiSVnMN6SJLUSuz1xf9aq8cb98vT1urxtOa8wiZJktapxYsXV7uEmmdgkyRJa2zmzJnsvPPOjBo1irq6Og455BDmz59Pt27dOO+889hrr734xS9+waRJkxg8eDB1dXUcfPDBvPnmm9UuvaYY2CRJUotMmzaN0aNH09DQwBZbbMEVV1wBQIcOHRg3bhyHHXYYRx99ND/4wQ9oaGigT58+/Md//EeVq64tBjZJktQiXbt2Zc899wTgqKOOYty4cQDLpquaN28eb731Fvvssw8Ao0aN4uGHH65OsTXKwCZJklokIppc3nTTTatRTqu0ysAWER0i4vGIeCoipkTEfxTt3SPisYiYHhFjIqJ90b5xsTyjWN+t0bHOKNqnRcTwRu0jirYZEfGdRu1NnkOSJJXHiy++yPjx4wG45ZZb2GuvvZZb36lTJ7baaiv++Mc/AvDzn/982dU2rZ7VGdbjfWC/zHwnItoB4yLiXuD/A36UmbdGxFXAccCVxfc3M/OfIuIw4AfAyIjYBTgM6AV8Arg/InYsznE5sD8wC5gQEXdl5jPFvk2dQ5IkraBaw3D07NmTG264geOPP54ePXpwwgkncNllly23zQ033MDXvvY15s+fzw477MBPf/rTqtRaq1YZ2DIzgXeKxXbFVwL7AUcU7TcA51IJUwcWrwHGAv8blWujBwK3Zub7wPMRMQMYWGw3IzOfA4iIW4EDI2JqM+eQJEkl0aZNG6666qrl2mbOnLnccr9+/Xj00UfXY1Wty2o9wxYRbSNiEvAacB/wLPBWZi4qNpkFbF+83h54CaBYPw/4eOP2FfZZWfvHmzmHJEnSBmO1AltmLs7MfkAXKlfFeja1WfE9VrJubbV/SESMjoj6iKifM2dOU5tIkqR1oFu3bkyePLnaZbR6H6mXaGa+BTwEDAa2jIilt1S7ALOL17OArgDF+k7A3MbtK+yzsvbXmznHinVdnZn9M7N/586dP8pbkiRJKr3V6SXaOSK2LF53BP4FmAo8CBxSbDYKuLN4fVexTLH+98VzcHcBhxW9SLsDPYDHgQlAj6JHaHsqHRPuKvZZ2TkkSZI2GKvTS3Q74IaIaEsl4N2Wmb+JiGeAWyPifOBJ4Npi+2uBnxedCuZSCWBk5pSIuA14BlgEfD0zFwNExEnAb4G2wHWZOaU41rdXcg5JkqQNxur0Em0Adm2i/Tn+3suzcfsC4NCVHOsC4IIm2u8B7lndc0iSJG1IVucKmyRJqgHd/+e/1+rxnj/lW2v1eCu6/vrrqa+v53//93/X6XlaA6emkiRJ69TixYurXULNM7BJkqQWOeigg9h9993p1asXV199NQCbbbYZZ599NoMGDWL8+PFMmDCBIUOG0LdvXwYOHMjbb78NwOzZsxkxYgQ9evTg9NNPX3bME044gf79+9OrVy/OOeecZe0TJ05kn332Yffdd2f48OG88sorAFxzzTUMGDCAvn378qUvfYn58+evx09g3TOwSZKkFrnuuuuYOHEi9fX1XHrppbzxxhu8++679O7dm8cee4yBAwcycuRIfvzjH/PUU09x//3307FjRwAmTZrEmDFjePrppxkzZgwvvVQZS/+CCy6gvr6ehoYG/vCHP9DQ0MDChQv5xje+wdixY5k4cSLHHnss3/3udwH44he/yIQJE3jqqafo2bMn117buvop+gybJElqkUsvvZQ77rgDgJdeeonp06fTtm1bvvSlLwEwbdo0tttuOwYMGADAFltssWzfYcOG0alTJwB22WUXXnjhBbp27cptt93G1VdfzaJFi3jllVd45plnaNOmDZMnT2b//fcHKrdat9tuOwAmT57M9773Pd566y3eeecdhg8fvt7e//pgYJMkSWvsoYce4v7772f8+PFssskm7LvvvixYsIAOHTrQtm1bADKTyrTiH7bxxhsve922bVsWLVrE888/z8UXX8yECRPYaqutOOaYY1iwYAGZSa9evRg/fvyHjnPMMcfwq1/9ir59+3L99dfz0EMPrZP3Wy3eEpUkSWts3rx5bLXVVmyyySb8+c9/bnKC95133pnZs2czYcIEAN5++20WLVr0oe2W+tvf/samm25Kp06dePXVV7n33nsB2GmnnZgzZ86ywLZw4UKmTJmy7JjbbbcdCxcu5Kabblrbb7PqvMImSVIrsa6H4WjKiBEjuOqqq6irq2OnnXZi8ODBH9qmffv2jBkzhm984xu89957dOzYkfvvv3+lx+zbty+77rorvXr1YocddmDPPfdcdpyxY8fyzW9+k3nz5rFo0SJOOeUUevXqxfe//30GDRrEJz/5Sfr06bOsU0NrEZUZoFqP/v37Z319fbXLkKSadtRjx1W7hGbdOKh1PVC+pqZOnUrPnj2rXYZWU1N/XhExMTP7r2pfb4lKkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUskZ2CRJkkrOcdgkSWol1vZwLKszfMrMmTP53Oc+x+TJk9f4PDNnzuSRRx7hiCOO+Mj7DhkyhEceeWSNz10rvMImSZKqaubMmdx8880faZ/FixcDbBBhDQxskiSphRYtWsSoUaOoq6vjkEMOYf78+Zx33nkMGDCA3r17M3r0aJYO1D9jxgz+5V/+hb59+7Lbbrvx7LPP8p3vfIc//vGP9OvXjx/96EcsXryY0047jQEDBlBXV8f/+3//D6jMWzp06FCOOOII+vTpA8Bmm20GwDvvvMOwYcPYbbfd6NOnD3feeWd1Pox1xMAmSZJaZNq0aYwePZqGhga22GILrrjiCk466SQmTJjA5MmTee+99/jNb34DwJFHHsnXv/51nnrqKR555BG22247LrroIv75n/+ZSZMmceqpp3LttdfSqVMnJkyYwIQJE7jmmmt4/vnnAXj88ce54IILeOaZZ5aroUOHDtxxxx088cQTPPjgg3zrW9+iNc3m5DNskiSpRbp27bpsvs+jjjqKSy+9lO7du/PDH/6Q+fPnM3fuXHr16sW+++7Lyy+/zMEHHwxUQlZTfve739HQ0MDYsWOBygTz06dPp3379gwcOJDu3bt/aJ/M5Mwzz+Thhx+mTZs2vPzyy7z66qtsu+226+hdr18GNkmS1CIR8aHlE088kfr6erp27cq5557LggULVvuKV2Zy2WWXMXz48OXaH3roITbddNMm97npppuYM2cOEydOpF27dnTr1o0FCxas2RsqIW+JSpKkFnnxxRcZP348ALfccgt77bUXAFtvvTXvvPPOsitlW2yxBV26dOFXv/oVAO+//z7z589n88035+233152vOHDh3PllVeycOFCAP7yl7/w7rvvNlvDvHnz2GabbWjXrh0PPvggL7zwwlp/n9XkFTZJklqJ1RmGY13o2bMnN9xwA8cffzw9evTghBNO4M0336RPnz5069aNAQMGLNv25z//Occffzxnn3027dq14xe/+AV1dXVstNFG9O3bl2OOOYaTTz6ZmTNnsttuu5GZdO7ceVnIW5kjjzySz3/+8/Tv359+/fqx8847r+u3vV5Fa3ogD6B///5ZX19f7TIkqaat7fG81rZqBZOymTp1Kj179qx2GVpNTf15RcTEzOy/qn29JSpJklRyBjZJkqSSM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjnHYZMkqZVY8tcea/V4bbadvlaPt6JJkyYxe/ZsPvOZz6zT87QGXmGTJElVMWnSJO65555ql1ETDGySJGmNzZw5k5133pl//dd/pXfv3hx55JHcf//97LnnnvTo0YPHH3+cd999l2OPPZYBAwaw6667cuedd/LBBx9w9tlnM2bMGPr168eYMWN4/PHHGTJkCLvuuitDhgxh2rRpAMyfP58vf/nL1NXVMXLkSAYNGsTSQfJ/97vfsccee7Dbbrtx6KGH8s477wDQrVs3zjnnHHbbbTf69OnDn//8ZwDmzp3LQQcdRF1dHYMHD6ahoQGAc889l1GjRnHAAQfQrVs3fvnLX3L66afTp08fRowYsWyarIkTJ7LPPvuw++67M3z4cF555RUALr30UnbZZRfq6uo47LDD1vrnbGCTJEktMmPGDE4++WQaGhr485//zM0338y4ceO4+OKLufDCC7ngggvYb7/9mDBhAg8++CCnnXYaCxcu5LzzzmPkyJFMmjSJkSNHsvPOO/Pwww/z5JNPct5553HmmWcCcMUVV7DVVlvR0NDAWWedxcSJEwF4/fXXOf/887n//vt54okn6N+/P5dccsmyurbeemueeOIJTjjhBC6++GIAzjnnHHbddVcaGhq48MILOfroo5dt/+yzz3L33Xdz5513ctRRRzF06FCefvppOnbsyN13383ChQv5xje+wdixY5k4cSLHHnss3/3udwG46KKLePLJJ2loaOCqq65a65+xz7BJkqQW6d69O3369AGgV69eDBs2jIigT58+zJw5k1mzZnHXXXctC00LFizgxRdf/NBx5s2bx6hRo5g+fToRseyq1rhx4zj55JMB6N27N3V1dQA8+uijPPPMM+y5554AfPDBB+yxxx7LjvfFL34RgN13351f/vKXy451++23A7DffvvxxhtvMG/ePAA+/elP065dO/r06cPixYsZMWIEwLL3MW3aNCZPnsz+++8PwOLFi9luu+0AqKur48gjj+Sggw7ioIMOWiufa2MGNkmS1CIbb7zxstdt2rRZttymTRsWLVpE27Ztuf3229lpp52W2++xxx5bbvmss85i6NCh3HHHHcycOZN9990XgJXNe56Z7L///txyyy3N1tW2bVsWLVq00mNFxHLbt2nThnbt2i1rX/o+MpNevXoxfvz4Dx3j7rvv5uGHH+auu+7i+9//PlOmTGGjjdZezPKWqCRJWqeGDx/OZZddtiwsPfnkkwBsvvnmvP3228u2mzdvHttvvz0A119//bL2vfbai9tuuw2AZ555hqeffhqAwYMH86c//YkZM2YAlWfd/vKXvzRby957781NN90EwEMPPcTWW2/NFltssVrvY6eddmLOnDnLAtvChQuZMmUKS5Ys4aWXXmLo0KH88Ic/5K233lr2LN3a4hU2SZJaiXU9DMeaOuusszjllFOoq6sjM+nWrRu/+c1vGDp0KBdddBH9+vXjjDPO4PTTT2fUqFFccskl7Lfffsv2P/HEExk1ahR1dXXsuuuu1NXV0alTJzp37sz111/P4Ycfzvvvvw/A+eefz4477rjSWs4991y++tWvUldXxyabbMINN9yw2u+jffv2jB07lm9+85vMmzePRYsWccopp7Djjjty1FFHMW/ePDKTU089lS233HLNP7AmxMouM9aq/v3759KeI5KkNXPUY8dVu4Rm3Tjo2mqXUApTp06lZ8+e1S5jnVu8eDELFy6kQ4cOPPvsswwbNoy//OUvtG/fvtqlfSRN/XlFxMTM7L+qfb3CJkmSSm3+/PkMHTqUhQsXkplceeWVNRfWWsrAJkmSSm3zzTdnQ797ZqcDSZJqWGt7tKm1aumfk4FNkqQa1aFDB9544w1DW8llJm+88QYdOnRY42N4S1SSpBrVpUsXZs2axZw5c6pdilahQ4cOdOnSZY33N7BJklSj2rVrR/fu3atdhtYDb4lKkiSV3CoDW0R0jYgHI2JqREyJiJOL9nMj4uWImFR8fabRPmdExIyImBYRwxu1jyjaZkTEdxq1d4+IxyJiekSMiYj2RfvGxfKMYn23tfnmJUmSasHqXGFbBHwrM3sCg4GvR8QuxbofZWa/4usegGLdYUAvYARwRUS0jYi2wOXAp4FdgMMbHecHxbF6AG8CS0dsPA54MzP/CfhRsZ0kSdIGZZWBLTNfycwnitdvA1OB7ZvZ5UDg1sx8PzOfB2YAA4uvGZn5XGZ+ANwKHBiVmVX3A8YW+98AHNToWEvnjBgLDIulM7FKkiRtID7SM2zFLcldgceKppMioiEirouIrYq27YGXGu02q2hbWfvHgbcyc9EK7csdq1g/r9hekiRpg7HagS0iNgNuB07JzL8BVwKfAvoBrwD/vXTTJnbPNWhv7lgr1jY6Iuojot6uzZIkqbVZrcAWEe2ohLWbMvOXAJn5amYuzswlwDVUbnlC5QpZ10a7dwFmN9P+OrBlRGy0QvtyxyrWdwLmrlhfZl6dmf0zs3/nzp1X5y1JkiTVjNXpJRrAtcDUzLykUft2jTY7GJhcvL4LOKzo4dkd6AE8DkwAehQ9QttT6ZhwV1aGZ34QOKTYfxRwZ6NjjSpeHwL8Ph3OWZIkbWBWZ+DcPYGvAE9HxKSi7UwqvTz7UblFORM4HiAzp0TEbcAzVHqYfj0zFwNExEnAb4G2wHWZOaU43reBWyPifOBJKgGR4vvPI2IGlStrh7XgvUqSJNWkVQa2zBxH08+S3dPMPhcAFzTRfk9T+2Xmc/z9lmrj9gXAoauqUZIkqTVzpgNJkqSSM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUskZ2CRJkkrOwCZJklRyBjZJkqSSM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUskZ2CRJkkrOwCZJklRyBjZJkqSSM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUskZ2CRJkkrOwCZJklRyBjZJkqSSM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUskZ2CRJkkrOwCZJklRyBjZJkqSSM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUskZ2CRJkkrOwCZJklRyBjZJkqSSM7BJkiSVnIFNkiSp5AxskiRJJWdgkyRJKjkDmyRJUsmtMrBFRNeIeDAipkbElIg4uWj/WETcFxHTi+9bFe0REZdGxIyIaIiI3Roda1Sx/fSIGNWoffeIeLrY59KIiObOIUmStCFZnStsi4BvZWZPYDDw9YjYBfgO8EBm9gAeKJYBPg30KL5GA1dCJXwB5wCDgIHAOY0C2JXFtkv3G1G0r+wckiRJG4xVBrbMfCUznyhevw1MBbYHDgRuKDa7ATioeH0g8LOseBTYMiK2A4YD92Xm3Mx8E7gPGFGs2yIzx2dmAj9b4VhNnUOSJGmD8ZGeYYuIbsCuwGPAP2TmK1AJdcA2xWbbAy812m1W0dZc+6wm2mnmHJIkSRuM1Q5sEbEZcDtwSmb+rblNm2jLNWhfbRExOiLqI6J+zpw5H2VXSZKk0lutwBYR7aiEtZsy85dF86vF7UyK768V7bOAro127wLMXkV7lybamzvHcjLz6szsn5n9O3fuvDpvSZIkqWZstKoNih6b1wJTM/OSRqvuAkYBFxXf72zUflJE3Eqlg8G8zHwlIn4LXNioo8EBwBmZOTci3o6IwVRutR4NXLaKc6w3e33xv9b3KT+Scb88rdolSJKkdWyVgQ3YE/gK8HRETCrazqQSom6LiOOAF4FDi3X3AJ8BZgDzga8CFMHs+8CEYrvzMnNu8foE4HqgI3Bv8UUz55AkSdpgrDKwZeY4mn7ODGBYE9sn8PWVHOs64Lom2uuB3k20v9HUOSRJkjYkznQgSZJUcgY2SZKkkjOwSZIklZyBTZIkqeQMbJIkSSVnYJMkSSo5A5skSVLJGdgkSZJKzsAmSZJUcgY2SZKkkjOwSZIklZyBTZIkqeQMbJIkSSVnYJMkSSo5A5skSVLJGdgkSZJKzsAmSZJUcgY2SZKkkjOwSZIklZyBTZIkqeQMbJIkSSVnYJMkSSo5A5skSVLJGdgkSZJKzsAmSZJUcgY2SZKkkjOwSZIklZyBTZIkqeQMbJIkSSVnYJMkSSo5A5skSVLJGdgkSZJKzsAmSZJUcgY2SZKkkjOwSZIklZyBTZIkqeQMbJIkSSVnYJMkSSo5A5skSVLJGdgkSZJKzsAmSZJUcgY2SZKkkjOwSZIklZyBTZIkqeQMbJIkSSVnYJMkSSq5VQa2iLguIl6LiMmN2s6NiJcjYlLx9ZlG686IiBkRMS0ihjdqH1G0zYiI7zRq7x4Rj0XE9IgYExHti/aNi+UZxfpua+tNS5Ik1ZLVucJ2PTCiifYfZWa/4usegIjYBTgM6FXsc0VEtI2ItsDlwKdEGw8BAAAcuUlEQVSBXYDDi20BflAcqwfwJnBc0X4c8GZm/hPwo2I7SZKkDc4qA1tmPgzMXc3jHQjcmpnvZ+bzwAxgYPE1IzOfy8wPgFuBAyMigP2AscX+NwAHNTrWDcXrscCwYntJkqQNSkueYTspIhqKW6ZbFW3bAy812mZW0bay9o8Db2XmohXalztWsX5esb0kSdIGZU0D25XAp4B+wCvAfxftTV0ByzVob+5YHxIRoyOiPiLq58yZ01zdkiRJNWeNAltmvpqZizNzCXANlVueULlC1rXRpl2A2c20vw5sGREbrdC+3LGK9Z1Yya3ZzLw6M/tnZv/OnTuvyVuSJEkqrTUKbBGxXaPFg4GlPUjvAg4renh2B3oAjwMTgB5Fj9D2VDom3JWZCTwIHFLsPwq4s9GxRhWvDwF+X2wvSZK0QdloVRtExC3AvsDWETELOAfYNyL6UblFORM4HiAzp0TEbcAzwCLg65m5uDjOScBvgbbAdZk5pTjFt4FbI+J84Eng2qL9WuDnETGDypW1w1r8biVJkmrQKgNbZh7eRPO1TbQt3f4C4IIm2u8B7mmi/Tn+fku1cfsC4NBV1SdJktTaOdOBJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkVhnYIuK6iHgtIiY3avtYRNwXEdOL71sV7RERl0bEjIhoiIjdGu0zqth+ekSMatS+e0Q8XexzaUREc+eQJEna0KzOFbbrgRErtH0HeCAzewAPFMsAnwZ6FF+jgSuhEr6Ac4BBwEDgnEYB7Mpi26X7jVjFOSRJkjYoqwxsmfkwMHeF5gOBG4rXNwAHNWr/WVY8CmwZEdsBw4H7MnNuZr4J3AeMKNZtkZnjMzOBn61wrKbOIUmStEFZ02fY/iEzXwEovm9TtG8PvNRou1lFW3Pts5pob+4ckiRJG5S13ekgmmjLNWj/aCeNGB0R9RFRP2fOnI+6uyRJUqmtaWB7tbidSfH9taJ9FtC10XZdgNmraO/SRHtz5/iQzLw6M/tnZv/OnTuv4VuSJEkqpzUNbHcBS3t6jgLubNR+dNFbdDAwr7id+VvggIjYquhscADw22Ld2xExuOgdevQKx2rqHJIkSRuUjVa1QUTcAuwLbB0Rs6j09rwIuC0ijgNeBA4tNr8H+AwwA5gPfBUgM+dGxPeBCcV252Xm0o4MJ1DpidoRuLf4oplzSJIkbVBWGdgy8/CVrBrWxLYJfH0lx7kOuK6J9nqgdxPtbzR1DkmSpA2NMx1IkiSVnIFNkiSp5AxskiRJJbfKZ9gkSZLWl6MeO67aJTTrxkHXVuW8XmGTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSybUosEXEzIh4OiImRUR90faxiLgvIqYX37cq2iMiLo2IGRHREBG7NTrOqGL76RExqlH77sXxZxT7RkvqlSRJqkVr4wrb0Mzsl5n9i+XvAA9kZg/ggWIZ4NNAj+JrNHAlVAIecA4wCBgInLM05BXbjG6034i1UK8kSVJN2WgdHPNAYN/i9Q3AQ8C3i/afZWYCj0bElhGxXbHtfZk5FyAi7gNGRMRDwBaZOb5o/xlwEHDvOqh5pTr+6vH1eTpJkqQPaekVtgR+FxETI2J00fYPmfkKQPF9m6J9e+ClRvvOKtqaa5/VRLskSdIGpaVX2PbMzNkRsQ1wX0T8uZltm3r+LNeg/cMHroTF0QD/+I//2HzFkiRJNaZFgS0zZxffX4uIO6g8g/ZqRGyXma8UtzxfKzafBXRttHsXYHbRvu8K7Q8V7V2a2L6pOq4Grgbo379/k6FO1bHkrz2qXcJKtdl2erVLkCRptazxLdGI2DQiNl/6GjgAmAzcBSzt6TkKuLN4fRdwdNFbdDAwr7hl+lvggIjYquhscADw22Ld2xExuOgdenSjY0mSJG0wWnKF7R+AO4qRNjYCbs7M/4uICcBtEXEc8CJwaLH9PcBngBnAfOCrAJk5NyK+D0wotjtvaQcE4ATgeqAjlc4G67XDgSRJUhmscWDLzOeAvk20vwEMa6I9ga+v5FjXAdc10V4P9F7TGiVJkloDZzqQJEkqOQObJElSya2LgXOlZY5+Ye9ql7BSN25b7QokSVo9XmGTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJIzsEmSJJWcgU2SJKnkDGySJEklZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGNkmSpJLbqNoFSJLUmiz5a49ql9CsNttOr3YJWgNeYZMkSSo5A5skSVLJGdgkSZJKzsAmSZJUcnY6kErqqMeOq3YJzbpx0LXVLkGSNhilD2wRMQL4MdAW+ElmXlTlkiTVAANvy/xp/C7VLqF5g6pdgLR+lfqWaES0BS4HPg3sAhweESX/KSJJkrR2lTqwAQOBGZn5XGZ+ANwKHFjlmiRJktarst8S3R54qdHyLLwQXlNKfVvFv0mSpBoRmVntGlYqIg4FhmfmvxbLXwEGZuY3VthuNDC6WNwJmLZeC/1otgZer3YRNczPb8352bWMn1/L+PmtOT+7lin75/fJzOy8qo3KfoVtFtC10XIXYPaKG2Xm1cDV66uoloiI+szsX+06apWf35rzs2sZP7+W8fNbc352LdNaPr+yP8M2AegREd0joj1wGHBXlWuSJElar0p9hS0zF0XEScBvqQzrcV1mTqlyWZIkSetVqQMbQGbeA9xT7TrWopq4dVtifn5rzs+uZfz8WsbPb8352bVMq/j8St3pQJIkSeV/hk2SJGmDZ2CTJEkqOQObJElSyRnYJEmSSs7AJkmSVHIGtvUsIrpGxGnVrkPSqkVE+4ioi4g+xeDdklQVpR+HrTWIiK2BQ4HDqUxof0d1Kyq/iPg1sNIxZzLzC+uxnJoUEfXAT4GbM/PNatdTayLis8BVwLNAAN0j4vjMvLe6lZVbRHyxufWZ+cv1VUutiYj/ycxTVvbzz597zYuIp2n6/40AMjPr1nNJa5WBbR2JiM2Bg4EjgB2phLQdMrNLVQurHRdXu4BW4DDgq8CERuHtd+ngi6vrv4GhmTkDICI+BdwNGNia9/lm1iVgYFu5nxff/fm3Zj5X7QLWJQfOXUci4j3gceB7wLjMzIh4LjN3qHJp2sBERBsqP8iuBJYA1wE/zsy5VS2s5CLi4czcu9FyAH9o3CapnCJiW2AglV8SJmTmX6tcUot5hW3dOZPKFY4rgZsjYkyV66kpEXEg0CUzLy+WHwM6F6tPz8yxVSuuhkREHZWrbJ8BbgduAvYCfg/0q2JppdXolt6UiLgHuI3KD/1DgQlVK6yGRMQ+wJuZ2RARXwb2pnJr+YrMfL+61ZVXM7f0AKj1W3rrS0T8K3A2lZ9zAVwWEedl5nXVraxlvMK2jkXEDlSeXTsM6AGcA9yRmX+pamElFxF/Ag7LzJeK5UnAMGBT4KeZOaya9dWCiJgIvAVcC9ze+D/KiPhlZjb7rNGGKiJ+2szqzMxj11sxNSgiLgfqgA7ANGAz4P+AIUDbzDyyiuWVWkR8srn1mfnC+qqllkXENGBIZr5RLH8ceCQzd6puZS1jYFuPIqIPlWfavpyZn6p2PWUWERMyc0Cj5f/NzJOK149m5uDqVVduEXFyZv44IvbOzIerXY82LBHxTGbuEhEdgJeBbTJzcXFLuSEz+1S5RLVyEfEA8OnM/KBYbg/ck5n/Ut3KWsbAplKKiBmZ+U8rWfesgXflImJSZvaLiCcyc7dq11OriittTfXU8wpbMxr/vVvx76B/J5sXEW+z/N+5KJaX9nLcoiqF1YiI+P+Kl/2APsCdVD6/A4HHM/Nr1aptbfAZtnUkIp6n6X94UPmHZ+Bo3mMR8W+ZeU3jxog4nkpnDq3c1IiYCXSOiIZG7a2ia/t69JtGrztQ6fU9u0q11JJtiv84o9FriuXOK99Nmbl5tWuocUs/v2eLr6XurEIta51X2NaR4p55Y22ALwP/DjyRmV9a/1XVjojYBvgV8D7wRNG8O7AxcFBmvlqt2mpB0UPqt8CHxm3yOZg1U/S2vT8z96t2LWUWEec0tz4z/2N91VLLIqIv8M/F4sOZ2dDc9mr9DGzrWPFD/ivAacAk4MLMfKa6VdWOiNgP6FUsTsnM31ezHm24ImIn4O6V3aqX1paIOBn4N/4+Zt3BwNWZeVn1qqodEdEf+C7wSRrdSaz1uwsGtnUkItoBxwKnAuOA/8zMZ5vfS0tFxMeaWf1+Zr673oqpMRFxW2Z+uYkhArwl+hE08TzRX4EzMvP2KpVUEyLih8BzmXnVCu2nAttm5rerU1ntKB5l2GPpz7mI2BQY77/d1VP0Ej0NeJrK2JNA7d9dMLCtIxExC1gE/A/w4orrnZ6leY2eAWz87F8U35f+xvSdzLxpfddWdhGxXWa+srIhAmr9h5bKLSKeAXpn5pIV2ttQ6SXauzqV1Y7il60BmbmgWO5AZfBXe9iuhogYl5l7VbuOtc1OB+vO/VSCRt/iqzGnZ1mFzOze3PqI6Az8gcpAsGokM18pvhvMWiAi9gQmZea7EXEUsBuVGSL8XJuXK4a1onFJMbSHVu2nVDpe3UHlF9UDqYynqNVzTkT8BHiAynPQQO1fKPEKm2pWRHw+M39d7TrKqhix/wfANlR+6Ds0wEdQ3JbqS2UQ2J9T+Q/zi5m5T1ULK7mImAAckZnTV2jvAdySmf2rU1ltiYjdqMxKElQ6HTxZ5ZJqRkTcCOwMTOHvt0RrftBrA9s6EhH/k5mnFK9PzswfN1p3fWYeU7XitEGIiBnA5zNzarVrqUVLxwyLiLOBlzPzWscRW7WI+DRwGXA+MLFo7g+cAZySmfdUq7ZaERGfAmZl5vsRsS+VXxp+lplvVbey2hART7fG28dtql1AK9Z4guhRK6zzwVGtD68a1lrk7Yg4AzgKuDsi2gLtqlxT6WXmvcBBwFDg+uJrKPAlw9pqux1YHBH/BPwE6A7cXN2SasqjEbFLtYtY23yGbd2JlbyW1pf6iBjD38ezA2r/OY71aCSVqeSOy8y/RsQ/Av9V5ZpqQmZO5sO/qGr1LcnMRcVjDT/OzMsiwluiq28vYFTRee19WkkPeQPbutMmIraichVz6eulwa1t9crSBmQLYD5wQKM2O7yspsz8K3BJo+UXgZ9Vr6LasLIpvQqZmcetz3pq1MKIOBw4Gvh80ebV3dU3otoFrAs+w7aOFFMDLaHpq2uZmTus34okfRQRMZjKs1g9gfZUftF6JzM7VbWwkouIpmZx+UfgFKBtZnZZzyXVnOJ23teojL12S0R0B0Zm5kVVLq1mRMReQI/M/GkxqsBmmfl8tetqCQOb1MpERC/gU5l5V7H8I2BpyPjfzHxipTtrmYioBw4DfkHlofmjqfwHcGZVC6shEbEDcCaVZ3p/BFybmR9Utyq1dsX0aP2BnTJzx4j4BPCLzNyzyqW1iJ0OpNbnIuD1RsvDgbuBB4Gzq1JRjcrMGVSuCi3OzJ8C+1a5pJoQET2LoRV+TWWml10y80rD2uqJiB4RMTYinomI55Z+VbuuGnIwlXmU3wXIzNn8fWL4muUzbFLrs11mPtJo+W9Lp1OKiOOrVFMtmh8R7YFJxXRLrwCbVrmm0ouIpVckL6YyNd9iYIulY+Zm5tzqVVczfgqcQ+Wq5FDgq9h57aP4IDMzIhKWTe1V87wlKrUyETEtM3daybq/ZOaO67umWlRM7fUqlefXTqVyW/ly5wRuXvH87tL/WFacVs7nd1dDREzMzN0bjycWEX/MzH+udm21ICL+HegB7A/8J5V5vW/OzMuqWlgLeYVNan1mR8SgzHyscWPxEP3sKtVUiw4qBrxeAPwHVAbBBn7c7F4buMzsVu0aWoEFxdyr0yPiJOBlKjOWaDVk5sURsT/wN2An4OzMvK/KZbWYV9ikViYiBgJjqAxYurSDwe5UxsUamZmPV6m0mtLUrAYR8WRm7lqtmrRhiIgBwFRgS+D7VIbo+a/MfLSqhdWI4hbogsxcHBE7UQlt92bmwiqX1iIGNqkVioh/AL4O9CqaplC5nfdq9aqqDcX4V0dQGXzzj41WbQEsysx/qUph2iAUQ1B8EpjhVFRrJiImAv8MbAU8CtQD8zPzyKoW1kIGNklqpHh2rTuVZ1++02jV20BDZi6qSmFq9SLiX4ELgWep/B0cvXR4Hq2+RvMAfwPomJk/bA1Xx32GTWplIqJhZatoBdOzrGuZ+QLwArBHEd56ZOb9EdER6EgluEnrwilAr8ycU4xhdxNgYPvoIiL2AI4Els6sUfN5p+bfgKQPWUKld97NVMbBeq+65dSmiPg3YDTwMeBTQBfgKmBYNetSq/ZBZs4ByMznImLjahdUo04GzgDuyMwpRfh9sMo1tZi3RKVWKCJ2Bg6nMg/hM1TC2++8nbf6ImISMBB4bOmtlMbDLEhrW0S8BtzaqOmwxsuZ+c31XpRKwytsUiuUmX+mMvDmORExksqk5T8A/quqhdWW9zPzg6UDvkbERqx8UnNpbThtheWJVamixkXEjsC/A91olHMyc79q1bQ2GNikVigitqfy2/nBwJtUBn69o6pF1Z4/RMSZQMdiTKcTqdxiltaJzLxhxbaI2DYz/1qNemrYL6g8vvATKjNttAreEpVamYj4A5V5824DxgLLTQXk1ECrpxi49DjgACodNn4L/CT9oan1qKnxANW8pTNFVLuOtc3AJrUyK5kaCP7eS9SpgaQa0RqGo1jfIuJc4DUqdxXeX9pe67+sGtgkqZFmhkUBwGFRtD5FxImZeUW166glEfF8E801/8uqgU2SGil6h650WJRinDZpnYuIbYAOS5cz88UqlqMqa1PtAiSpTDKzH5UhUTajEtouoDLF18uGNa0PEfGFiJgOPA/8AZgJ3FvVompIRGwSEd+LiKuL5R4R8blq19VSBjZJWkFm/jkzzyke9v41lWFRTq1yWdpwfB8YDPwlM7tTGaz5T9Utqab8FPgAGFIszwLOr145a4eBTZJWEBHbR8S3ImIccBSVsHZllcvShmNhZr4BtImINpn5INCv2kXVkE9l5g+BhQCZ+R6VTlc1zXHYJKmRFYZFOYa/D4vSPiI+Vus9zVQT3oqIzYCHgZuKGRCcpWT1fVDM/ZsAEfEpGvUWrVV2OpCkRhwWRdUWEZtS6ezShsoE5p2AG/1lYfUUA11/D9gF+B2wJ3BMZj5UzbpaysAmSVKJRMQPMvPbq2rTykXEx6k8BxjAo5n5epVLajEDmyRJJdLU7AYR0eAYgBs2n2GTJKkEIuIEKnPW7rDCAM6bYy/RDZ5X2CRJKoGI6ARsBfwn8J1Gq972+TUZ2CRJKiFnOlBjjsMmSVKJRMTnnelAKzKwSZJULufjTAdagYFNkqRycaYDfYi9RCVJKhdnOtCH2OlAkqQSKWY6WEBl0NelMx3cVFx10wbKwCZJklRy3hKVJKlEIuJtlp/HFmAeUA98KzOfW/9VqdoMbJIklcslwGzgZiq3RQ8DtgWmAdcB+1atMlWNt0QlSSqRiHgsMwet0PZoZg6OiKcys2+1alP1OKyHJEnlsiQivhwRbYqvLzda51WWDZRX2CRJKpGI2AH4MbAHlYD2KHAq8DKwe2aOq2J5qhIDmyRJUsl5S1SSpBKJiB0j4oGImFws10XE96pdl6rLwCZJUrlcA5wBLATIzAYqPUW1ATOwSZJULptk5uMrtDk11QbOwCZJUrm8HhGfougRGhGHAK9UtyRVm50OJEkqkaKX6NXAEOBN4HngyMx8oaqFqaoMbJIklVAxCXybzHy72rWo+rwlKklSCUTE5yPik42avgWMi4i7IqJ7tepSORjYJEkqhwuAOQAR8TngKOBY4C7gqirWpRIwsEmSVA6ZmfOL118Ers3MiZn5E6BzFetSCRjYJEkqh4iIzSKiDTAM/v/27iZEqyqO4/j3l9nYoBQElbQoGrOIoFcQDMtFBBXFBLqISNoErYKoaFFBRK3cCIYIUg20sDcrok1URCNhYQwzFIQKRW3EIoLKECL/LZ4DDaGPE0w+1zvfDzybc/73cO5d/Z5zXw4fz+tbMaI5qSPOHvUEJEkSANuAWeBX4Juq+hIgyfX4WY8lz7dEJUnqiCSXABcCc1V1vLWtBpZX1Q8jnZxGysAmSZLUcT7DJkmS1HEGNkmSpI4zsEmSJHWcgU2SJKnjDGySJEkdZ2CT1FtJnk3y+JD+ySRXL2CcqSSbFnd2krRwBjZJS9kkcMrAJkmjZmCT1CtJnkpyIMlHwJWt7aEk+5PMJdmTZDzJeuAeYGuS2SQTJ6qbN/RtSfYmOdg25ibJZa1tpv3Wt/bVSabbuF8n2dDab0+yr9W+mWTl6b06ks5UfjhXUm8kuRGYAtYx2HpvBtgJvFJVP7ea54EjVbU9yRTwflW91fouGFJ3MXAnMAF8Aqxh8Kf3eFUdS3IFsLuqbkryGLCiql5IsgwYB8aAt4E7qupokieBsap67v+/MpLOdO4lKqlPNgDvVNUfAEnea+3XtAB2PrAS+OAkxw+re6NtFXQoybfAVcB3wItJrgP+Ata22v3Ay0mWA+9W1WySWxncfv0sCcA5wL7FOGlJ/Wdgk9Q3J7ptMAVMVtVckgeBjSc5dljdv8ct4FHgCHAtg9W2YwBVNZ3kFuAu4NUkW4FfgA+r6r7/fEaSljyfYZPUJ9PAvUnOTbIKuLu1rwIOtxWv++fV/9b6OEUdwOYkZyWZAC4HDgDnAYfbytsDwDKAJJcCP1bVLuAl4Abgc+DmJGtazXiStUjSArjCJqk3qmomyevALPA9sLd1PQN80dq+4p+Q9hqwK8kjwKYhdTAIaJ8CFwEPt+fWdgB7kmxm8Fzb0Va7EXgiyZ/A78CWqvqprdrtTjLW6p4GDi7eFZDUV750IEmS1HHeEpUkSeo4A5skSVLHGdgkSZI6zsAmSZLUcQY2SZKkjjOwSZIkdZyBTZIkqeMMbJIkSR33N/ZJAnZSMxsiAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 720x504 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "pivot_df.loc[:,sub_dbs].plot.bar(stacked=True, color=viridis(5), figsize=(10,7));"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>sub_database</th>\n",
       "      <th>archaea</th>\n",
       "      <th>bacteria</th>\n",
       "      <th>env</th>\n",
       "      <th>metagenomes</th>\n",
       "      <th>pro</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>database</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENA</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>51876.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>146559.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JGI</th>\n",
       "      <td>958.0</td>\n",
       "      <td>42908.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5072.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGnify</th>\n",
       "      <td>NaN</td>\n",
       "      <td>4644.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metasub</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3840.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI</th>\n",
       "      <td>3541.0</td>\n",
       "      <td>315772.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10402.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Segata-Pasolli</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9395.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl</th>\n",
       "      <td>NaN</td>\n",
       "      <td>93221.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "sub_database      archaea  bacteria      env  metagenomes       pro\n",
       "database                                                           \n",
       " ENA                  NaN       NaN  51876.0          NaN  146559.0\n",
       " JGI                958.0   42908.0      NaN       5072.0       NaN\n",
       " MGnify               NaN    4644.0      NaN          NaN       NaN\n",
       " Metasub              NaN    3840.0      NaN          NaN       NaN\n",
       " NCBI              3541.0  315772.0      NaN      10402.0       NaN\n",
       " Segata-Pasolli       NaN       NaN      NaN       9395.0       NaN\n",
       " ensembl              NaN   93221.0      NaN          NaN       NaN"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pivot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "databases = list(by_db_type.database.unique())\n",
    "\n",
    "sub_dbs = list(by_db_type.sub_database.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i,v in enumerate(databases):\n",
    "    databases[i]=v.strip()\n",
    "\n",
    "for i,v in enumerate(sub_dbs):\n",
    "    sub_dbs[i]=v.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_dir={\n",
    "    \"databases\": databases,\n",
    "}\n",
    "for i in sub_db:\n",
    "    data_dir[i] = pivot_df[i].fillna(0).tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'databases': ['ENA',\n",
       "  'JGI',\n",
       "  'MGnify',\n",
       "  'Metasub',\n",
       "  'NCBI',\n",
       "  'Segata-Pasolli',\n",
       "  'ensembl'],\n",
       " 'env': [51876.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],\n",
       " 'pro': [146559.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],\n",
       " 'archaea': [0.0, 958.0, 0.0, 0.0, 3541.0, 0.0, 0.0],\n",
       " 'bacteria': [0.0, 42908.0, 4644.0, 3840.0, 315772.0, 0.0, 93221.0],\n",
       " 'metagenomes': [0.0, 5072.0, 0.0, 0.0, 10402.0, 9395.0, 0.0]}"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_accu\n",
    "\n",
    "{'databases': ['NCBI',\n",
    "  'ENA',\n",
    "  'JGI',\n",
    "  'ensembl',\n",
    "  'MGnify',\n",
    "  'Metasub',\n",
    "  'Segata-Pasolli'],\n",
    " 'metagenomes': [10402, 0, 5072, 0, 0, 0, 9395],\n",
    " 'archaea': [3541, 0, 958, 0, 0, 0, 0],\n",
    " 'bacteria': [315772, 0, 42908, 93221, 4644, 3840, 0],\n",
    " 'pro': [0, 146559, 0, 0, 0, 0, 0],\n",
    " 'env': [0, 51876, 0, 0, 0, 0, 0]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"5f1e9261-8d44-49dc-80ec-45fb7b4fbcff\" data-root-id=\"3763\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"732b9776-99f0-4e1a-8de2-ea752639ca72\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"3773\",\"type\":\"CategoricalAxis\"}],\"left\":[{\"id\":\"3777\",\"type\":\"LinearAxis\"}],\"outline_line_color\":{\"value\":null},\"plot_height\":700,\"renderers\":[{\"id\":\"3773\",\"type\":\"CategoricalAxis\"},{\"id\":\"3776\",\"type\":\"Grid\"},{\"id\":\"3777\",\"type\":\"LinearAxis\"},{\"id\":\"3781\",\"type\":\"Grid\"},{\"id\":\"3799\",\"type\":\"GlyphRenderer\"},{\"id\":\"3805\",\"type\":\"GlyphRenderer\"},{\"id\":\"3811\",\"type\":\"GlyphRenderer\"},{\"id\":\"3817\",\"type\":\"GlyphRenderer\"},{\"id\":\"3823\",\"type\":\"GlyphRenderer\"},{\"id\":\"3825\",\"type\":\"Legend\"}],\"right\":[{\"id\":\"3825\",\"type\":\"Legend\"}],\"title\":{\"id\":\"3762\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"3783\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"3765\",\"type\":\"FactorRange\"},\"x_scale\":{\"id\":\"3769\",\"type\":\"CategoricalScale\"},\"y_range\":{\"id\":\"3767\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"3771\",\"type\":\"LinearScale\"}},\"id\":\"3763\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"callback\":null,\"data\":{\"archaea\":[0.0,958.0,0.0,0.0,3541.0,0.0,0.0],\"bacteria\":[0.0,42908.0,4644.0,3840.0,315772.0,0.0,93221.0],\"databases\":[\"ENA\",\"JGI\",\"MGnify\",\"Metasub\",\"NCBI\",\"Segata-Pasolli\",\"ensembl\"],\"env\":[51876.0,0.0,0.0,0.0,0.0,0.0,0.0],\"metagenomes\":[0.0,5072.0,0.0,0.0,10402.0,9395.0,0.0],\"pro\":[146559.0,0.0,0.0,0.0,0.0,0.0,0.0]},\"selected\":{\"id\":\"3974\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"3975\",\"type\":\"UnionRenderers\"}},\"id\":\"3813\",\"type\":\"ColumnDataSource\"},{\"attributes\":{},\"id\":\"3969\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"callback\":null,\"factors\":[\"ENA\",\"JGI\",\"MGnify\",\"Metasub\",\"NCBI\",\"Segata-Pasolli\",\"ensembl\"],\"range_padding\":0.1},\"id\":\"3765\",\"type\":\"FactorRange\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3791\",\"type\":\"Stack\"}},\"fill_color\":{\"value\":\"#5BC862\"},\"line_color\":{\"value\":\"#5BC862\"},\"top\":{\"expr\":{\"id\":\"3792\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3815\",\"type\":\"VBar\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3791\",\"type\":\"Stack\"}},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"top\":{\"expr\":{\"id\":\"3792\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3816\",\"type\":\"VBar\"},{\"attributes\":{\"data_source\":{\"id\":\"3795\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"3797\",\"type\":\"VBar\"},\"hover_glyph\":null,\"muted_glyph\":null,\"name\":\"env\",\"nonselection_glyph\":{\"id\":\"3798\",\"type\":\"VBar\"},\"selection_glyph\":null,\"view\":{\"id\":\"3800\",\"type\":\"CDSView\"}},\"id\":\"3799\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"callback\":null,\"start\":0},\"id\":\"3767\",\"type\":\"DataRange1d\"},{\"attributes\":{\"formatter\":{\"id\":\"3966\",\"type\":\"BasicTickFormatter\"},\"minor_tick_line_color\":{\"value\":null},\"plot\":{\"id\":\"3763\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"3778\",\"type\":\"BasicTicker\"}},\"id\":\"3777\",\"type\":\"LinearAxis\"},{\"attributes\":{},\"id\":\"3964\",\"type\":\"CategoricalTickFormatter\"},{\"attributes\":{\"dimension\":1,\"plot\":{\"id\":\"3763\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"3778\",\"type\":\"BasicTicker\"}},\"id\":\"3781\",\"type\":\"Grid\"},{\"attributes\":{\"fields\":[\"env\",\"pro\",\"archaea\",\"bacteria\"]},\"id\":\"3793\",\"type\":\"Stack\"},{\"attributes\":{\"source\":{\"id\":\"3813\",\"type\":\"ColumnDataSource\"}},\"id\":\"3818\",\"type\":\"CDSView\"},{\"attributes\":{\"source\":{\"id\":\"3807\",\"type\":\"ColumnDataSource\"}},\"id\":\"3812\",\"type\":\"CDSView\"},{\"attributes\":{\"callback\":null,\"tooltips\":\"@databases $name: @$name\"},\"id\":\"3782\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"3778\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"3968\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"3977\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3785\",\"type\":\"Stack\"}},\"fill_color\":{\"value\":\"#440154\"},\"line_color\":{\"value\":\"#440154\"},\"top\":{\"expr\":{\"id\":\"3786\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3797\",\"type\":\"VBar\"},{\"attributes\":{\"label\":{\"value\":\"metagenomes\"},\"renderers\":[{\"id\":\"3823\",\"type\":\"GlyphRenderer\"}]},\"id\":\"3830\",\"type\":\"LegendItem\"},{\"attributes\":{\"fields\":[\"env\"]},\"id\":\"3786\",\"type\":\"Stack\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3785\",\"type\":\"Stack\"}},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"top\":{\"expr\":{\"id\":\"3786\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3798\",\"type\":\"VBar\"},{\"attributes\":{\"source\":{\"id\":\"3819\",\"type\":\"ColumnDataSource\"}},\"id\":\"3824\",\"type\":\"CDSView\"},{\"attributes\":{\"callback\":null,\"data\":{\"archaea\":[0.0,958.0,0.0,0.0,3541.0,0.0,0.0],\"bacteria\":[0.0,42908.0,4644.0,3840.0,315772.0,0.0,93221.0],\"databases\":[\"ENA\",\"JGI\",\"MGnify\",\"Metasub\",\"NCBI\",\"Segata-Pasolli\",\"ensembl\"],\"env\":[51876.0,0.0,0.0,0.0,0.0,0.0,0.0],\"metagenomes\":[0.0,5072.0,0.0,0.0,10402.0,9395.0,0.0],\"pro\":[146559.0,0.0,0.0,0.0,0.0,0.0,0.0]},\"selected\":{\"id\":\"3970\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"3971\",\"type\":\"UnionRenderers\"}},\"id\":\"3801\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"fields\":[\"env\",\"pro\",\"archaea\",\"bacteria\"]},\"id\":\"3792\",\"type\":\"Stack\"},{\"attributes\":{\"callback\":null,\"data\":{\"archaea\":[0.0,958.0,0.0,0.0,3541.0,0.0,0.0],\"bacteria\":[0.0,42908.0,4644.0,3840.0,315772.0,0.0,93221.0],\"databases\":[\"ENA\",\"JGI\",\"MGnify\",\"Metasub\",\"NCBI\",\"Segata-Pasolli\",\"ensembl\"],\"env\":[51876.0,0.0,0.0,0.0,0.0,0.0,0.0],\"metagenomes\":[0.0,5072.0,0.0,0.0,10402.0,9395.0,0.0],\"pro\":[146559.0,0.0,0.0,0.0,0.0,0.0,0.0]},\"selected\":{\"id\":\"3976\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"3977\",\"type\":\"UnionRenderers\"}},\"id\":\"3819\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3789\",\"type\":\"Stack\"}},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"top\":{\"expr\":{\"id\":\"3790\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3810\",\"type\":\"VBar\"},{\"attributes\":{},\"id\":\"3976\",\"type\":\"Selection\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3793\",\"type\":\"Stack\"}},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"top\":{\"expr\":{\"id\":\"3794\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3822\",\"type\":\"VBar\"},{\"attributes\":{\"fields\":[]},\"id\":\"3785\",\"type\":\"Stack\"},{\"attributes\":{\"source\":{\"id\":\"3795\",\"type\":\"ColumnDataSource\"}},\"id\":\"3800\",\"type\":\"CDSView\"},{\"attributes\":{\"callback\":null,\"data\":{\"archaea\":[0.0,958.0,0.0,0.0,3541.0,0.0,0.0],\"bacteria\":[0.0,42908.0,4644.0,3840.0,315772.0,0.0,93221.0],\"databases\":[\"ENA\",\"JGI\",\"MGnify\",\"Metasub\",\"NCBI\",\"Segata-Pasolli\",\"ensembl\"],\"env\":[51876.0,0.0,0.0,0.0,0.0,0.0,0.0],\"metagenomes\":[0.0,5072.0,0.0,0.0,10402.0,9395.0,0.0],\"pro\":[146559.0,0.0,0.0,0.0,0.0,0.0,0.0]},\"selected\":{\"id\":\"3968\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"3969\",\"type\":\"UnionRenderers\"}},\"id\":\"3795\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"items\":[{\"id\":\"3826\",\"type\":\"LegendItem\"},{\"id\":\"3827\",\"type\":\"LegendItem\"},{\"id\":\"3828\",\"type\":\"LegendItem\"},{\"id\":\"3829\",\"type\":\"LegendItem\"},{\"id\":\"3830\",\"type\":\"LegendItem\"}],\"location\":[0,30],\"plot\":{\"id\":\"3763\",\"subtype\":\"Figure\",\"type\":\"Plot\"}},\"id\":\"3825\",\"type\":\"Legend\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3787\",\"type\":\"Stack\"}},\"fill_color\":{\"value\":\"#3B518A\"},\"line_color\":{\"value\":\"#3B518A\"},\"top\":{\"expr\":{\"id\":\"3788\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3803\",\"type\":\"VBar\"},{\"attributes\":{\"data_source\":{\"id\":\"3801\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"3803\",\"type\":\"VBar\"},\"hover_glyph\":null,\"muted_glyph\":null,\"name\":\"pro\",\"nonselection_glyph\":{\"id\":\"3804\",\"type\":\"VBar\"},\"selection_glyph\":null,\"view\":{\"id\":\"3806\",\"type\":\"CDSView\"}},\"id\":\"3805\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3793\",\"type\":\"Stack\"}},\"fill_color\":{\"value\":\"#FDE724\"},\"line_color\":{\"value\":\"#FDE724\"},\"top\":{\"expr\":{\"id\":\"3794\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3821\",\"type\":\"VBar\"},{\"attributes\":{\"fields\":[\"env\"]},\"id\":\"3787\",\"type\":\"Stack\"},{\"attributes\":{\"callback\":null,\"data\":{\"archaea\":[0.0,958.0,0.0,0.0,3541.0,0.0,0.0],\"bacteria\":[0.0,42908.0,4644.0,3840.0,315772.0,0.0,93221.0],\"databases\":[\"ENA\",\"JGI\",\"MGnify\",\"Metasub\",\"NCBI\",\"Segata-Pasolli\",\"ensembl\"],\"env\":[51876.0,0.0,0.0,0.0,0.0,0.0,0.0],\"metagenomes\":[0.0,5072.0,0.0,0.0,10402.0,9395.0,0.0],\"pro\":[146559.0,0.0,0.0,0.0,0.0,0.0,0.0]},\"selected\":{\"id\":\"3972\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"3973\",\"type\":\"UnionRenderers\"}},\"id\":\"3807\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"grid_line_color\":{\"value\":null},\"plot\":{\"id\":\"3763\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"3774\",\"type\":\"CategoricalTicker\"}},\"id\":\"3776\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"3970\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"3769\",\"type\":\"CategoricalScale\"},{\"attributes\":{\"label\":{\"value\":\"bacteria\"},\"renderers\":[{\"id\":\"3817\",\"type\":\"GlyphRenderer\"}]},\"id\":\"3829\",\"type\":\"LegendItem\"},{\"attributes\":{},\"id\":\"3971\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"plot\":null,\"text\":\"File count by database\"},\"id\":\"3762\",\"type\":\"Title\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3789\",\"type\":\"Stack\"}},\"fill_color\":{\"value\":\"#208F8C\"},\"line_color\":{\"value\":\"#208F8C\"},\"top\":{\"expr\":{\"id\":\"3790\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3809\",\"type\":\"VBar\"},{\"attributes\":{\"label\":{\"value\":\"pro\"},\"renderers\":[{\"id\":\"3805\",\"type\":\"GlyphRenderer\"}]},\"id\":\"3827\",\"type\":\"LegendItem\"},{\"attributes\":{\"data_source\":{\"id\":\"3807\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"3809\",\"type\":\"VBar\"},\"hover_glyph\":null,\"muted_glyph\":null,\"name\":\"archaea\",\"nonselection_glyph\":{\"id\":\"3810\",\"type\":\"VBar\"},\"selection_glyph\":null,\"view\":{\"id\":\"3812\",\"type\":\"CDSView\"}},\"id\":\"3811\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"fields\":[\"env\",\"pro\",\"archaea\"]},\"id\":\"3791\",\"type\":\"Stack\"},{\"attributes\":{},\"id\":\"3974\",\"type\":\"Selection\"},{\"attributes\":{},\"id\":\"3774\",\"type\":\"CategoricalTicker\"},{\"attributes\":{\"bottom\":{\"expr\":{\"id\":\"3787\",\"type\":\"Stack\"}},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"top\":{\"expr\":{\"id\":\"3788\",\"type\":\"Stack\"}},\"width\":{\"value\":0.9},\"x\":{\"field\":\"databases\"}},\"id\":\"3804\",\"type\":\"VBar\"},{\"attributes\":{\"label\":{\"value\":\"archaea\"},\"renderers\":[{\"id\":\"3811\",\"type\":\"GlyphRenderer\"}]},\"id\":\"3828\",\"type\":\"LegendItem\"},{\"attributes\":{\"fields\":[\"env\",\"pro\",\"archaea\"]},\"id\":\"3790\",\"type\":\"Stack\"},{\"attributes\":{\"fields\":[\"env\",\"pro\",\"archaea\",\"bacteria\",\"metagenomes\"]},\"id\":\"3794\",\"type\":\"Stack\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"3782\",\"type\":\"HoverTool\"}]},\"id\":\"3783\",\"type\":\"Toolbar\"},{\"attributes\":{\"data_source\":{\"id\":\"3819\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"3821\",\"type\":\"VBar\"},\"hover_glyph\":null,\"muted_glyph\":null,\"name\":\"metagenomes\",\"nonselection_glyph\":{\"id\":\"3822\",\"type\":\"VBar\"},\"selection_glyph\":null,\"view\":{\"id\":\"3824\",\"type\":\"CDSView\"}},\"id\":\"3823\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"fields\":[\"env\",\"pro\"]},\"id\":\"3788\",\"type\":\"Stack\"},{\"attributes\":{},\"id\":\"3975\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"3972\",\"type\":\"Selection\"},{\"attributes\":{\"source\":{\"id\":\"3801\",\"type\":\"ColumnDataSource\"}},\"id\":\"3806\",\"type\":\"CDSView\"},{\"attributes\":{\"fields\":[\"env\",\"pro\"]},\"id\":\"3789\",\"type\":\"Stack\"},{\"attributes\":{\"formatter\":{\"id\":\"3964\",\"type\":\"CategoricalTickFormatter\"},\"minor_tick_line_color\":{\"value\":null},\"plot\":{\"id\":\"3763\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"3774\",\"type\":\"CategoricalTicker\"}},\"id\":\"3773\",\"type\":\"CategoricalAxis\"},{\"attributes\":{},\"id\":\"3771\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"3973\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"3966\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"label\":{\"value\":\"env\"},\"renderers\":[{\"id\":\"3799\",\"type\":\"GlyphRenderer\"}]},\"id\":\"3826\",\"type\":\"LegendItem\"},{\"attributes\":{\"data_source\":{\"id\":\"3813\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"3815\",\"type\":\"VBar\"},\"hover_glyph\":null,\"muted_glyph\":null,\"name\":\"bacteria\",\"nonselection_glyph\":{\"id\":\"3816\",\"type\":\"VBar\"},\"selection_glyph\":null,\"view\":{\"id\":\"3818\",\"type\":\"CDSView\"}},\"id\":\"3817\",\"type\":\"GlyphRenderer\"}],\"root_ids\":[\"3763\"]},\"title\":\"Bokeh Application\",\"version\":\"1.0.4\"}};\n",
       "  var render_items = [{\"docid\":\"732b9776-99f0-4e1a-8de2-ea752639ca72\",\"roots\":{\"3763\":\"5f1e9261-8d44-49dc-80ec-45fb7b4fbcff\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "3763"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "from bokeh.io import show, output_file\n",
    "from bokeh.plotting import figure\n",
    "from bokeh.models import Legend\n",
    "from bokeh.palettes import Viridis256, viridis\n",
    "\n",
    "#output_file(\"stacked.html\")\n",
    "\n",
    "\n",
    "colors =  viridis(5)\n",
    "\n",
    "p = figure(x_range=databases, plot_height=700, title=\"File count by database\",\n",
    "           toolbar_location=None, tools=\"hover\", tooltips=\"@databases $name: @$name\")\n",
    "\n",
    "rs = p.vbar_stack(sub_dbs, x='databases', width=0.9, color=colors, source=data_dir)\n",
    "\n",
    "\n",
    "p.y_range.start = 0\n",
    "p.x_range.range_padding = 0.1\n",
    "p.xgrid.grid_line_color = None\n",
    "p.axis.minor_tick_line_color = None\n",
    "p.outline_line_color = None\n",
    "#p.legend.location = \"top_left\"\n",
    "#p.legend.orientation = \"horizontal\"\n",
    "\n",
    "\n",
    "legend = Legend(items=[(sub, [r]) for (sub, r) in zip(sub_dbs, rs)], location=(0, 30))\n",
    "p.add_layout(legend, 'right')\n",
    "\n",
    "show(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## File Sizes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3020: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "data = pd.read_csv('fasta_index_file_oa.txt', sep=\"\\t\", header=None, names = [\"id\", \"filename_id\", \"filename\", \"database\",\"sub_database\",\"filename_2\",\"7\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "sizes = pd.read_csv('all.txt', sep=\" \", header=None, names = [\"filename_id\", \"size\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "sizes[\"filename_id\"]=sizes[\"filename_id\"].apply(lambda x: x.strip())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [],
   "source": [
    "data[\"filename_id\"]=data[\"filename_id\"].apply(lambda x: x.strip())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "size_data=data.set_index(\"filename_id\").join(sizes.set_index(\"filename_id\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot by database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot by subdatabase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot by datatype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Bar graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db=size_data.groupby(\"database\")[\"size\"].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db=by_db.reset_index(name='value').rename(columns={'index':'database'})\n",
    "#by_db['angle'] = by_db['value']/by_db['value'].sum() * 2*pi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db['angle'] = by_db['value']/by_db['value'].sum() * 2*pi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db[\"value\"] = by_db[\"value\"].apply(lambda x: round(x/(10**9),4)) # Gigabyte"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>2588.9170</td>\n",
       "      <td>1.979102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JGI</td>\n",
       "      <td>991.7704</td>\n",
       "      <td>0.758160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>11.3817</td>\n",
       "      <td>0.008701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>680.2256</td>\n",
       "      <td>0.520000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>2145.8124</td>\n",
       "      <td>1.640370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>1461.6472</td>\n",
       "      <td>1.117358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>339.4527</td>\n",
       "      <td>0.259495</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           database      value     angle\n",
       "0              ENA   2588.9170  1.979102\n",
       "1              JGI    991.7704  0.758160\n",
       "2           MGnify     11.3817  0.008701\n",
       "3          Metasub    680.2256  0.520000\n",
       "4             NCBI   2145.8124  1.640370\n",
       "5   Segata-Pasolli   1461.6472  1.117358\n",
       "6          ensembl    339.4527  0.259495"
      ]
     },
     "execution_count": 367,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "metadata": {},
   "outputs": [],
   "source": [
    "color_dict={'NCBI': Blues[3].copy(),\n",
    " 'JGI': Oranges[6].copy(),\n",
    " 'ENA': Greens[3].copy(),\n",
    " 'ensembl': [ \"#d62728\"],\n",
    " 'MGnify': [\"#9467bd\"],\n",
    " 'Metasub': [\"#8c564b\"],\n",
    " 'Segata-Pasolli': [\"#e377c2\"]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "metadata": {},
   "outputs": [],
   "source": [
    "#colors= [i[0] for i in color_dict.values()]\n",
    "colors=Greens[len(color_dict.values())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['#005a32', '#238b45', '#41ab5d', '#74c476', '#a1d99b', '#c7e9c0', '#edf8e9']"
      ]
     },
     "execution_count": 370,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "colors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"10605\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"10605\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n",
       "    }\n",
       "    finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.info(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(js_urls, callback) {\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = js_urls.length;\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var s = document.createElement('script');\n",
       "      s.src = url;\n",
       "      s.async = false;\n",
       "      s.onreadystatechange = s.onload = function() {\n",
       "        root._bokeh_is_loading--;\n",
       "        if (root._bokeh_is_loading === 0) {\n",
       "          console.log(\"Bokeh: all BokehJS libraries loaded\");\n",
       "          run_callbacks()\n",
       "        }\n",
       "      };\n",
       "      s.onerror = function() {\n",
       "        console.warn(\"failed to load library \" + url);\n",
       "      };\n",
       "      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "    }\n",
       "  };var element = document.getElementById(\"10605\");\n",
       "  if (element == null) {\n",
       "    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '10605' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    \n",
       "    function(Bokeh) {\n",
       "      \n",
       "    },\n",
       "    function(Bokeh) {\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if ((root.Bokeh !== undefined) || (force === true)) {\n",
       "      for (var i = 0; i < inline_js.length; i++) {\n",
       "        inline_js[i].call(root, root.Bokeh);\n",
       "      }if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"10605\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(js_urls, function() {\n",
       "      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"10605\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n    }\n    finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.info(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(js_urls, callback) {\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = js_urls.length;\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var s = document.createElement('script');\n      s.src = url;\n      s.async = false;\n      s.onreadystatechange = s.onload = function() {\n        root._bokeh_is_loading--;\n        if (root._bokeh_is_loading === 0) {\n          console.log(\"Bokeh: all BokehJS libraries loaded\");\n          run_callbacks()\n        }\n      };\n      s.onerror = function() {\n        console.warn(\"failed to load library \" + url);\n      };\n      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.getElementsByTagName(\"head\")[0].appendChild(s);\n    }\n  };var element = document.getElementById(\"10605\");\n  if (element == null) {\n    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '10605' but no matching script tag was found. \")\n    return false;\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n    }\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"10605\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(js_urls, function() {\n      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"f7e223fe-9374-475e-b12d-df5854abde1b\" data-root-id=\"10607\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"d50298ea-a73d-4a18-bdd7-74af340530ff\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"10617\",\"type\":\"LinearAxis\"}],\"left\":[{\"id\":\"10622\",\"type\":\"LinearAxis\"}],\"plot_height\":500,\"plot_width\":800,\"renderers\":[{\"id\":\"10617\",\"type\":\"LinearAxis\"},{\"id\":\"10621\",\"type\":\"Grid\"},{\"id\":\"10622\",\"type\":\"LinearAxis\"},{\"id\":\"10626\",\"type\":\"Grid\"},{\"id\":\"10643\",\"type\":\"Legend\"},{\"id\":\"10636\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"10606\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"10628\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"10609\",\"type\":\"DataRange1d\"},\"x_scale\":{\"id\":\"10613\",\"type\":\"LinearScale\"},\"y_range\":{\"id\":\"10611\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"10615\",\"type\":\"LinearScale\"}},\"id\":\"10607\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{},\"id\":\"10679\",\"type\":\"Selection\"},{\"attributes\":{\"plot\":null,\"text\":\"Genomic Data on Server by Database (in GB)\"},\"id\":\"10606\",\"type\":\"Title\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"10631\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"10630\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"10635\",\"type\":\"Wedge\"},{\"attributes\":{\"field\":\"angle\"},\"id\":\"10631\",\"type\":\"CumSum\"},{\"attributes\":{\"formatter\":{\"id\":\"10641\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"10607\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10623\",\"type\":\"BasicTicker\"}},\"id\":\"10622\",\"type\":\"LinearAxis\"},{\"attributes\":{},\"id\":\"10613\",\"type\":\"LinearScale\"},{\"attributes\":{\"source\":{\"id\":\"10632\",\"type\":\"ColumnDataSource\"}},\"id\":\"10637\",\"type\":\"CDSView\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"10631\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_color\":{\"field\":\"color\"},\"line_color\":{\"value\":\"white\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"10630\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"10634\",\"type\":\"Wedge\"},{\"attributes\":{\"callback\":null},\"id\":\"10611\",\"type\":\"DataRange1d\"},{\"attributes\":{},\"id\":\"10639\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"items\":[{\"id\":\"10644\",\"type\":\"LegendItem\"}],\"plot\":{\"id\":\"10607\",\"subtype\":\"Figure\",\"type\":\"Plot\"}},\"id\":\"10643\",\"type\":\"Legend\"},{\"attributes\":{},\"id\":\"10618\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"10623\",\"type\":\"BasicTicker\"},{\"attributes\":{\"label\":{\"field\":\"database\"},\"renderers\":[{\"id\":\"10636\",\"type\":\"GlyphRenderer\"}]},\"id\":\"10644\",\"type\":\"LegendItem\"},{\"attributes\":{\"formatter\":{\"id\":\"10639\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"10607\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10618\",\"type\":\"BasicTicker\"}},\"id\":\"10617\",\"type\":\"LinearAxis\"},{\"attributes\":{\"dimension\":1,\"plot\":{\"id\":\"10607\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10623\",\"type\":\"BasicTicker\"}},\"id\":\"10626\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"10680\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"10641\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"plot\":{\"id\":\"10607\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10618\",\"type\":\"BasicTicker\"}},\"id\":\"10621\",\"type\":\"Grid\"},{\"attributes\":{},\"id\":\"10615\",\"type\":\"LinearScale\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"10627\",\"type\":\"HoverTool\"}]},\"id\":\"10628\",\"type\":\"Toolbar\"},{\"attributes\":{\"data_source\":{\"id\":\"10632\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"10634\",\"type\":\"Wedge\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"10635\",\"type\":\"Wedge\"},\"selection_glyph\":null,\"view\":{\"id\":\"10637\",\"type\":\"CDSView\"}},\"id\":\"10636\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"field\":\"angle\",\"include_zero\":true},\"id\":\"10630\",\"type\":\"CumSum\"},{\"attributes\":{\"callback\":null,\"data\":{\"angle\":{\"__ndarray__\":\"XQ7WWGaq/z96RNOY2ULoP/iCh/aw0YE/N8SnB9aj4D9H+Rkl9D76PygJ1EKz4PE/0kCIeJGb0D8=\",\"dtype\":\"float64\",\"shape\":[7]},\"color\":[\"#005a32\",\"#238b45\",\"#41ab5d\",\"#74c476\",\"#a1d99b\",\"#c7e9c0\",\"#edf8e9\"],\"database\":[\" ENA \",\" JGI \",\" MGnify \",\" Metasub \",\" NCBI \",\" Segata-Pasolli \",\" ensembl \"],\"index\":[0,1,2,3,4,5,6],\"value\":{\"__ndarray__\":\"3SQGgdU5pEC1pnnHKf6OQDPEsS5uwyZA0m9fB85BhUCKjuTyn8OgQOPHmLuW1pZAY+5aQj43dUA=\",\"dtype\":\"float64\",\"shape\":[7]}},\"selected\":{\"id\":\"10679\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"10680\",\"type\":\"UnionRenderers\"}},\"id\":\"10632\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"callback\":null,\"tooltips\":\"@database: @value GB\"},\"id\":\"10627\",\"type\":\"HoverTool\"},{\"attributes\":{\"callback\":null},\"id\":\"10609\",\"type\":\"DataRange1d\"}],\"root_ids\":[\"10607\"]},\"title\":\"Bokeh Application\",\"version\":\"1.0.4\"}};\n",
       "  var render_items = [{\"docid\":\"d50298ea-a73d-4a18-bdd7-74af340530ff\",\"roots\":{\"10607\":\"f7e223fe-9374-475e-b12d-df5854abde1b\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "10607"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "output_notebook();\n",
    "output_file(\"by_db.html\")\n",
    "by_db['color'] = colors\n",
    "\n",
    "p = figure(plot_height=500, plot_width=800,title=\"Genomic Data on Server by Database (in GB)\", toolbar_location=None,\n",
    "        tools=\"hover\", tooltips=\"@database: @value GB\")\n",
    "\n",
    "p.wedge(x=0, y=1, radius=0.4,\n",
    "        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),\n",
    "        line_color=\"white\", fill_color='color', legend='database', source=by_db)\n",
    "\n",
    "show(p);"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### By subdatabase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub=size_data.groupby([\"database\",\"sub_database\"])[\"size\"].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub.reset_index(name='value').rename(columns={'index':'sub_database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot[\"value\"] = by_sub_plot[\"value\"].apply(lambda x: round(x/(10**9),4)) # Gigabyte"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot['angle'] = by_sub_plot['value']/by_sub_plot['value'].sum() * 2*pi\n",
    "by_sub_plot['complete_database'] = by_sub_plot['database']+by_sub_plot['sub_database']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "      <th>complete_database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>1959.3144</td>\n",
       "      <td>1.497801</td>\n",
       "      <td>ENA  env</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>629.6026</td>\n",
       "      <td>0.481301</td>\n",
       "      <td>ENA  pro</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>1.4988</td>\n",
       "      <td>0.001146</td>\n",
       "      <td>JGI  JGI_archaea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>0.3575</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>JGI  JGI_archaea_v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>254.7375</td>\n",
       "      <td>0.194734</td>\n",
       "      <td>JGI  JGI_bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>2.2356</td>\n",
       "      <td>0.001709</td>\n",
       "      <td>JGI  JGI_bacteria_v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>709.9082</td>\n",
       "      <td>0.542690</td>\n",
       "      <td>JGI  JGI_metagenomes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>23.0328</td>\n",
       "      <td>0.017607</td>\n",
       "      <td>JGI  JGI_metagenomes_v2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>11.3817</td>\n",
       "      <td>0.008701</td>\n",
       "      <td>MGnify  bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>680.2256</td>\n",
       "      <td>0.519999</td>\n",
       "      <td>Metasub  bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>6.9064</td>\n",
       "      <td>0.005280</td>\n",
       "      <td>NCBI  archaea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>1200.5480</td>\n",
       "      <td>0.917761</td>\n",
       "      <td>NCBI  bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>938.3580</td>\n",
       "      <td>0.717329</td>\n",
       "      <td>NCBI  metagenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>1461.6472</td>\n",
       "      <td>1.117358</td>\n",
       "      <td>Segata-Pasolli  metagenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>339.4527</td>\n",
       "      <td>0.259495</td>\n",
       "      <td>ensembl  bacteria</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database      value     angle  \\\n",
       "0               ENA                   env   1959.3144  1.497801   \n",
       "1               ENA                   pro    629.6026  0.481301   \n",
       "2               JGI           JGI_archaea      1.4988  0.001146   \n",
       "3               JGI        JGI_archaea_v2      0.3575  0.000273   \n",
       "4               JGI          JGI_bacteria    254.7375  0.194734   \n",
       "5               JGI       JGI_bacteria_v2      2.2356  0.001709   \n",
       "6               JGI       JGI_metagenomes    709.9082  0.542690   \n",
       "7               JGI    JGI_metagenomes_v2     23.0328  0.017607   \n",
       "8            MGnify              bacteria     11.3817  0.008701   \n",
       "9           Metasub              bacteria    680.2256  0.519999   \n",
       "10             NCBI               archaea      6.9064  0.005280   \n",
       "11             NCBI              bacteria   1200.5480  0.917761   \n",
       "12             NCBI            metagenome    938.3580  0.717329   \n",
       "13   Segata-Pasolli            metagenome   1461.6472  1.117358   \n",
       "14          ensembl              bacteria    339.4527  0.259495   \n",
       "\n",
       "               complete_database  \n",
       "0                      ENA  env   \n",
       "1                      ENA  pro   \n",
       "2              JGI  JGI_archaea   \n",
       "3           JGI  JGI_archaea_v2   \n",
       "4             JGI  JGI_bacteria   \n",
       "5          JGI  JGI_bacteria_v2   \n",
       "6          JGI  JGI_metagenomes   \n",
       "7       JGI  JGI_metagenomes_v2   \n",
       "8              MGnify  bacteria   \n",
       "9             Metasub  bacteria   \n",
       "10                NCBI  archaea   \n",
       "11               NCBI  bacteria   \n",
       "12             NCBI  metagenome   \n",
       "13   Segata-Pasolli  metagenome   \n",
       "14            ensembl  bacteria   "
      ]
     },
     "execution_count": 376,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 377,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\ipykernel_launcher.py:7: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  import sys\n"
     ]
    }
   ],
   "source": [
    "for index, row in by_sub_plot.iterrows():\n",
    "    #print (color_dict)\n",
    "    \n",
    "    db_name=row['database'].split()[0]\n",
    "    \n",
    "    co=color_dict[db_name].pop()\n",
    "    by_sub_plot.set_value(index,\"color\",co)\n",
    "\n",
    "#by_sub_plot = by_sub_plot.sort_values(by=\"database\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "      <th>complete_database</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>1959.3144</td>\n",
       "      <td>1.497801</td>\n",
       "      <td>ENA  env</td>\n",
       "      <td>#e5f5e0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>629.6026</td>\n",
       "      <td>0.481301</td>\n",
       "      <td>ENA  pro</td>\n",
       "      <td>#a1d99b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>1.4988</td>\n",
       "      <td>0.001146</td>\n",
       "      <td>JGI  JGI_archaea</td>\n",
       "      <td>#feedde</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>0.3575</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>JGI  JGI_archaea_v2</td>\n",
       "      <td>#fdd0a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>254.7375</td>\n",
       "      <td>0.194734</td>\n",
       "      <td>JGI  JGI_bacteria</td>\n",
       "      <td>#fdae6b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>2.2356</td>\n",
       "      <td>0.001709</td>\n",
       "      <td>JGI  JGI_bacteria_v2</td>\n",
       "      <td>#fd8d3c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>709.9082</td>\n",
       "      <td>0.542690</td>\n",
       "      <td>JGI  JGI_metagenomes</td>\n",
       "      <td>#e6550d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>23.0328</td>\n",
       "      <td>0.017607</td>\n",
       "      <td>JGI  JGI_metagenomes_v2</td>\n",
       "      <td>#a63603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>11.3817</td>\n",
       "      <td>0.008701</td>\n",
       "      <td>MGnify  bacteria</td>\n",
       "      <td>#9467bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>680.2256</td>\n",
       "      <td>0.519999</td>\n",
       "      <td>Metasub  bacteria</td>\n",
       "      <td>#8c564b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>6.9064</td>\n",
       "      <td>0.005280</td>\n",
       "      <td>NCBI  archaea</td>\n",
       "      <td>#deebf7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>1200.5480</td>\n",
       "      <td>0.917761</td>\n",
       "      <td>NCBI  bacteria</td>\n",
       "      <td>#9ecae1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>938.3580</td>\n",
       "      <td>0.717329</td>\n",
       "      <td>NCBI  metagenome</td>\n",
       "      <td>#3182bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>1461.6472</td>\n",
       "      <td>1.117358</td>\n",
       "      <td>Segata-Pasolli  metagenome</td>\n",
       "      <td>#e377c2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>339.4527</td>\n",
       "      <td>0.259495</td>\n",
       "      <td>ensembl  bacteria</td>\n",
       "      <td>#d62728</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database      value     angle  \\\n",
       "0               ENA                   env   1959.3144  1.497801   \n",
       "1               ENA                   pro    629.6026  0.481301   \n",
       "2               JGI           JGI_archaea      1.4988  0.001146   \n",
       "3               JGI        JGI_archaea_v2      0.3575  0.000273   \n",
       "4               JGI          JGI_bacteria    254.7375  0.194734   \n",
       "5               JGI       JGI_bacteria_v2      2.2356  0.001709   \n",
       "6               JGI       JGI_metagenomes    709.9082  0.542690   \n",
       "7               JGI    JGI_metagenomes_v2     23.0328  0.017607   \n",
       "8            MGnify              bacteria     11.3817  0.008701   \n",
       "9           Metasub              bacteria    680.2256  0.519999   \n",
       "10             NCBI               archaea      6.9064  0.005280   \n",
       "11             NCBI              bacteria   1200.5480  0.917761   \n",
       "12             NCBI            metagenome    938.3580  0.717329   \n",
       "13   Segata-Pasolli            metagenome   1461.6472  1.117358   \n",
       "14          ensembl              bacteria    339.4527  0.259495   \n",
       "\n",
       "               complete_database    color  \n",
       "0                      ENA  env   #e5f5e0  \n",
       "1                      ENA  pro   #a1d99b  \n",
       "2              JGI  JGI_archaea   #feedde  \n",
       "3           JGI  JGI_archaea_v2   #fdd0a2  \n",
       "4             JGI  JGI_bacteria   #fdae6b  \n",
       "5          JGI  JGI_bacteria_v2   #fd8d3c  \n",
       "6          JGI  JGI_metagenomes   #e6550d  \n",
       "7       JGI  JGI_metagenomes_v2   #a63603  \n",
       "8              MGnify  bacteria   #9467bd  \n",
       "9             Metasub  bacteria   #8c564b  \n",
       "10                NCBI  archaea   #deebf7  \n",
       "11               NCBI  bacteria   #9ecae1  \n",
       "12             NCBI  metagenome   #3182bd  \n",
       "13   Segata-Pasolli  metagenome   #e377c2  \n",
       "14            ensembl  bacteria   #d62728  "
      ]
     },
     "execution_count": 378,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 379,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"10787\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"10787\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n",
       "    }\n",
       "    finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.info(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(js_urls, callback) {\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = js_urls.length;\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var s = document.createElement('script');\n",
       "      s.src = url;\n",
       "      s.async = false;\n",
       "      s.onreadystatechange = s.onload = function() {\n",
       "        root._bokeh_is_loading--;\n",
       "        if (root._bokeh_is_loading === 0) {\n",
       "          console.log(\"Bokeh: all BokehJS libraries loaded\");\n",
       "          run_callbacks()\n",
       "        }\n",
       "      };\n",
       "      s.onerror = function() {\n",
       "        console.warn(\"failed to load library \" + url);\n",
       "      };\n",
       "      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "    }\n",
       "  };var element = document.getElementById(\"10787\");\n",
       "  if (element == null) {\n",
       "    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '10787' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    \n",
       "    function(Bokeh) {\n",
       "      \n",
       "    },\n",
       "    function(Bokeh) {\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if ((root.Bokeh !== undefined) || (force === true)) {\n",
       "      for (var i = 0; i < inline_js.length; i++) {\n",
       "        inline_js[i].call(root, root.Bokeh);\n",
       "      }if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"10787\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(js_urls, function() {\n",
       "      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"10787\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n    }\n    finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.info(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(js_urls, callback) {\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = js_urls.length;\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var s = document.createElement('script');\n      s.src = url;\n      s.async = false;\n      s.onreadystatechange = s.onload = function() {\n        root._bokeh_is_loading--;\n        if (root._bokeh_is_loading === 0) {\n          console.log(\"Bokeh: all BokehJS libraries loaded\");\n          run_callbacks()\n        }\n      };\n      s.onerror = function() {\n        console.warn(\"failed to load library \" + url);\n      };\n      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.getElementsByTagName(\"head\")[0].appendChild(s);\n    }\n  };var element = document.getElementById(\"10787\");\n  if (element == null) {\n    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '10787' but no matching script tag was found. \")\n    return false;\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n    }\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"10787\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(js_urls, function() {\n      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"1abba0da-aee1-4c05-b1ed-2bced1f1d80f\" data-root-id=\"10789\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"8daa3a36-9f1f-4ccf-814e-d977dcce9fb4\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"10799\",\"type\":\"LinearAxis\"}],\"left\":[{\"id\":\"10804\",\"type\":\"LinearAxis\"}],\"plot_height\":500,\"plot_width\":800,\"renderers\":[{\"id\":\"10799\",\"type\":\"LinearAxis\"},{\"id\":\"10803\",\"type\":\"Grid\"},{\"id\":\"10804\",\"type\":\"LinearAxis\"},{\"id\":\"10808\",\"type\":\"Grid\"},{\"id\":\"10825\",\"type\":\"Legend\"},{\"id\":\"10818\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"10788\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"10810\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"10791\",\"type\":\"DataRange1d\"},\"x_scale\":{\"id\":\"10795\",\"type\":\"LinearScale\"},\"y_range\":{\"id\":\"10793\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"10797\",\"type\":\"LinearScale\"}},\"id\":\"10789\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{},\"id\":\"10800\",\"type\":\"BasicTicker\"},{\"attributes\":{\"items\":[{\"id\":\"10826\",\"type\":\"LegendItem\"}],\"plot\":{\"id\":\"10789\",\"subtype\":\"Figure\",\"type\":\"Plot\"}},\"id\":\"10825\",\"type\":\"Legend\"},{\"attributes\":{\"callback\":null},\"id\":\"10791\",\"type\":\"DataRange1d\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"10809\",\"type\":\"HoverTool\"}]},\"id\":\"10810\",\"type\":\"Toolbar\"},{\"attributes\":{},\"id\":\"10868\",\"type\":\"Selection\"},{\"attributes\":{\"source\":{\"id\":\"10814\",\"type\":\"ColumnDataSource\"}},\"id\":\"10819\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"10797\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"10823\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"callback\":null,\"tooltips\":\"@complete_database: @value GB\"},\"id\":\"10809\",\"type\":\"HoverTool\"},{\"attributes\":{\"plot\":null,\"text\":\"Genomic Data on Server by Sub Database (in GB)\"},\"id\":\"10788\",\"type\":\"Title\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"10813\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_color\":{\"field\":\"color\"},\"line_color\":{\"value\":\"white\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"10812\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"10816\",\"type\":\"Wedge\"},{\"attributes\":{\"label\":{\"field\":\"complete_database\"},\"renderers\":[{\"id\":\"10818\",\"type\":\"GlyphRenderer\"}]},\"id\":\"10826\",\"type\":\"LegendItem\"},{\"attributes\":{\"data_source\":{\"id\":\"10814\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"10816\",\"type\":\"Wedge\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"10817\",\"type\":\"Wedge\"},\"selection_glyph\":null,\"view\":{\"id\":\"10819\",\"type\":\"CDSView\"}},\"id\":\"10818\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"plot\":{\"id\":\"10789\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10800\",\"type\":\"BasicTicker\"}},\"id\":\"10803\",\"type\":\"Grid\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"10813\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"10812\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"10817\",\"type\":\"Wedge\"},{\"attributes\":{},\"id\":\"10805\",\"type\":\"BasicTicker\"},{\"attributes\":{},\"id\":\"10821\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"callback\":null},\"id\":\"10793\",\"type\":\"DataRange1d\"},{\"attributes\":{\"formatter\":{\"id\":\"10823\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"10789\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10805\",\"type\":\"BasicTicker\"}},\"id\":\"10804\",\"type\":\"LinearAxis\"},{\"attributes\":{},\"id\":\"10795\",\"type\":\"LinearScale\"},{\"attributes\":{\"field\":\"angle\"},\"id\":\"10813\",\"type\":\"CumSum\"},{\"attributes\":{\"dimension\":1,\"plot\":{\"id\":\"10789\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10805\",\"type\":\"BasicTicker\"}},\"id\":\"10808\",\"type\":\"Grid\"},{\"attributes\":{\"formatter\":{\"id\":\"10821\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"10789\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10800\",\"type\":\"BasicTicker\"}},\"id\":\"10799\",\"type\":\"LinearAxis\"},{\"attributes\":{},\"id\":\"10869\",\"type\":\"UnionRenderers\"},{\"attributes\":{\"field\":\"angle\",\"include_zero\":true},\"id\":\"10812\",\"type\":\"CumSum\"},{\"attributes\":{\"callback\":null,\"data\":{\"angle\":{\"__ndarray__\":\"eUNUDf729z9YdVdPoc3eP2KKz16qxVI/3e1ktRHpMT+c/48hD+3IP/ppsy0ZAFw/kudoRLhd4T9QN92SsAeSPzcC8PSz0YE/iTmG+9Wj4D+dU+v6EaB1P8CGHclLXu0/FFHjXlz05j/gSRU7s+DxPw1Y4nmRm9A/\",\"dtype\":\"float64\",\"shape\":[15]},\"color\":[\"#e5f5e0\",\"#a1d99b\",\"#feedde\",\"#fdd0a2\",\"#fdae6b\",\"#fd8d3c\",\"#e6550d\",\"#a63603\",\"#9467bd\",\"#8c564b\",\"#deebf7\",\"#9ecae1\",\"#3182bd\",\"#e377c2\",\"#d62728\"],\"complete_database\":[\" ENA  env \",\" ENA  pro \",\" JGI  JGI_archaea \",\" JGI  JGI_archaea_v2 \",\" JGI  JGI_bacteria \",\" JGI  JGI_bacteria_v2 \",\" JGI  JGI_metagenomes \",\" JGI  JGI_metagenomes_v2 \",\" MGnify  bacteria \",\" Metasub  bacteria \",\" NCBI  archaea \",\" NCBI  bacteria \",\" NCBI  metagenome \",\" Segata-Pasolli  metagenome \",\" ensembl  bacteria \"],\"database\":[\" ENA \",\" ENA \",\" JGI \",\" JGI \",\" JGI \",\" JGI \",\" JGI \",\" JGI \",\" MGnify \",\" Metasub \",\" NCBI \",\" NCBI \",\" NCBI \",\" Segata-Pasolli \",\" ensembl \"],\"index\":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14],\"sub_database\":[\" env \",\" pro \",\" JGI_archaea \",\" JGI_archaea_v2 \",\" JGI_bacteria \",\" JGI_bacteria_v2 \",\" JGI_metagenomes \",\" JGI_metagenomes_v2 \",\" bacteria \",\" bacteria \",\" archaea \",\" bacteria \",\" metagenome \",\" metagenome \",\" bacteria \"],\"value\":{\"__ndarray__\":\"c9cS8kGdnkCP5PIf0qyDQLPqc7UV+/c/4XoUrkfh1j+amZmZmddvQDSAt0CC4gFA0ZFc/kMvhkAOT6+UZQg3QDPEsS5uwyZA0m9fB85BhUCqYFRSJ6AbQNV46SYxwpJAvp8aL91SjUDjx5i7ltaWQGPuWkI+N3VA\",\"dtype\":\"float64\",\"shape\":[15]}},\"selected\":{\"id\":\"10868\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"10869\",\"type\":\"UnionRenderers\"}},\"id\":\"10814\",\"type\":\"ColumnDataSource\"}],\"root_ids\":[\"10789\"]},\"title\":\"Bokeh Application\",\"version\":\"1.0.4\"}};\n",
       "  var render_items = [{\"docid\":\"8daa3a36-9f1f-4ccf-814e-d977dcce9fb4\",\"roots\":{\"10789\":\"1abba0da-aee1-4c05-b1ed-2bced1f1d80f\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "10789"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "output_notebook();\n",
    "output_file(\"by_sub_db.html\")\n",
    "\n",
    "p = figure(plot_height=500, plot_width=800,title=\"Genomic Data on Server by Sub Database (in GB)\", toolbar_location=None,\n",
    "        tools=\"hover\", tooltips=\"@complete_database: @value GB\")\n",
    "\n",
    "p.wedge(x=0, y=1, radius=0.4,\n",
    "        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),\n",
    "        line_color=\"white\", fill_color='color', legend='complete_database', source=by_sub_plot)\n",
    "\n",
    "show(p);"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### By Data Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub.reset_index(name='value').rename(columns={'index':'sub_database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>1959314354852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>629602608048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>1498837900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>357506596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>254737477139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>2235606878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>709908200134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>23032781088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>11381670837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>680225630474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>6906394710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>1200548046299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>938357957590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>1461647239711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>339452698810</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database          value\n",
       "0               ENA                   env   1959314354852\n",
       "1               ENA                   pro    629602608048\n",
       "2               JGI           JGI_archaea      1498837900\n",
       "3               JGI        JGI_archaea_v2       357506596\n",
       "4               JGI          JGI_bacteria    254737477139\n",
       "5               JGI       JGI_bacteria_v2      2235606878\n",
       "6               JGI       JGI_metagenomes    709908200134\n",
       "7               JGI    JGI_metagenomes_v2     23032781088\n",
       "8            MGnify              bacteria     11381670837\n",
       "9           Metasub              bacteria    680225630474\n",
       "10             NCBI               archaea      6906394710\n",
       "11             NCBI              bacteria   1200548046299\n",
       "12             NCBI            metagenome    938357957590\n",
       "13   Segata-Pasolli            metagenome   1461647239711\n",
       "14          ensembl              bacteria    339452698810"
      ]
     },
     "execution_count": 381,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\ipykernel_launcher.py:16: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['env', 'pro', 'archaea', 'bacteria', 'metagenomes']"
      ]
     },
     "execution_count": 382,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_dbs_dir = {'metagenome': 'metagenomes',\n",
    " 'JGI_archaea':'archaea',\n",
    " 'JGI_bacteria':'bacteria',\n",
    " 'JGI_metagenomes':'metagenomes',\n",
    " 'JGI_metagenomes_v2':'metagenomes',\n",
    " 'JGI_bacteria_v2':\"bacteria\",\n",
    " 'JGI_archaea_v2':\"archaea\"}\n",
    "\n",
    "\n",
    "for index, row in by_sub_plot.iterrows():\n",
    "    #print (color_dict)\n",
    "    \n",
    "    db_name=row['sub_database']\n",
    "    \n",
    "    new_db = sub_dbs_dir.get(db_name.strip(),db_name.strip())\n",
    "    by_sub_plot.set_value(index,\"sub_database\",new_db)\n",
    "\n",
    "    \n",
    "sub_db = by_sub_plot.sub_database.unique()\n",
    "list(sub_db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub_plot.groupby(\"sub_database\")[\"value\"].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot=by_sub_plot.reset_index(name='value').rename(columns={'index':'sub_database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot[\"value\"] = by_sub_plot[\"value\"].apply(lambda x: round(x/(10**9),4)) # Gigabyte"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot['angle'] = by_sub_plot['value']/by_sub_plot['value'].sum() * 2*pi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 387,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_sub_plot['color'] = Blues[len(by_sub_plot)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "      <th>angle</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>archaea</td>\n",
       "      <td>8.7627</td>\n",
       "      <td>0.006699</td>\n",
       "      <td>#08519c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bacteria</td>\n",
       "      <td>2488.5811</td>\n",
       "      <td>1.902400</td>\n",
       "      <td>#3182bd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>env</td>\n",
       "      <td>1959.3144</td>\n",
       "      <td>1.497801</td>\n",
       "      <td>#6baed6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>metagenomes</td>\n",
       "      <td>3132.9462</td>\n",
       "      <td>2.394985</td>\n",
       "      <td>#bdd7e7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pro</td>\n",
       "      <td>629.6026</td>\n",
       "      <td>0.481301</td>\n",
       "      <td>#eff3ff</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sub_database      value     angle    color\n",
       "0      archaea     8.7627  0.006699  #08519c\n",
       "1     bacteria  2488.5811  1.902400  #3182bd\n",
       "2          env  1959.3144  1.497801  #6baed6\n",
       "3  metagenomes  3132.9462  2.394985  #bdd7e7\n",
       "4          pro   629.6026  0.481301  #eff3ff"
      ]
     },
     "execution_count": 388,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_sub_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div class=\"bk-root\">\n",
       "        <a href=\"https://bokeh.pydata.org\" target=\"_blank\" class=\"bk-logo bk-logo-small bk-logo-notebook\"></a>\n",
       "        <span id=\"10976\">Loading BokehJS ...</span>\n",
       "    </div>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  var force = true;\n",
       "\n",
       "  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "  var JS_MIME_TYPE = 'application/javascript';\n",
       "  var HTML_MIME_TYPE = 'text/html';\n",
       "  var EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  var CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    var script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    var cell = handle.cell;\n",
       "\n",
       "    var id = cell.output_area._bokeh_element_id;\n",
       "    var server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      var cmd = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            var id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      var cmd = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    var output_area = handle.output_area;\n",
       "    var output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      var bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      var script_attrs = bk_div.children[0].attributes;\n",
       "      for (var i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      var toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    var events = require('base/js/events');\n",
       "    var OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  \n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  var NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    var el = document.getElementById(\"10976\");\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n",
       "    }\n",
       "    finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.info(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(js_urls, callback) {\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = js_urls.length;\n",
       "    for (var i = 0; i < js_urls.length; i++) {\n",
       "      var url = js_urls[i];\n",
       "      var s = document.createElement('script');\n",
       "      s.src = url;\n",
       "      s.async = false;\n",
       "      s.onreadystatechange = s.onload = function() {\n",
       "        root._bokeh_is_loading--;\n",
       "        if (root._bokeh_is_loading === 0) {\n",
       "          console.log(\"Bokeh: all BokehJS libraries loaded\");\n",
       "          run_callbacks()\n",
       "        }\n",
       "      };\n",
       "      s.onerror = function() {\n",
       "        console.warn(\"failed to load library \" + url);\n",
       "      };\n",
       "      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "    }\n",
       "  };var element = document.getElementById(\"10976\");\n",
       "  if (element == null) {\n",
       "    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '10976' but no matching script tag was found. \")\n",
       "    return false;\n",
       "  }\n",
       "\n",
       "  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n",
       "\n",
       "  var inline_js = [\n",
       "    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "    \n",
       "    function(Bokeh) {\n",
       "      \n",
       "    },\n",
       "    function(Bokeh) {\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n",
       "      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    \n",
       "    if ((root.Bokeh !== undefined) || (force === true)) {\n",
       "      for (var i = 0; i < inline_js.length; i++) {\n",
       "        inline_js[i].call(root, root.Bokeh);\n",
       "      }if (force === true) {\n",
       "        display_loaded();\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      var cell = $(document.getElementById(\"10976\")).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(js_urls, function() {\n",
       "      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  var force = true;\n\n  if (typeof (root._bokeh_onload_callbacks) === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n  \n\n  \n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  var NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    var el = document.getElementById(\"10976\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) { callback() });\n    }\n    finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.info(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(js_urls, callback) {\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.log(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.log(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = js_urls.length;\n    for (var i = 0; i < js_urls.length; i++) {\n      var url = js_urls[i];\n      var s = document.createElement('script');\n      s.src = url;\n      s.async = false;\n      s.onreadystatechange = s.onload = function() {\n        root._bokeh_is_loading--;\n        if (root._bokeh_is_loading === 0) {\n          console.log(\"Bokeh: all BokehJS libraries loaded\");\n          run_callbacks()\n        }\n      };\n      s.onerror = function() {\n        console.warn(\"failed to load library \" + url);\n      };\n      console.log(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.getElementsByTagName(\"head\")[0].appendChild(s);\n    }\n  };var element = document.getElementById(\"10976\");\n  if (element == null) {\n    console.log(\"Bokeh: ERROR: autoload.js configured with elementid '10976' but no matching script tag was found. \")\n    return false;\n  }\n\n  var js_urls = [\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.js\", \"https://cdn.pydata.org/bokeh/release/bokeh-gl-1.0.4.min.js\"];\n\n  var inline_js = [\n    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\n    \n    function(Bokeh) {\n      \n    },\n    function(Bokeh) {\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-widgets-1.0.4.min.css\");\n      console.log(\"Bokeh: injecting CSS: https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n      Bokeh.embed.inject_css(\"https://cdn.pydata.org/bokeh/release/bokeh-tables-1.0.4.min.css\");\n    }\n  ];\n\n  function run_inline_js() {\n    \n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (var i = 0; i < inline_js.length; i++) {\n        inline_js[i].call(root, root.Bokeh);\n      }if (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      var cell = $(document.getElementById(\"10976\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.log(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(js_urls, function() {\n      console.log(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"bk-root\" id=\"9fe2279a-1621-4f7f-9785-d71b8b63832d\" data-root-id=\"10978\"></div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function embed_document(root) {\n",
       "    \n",
       "  var docs_json = {\"22d3149a-16f2-44ea-a8ea-66be0235d60d\":{\"roots\":{\"references\":[{\"attributes\":{\"below\":[{\"id\":\"10988\",\"type\":\"LinearAxis\"}],\"left\":[{\"id\":\"10993\",\"type\":\"LinearAxis\"}],\"plot_height\":500,\"plot_width\":800,\"renderers\":[{\"id\":\"10988\",\"type\":\"LinearAxis\"},{\"id\":\"10992\",\"type\":\"Grid\"},{\"id\":\"10993\",\"type\":\"LinearAxis\"},{\"id\":\"10997\",\"type\":\"Grid\"},{\"id\":\"11014\",\"type\":\"Legend\"},{\"id\":\"11007\",\"type\":\"GlyphRenderer\"}],\"title\":{\"id\":\"10977\",\"type\":\"Title\"},\"toolbar\":{\"id\":\"10999\",\"type\":\"Toolbar\"},\"toolbar_location\":null,\"x_range\":{\"id\":\"10980\",\"type\":\"DataRange1d\"},\"x_scale\":{\"id\":\"10984\",\"type\":\"LinearScale\"},\"y_range\":{\"id\":\"10982\",\"type\":\"DataRange1d\"},\"y_scale\":{\"id\":\"10986\",\"type\":\"LinearScale\"}},\"id\":\"10978\",\"subtype\":\"Figure\",\"type\":\"Plot\"},{\"attributes\":{\"field\":\"angle\",\"include_zero\":true},\"id\":\"11001\",\"type\":\"CumSum\"},{\"attributes\":{},\"id\":\"10986\",\"type\":\"LinearScale\"},{\"attributes\":{},\"id\":\"11010\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{},\"id\":\"11064\",\"type\":\"Selection\"},{\"attributes\":{\"callback\":null,\"data\":{\"angle\":{\"__ndarray__\":\"FIX1rQ1wez/6wjGTOnD+P3lDVA3+9vc/iG2DJ+4oA0BYdVdPoc3ePw==\",\"dtype\":\"float64\",\"shape\":[5]},\"color\":[\"#08519c\",\"#3182bd\",\"#6baed6\",\"#bdd7e7\",\"#eff3ff\"],\"index\":[0,1,2,3,4],\"sub_database\":[\"archaea\",\"bacteria\",\"env\",\"metagenomes\",\"pro\"],\"value\":{\"__ndarray__\":\"g1FJnYCGIUBpb/CFKXGjQHPXEvJBnZ5A845TdOR5qECP5PIf0qyDQA==\",\"dtype\":\"float64\",\"shape\":[5]}},\"selected\":{\"id\":\"11064\",\"type\":\"Selection\"},\"selection_policy\":{\"id\":\"11065\",\"type\":\"UnionRenderers\"}},\"id\":\"11003\",\"type\":\"ColumnDataSource\"},{\"attributes\":{\"callback\":null},\"id\":\"10980\",\"type\":\"DataRange1d\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"11002\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_color\":{\"field\":\"color\"},\"line_color\":{\"value\":\"white\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"11001\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"11005\",\"type\":\"Wedge\"},{\"attributes\":{\"formatter\":{\"id\":\"11010\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"10978\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10989\",\"type\":\"BasicTicker\"}},\"id\":\"10988\",\"type\":\"LinearAxis\"},{\"attributes\":{},\"id\":\"11012\",\"type\":\"BasicTickFormatter\"},{\"attributes\":{\"dimension\":1,\"plot\":{\"id\":\"10978\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10994\",\"type\":\"BasicTicker\"}},\"id\":\"10997\",\"type\":\"Grid\"},{\"attributes\":{\"active_drag\":\"auto\",\"active_inspect\":\"auto\",\"active_multi\":null,\"active_scroll\":\"auto\",\"active_tap\":\"auto\",\"tools\":[{\"id\":\"10998\",\"type\":\"HoverTool\"}]},\"id\":\"10999\",\"type\":\"Toolbar\"},{\"attributes\":{\"end_angle\":{\"expr\":{\"id\":\"11002\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"fill_alpha\":{\"value\":0.1},\"fill_color\":{\"value\":\"#1f77b4\"},\"line_alpha\":{\"value\":0.1},\"line_color\":{\"value\":\"#1f77b4\"},\"radius\":{\"units\":\"data\",\"value\":0.4},\"start_angle\":{\"expr\":{\"id\":\"11001\",\"type\":\"CumSum\"},\"units\":\"rad\"},\"x\":{\"value\":0},\"y\":{\"value\":1}},\"id\":\"11006\",\"type\":\"Wedge\"},{\"attributes\":{\"callback\":null},\"id\":\"10982\",\"type\":\"DataRange1d\"},{\"attributes\":{\"formatter\":{\"id\":\"11012\",\"type\":\"BasicTickFormatter\"},\"plot\":{\"id\":\"10978\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10994\",\"type\":\"BasicTicker\"}},\"id\":\"10993\",\"type\":\"LinearAxis\"},{\"attributes\":{\"callback\":null,\"tooltips\":\"@sub_database: @value GB\"},\"id\":\"10998\",\"type\":\"HoverTool\"},{\"attributes\":{},\"id\":\"10994\",\"type\":\"BasicTicker\"},{\"attributes\":{\"data_source\":{\"id\":\"11003\",\"type\":\"ColumnDataSource\"},\"glyph\":{\"id\":\"11005\",\"type\":\"Wedge\"},\"hover_glyph\":null,\"muted_glyph\":null,\"nonselection_glyph\":{\"id\":\"11006\",\"type\":\"Wedge\"},\"selection_glyph\":null,\"view\":{\"id\":\"11008\",\"type\":\"CDSView\"}},\"id\":\"11007\",\"type\":\"GlyphRenderer\"},{\"attributes\":{\"plot\":null,\"text\":\"Genomic Data on Server (in GB)\"},\"id\":\"10977\",\"type\":\"Title\"},{\"attributes\":{\"source\":{\"id\":\"11003\",\"type\":\"ColumnDataSource\"}},\"id\":\"11008\",\"type\":\"CDSView\"},{\"attributes\":{},\"id\":\"10989\",\"type\":\"BasicTicker\"},{\"attributes\":{\"field\":\"angle\"},\"id\":\"11002\",\"type\":\"CumSum\"},{\"attributes\":{\"label\":{\"field\":\"sub_database\"},\"renderers\":[{\"id\":\"11007\",\"type\":\"GlyphRenderer\"}]},\"id\":\"11015\",\"type\":\"LegendItem\"},{\"attributes\":{\"items\":[{\"id\":\"11015\",\"type\":\"LegendItem\"}],\"plot\":{\"id\":\"10978\",\"subtype\":\"Figure\",\"type\":\"Plot\"}},\"id\":\"11014\",\"type\":\"Legend\"},{\"attributes\":{},\"id\":\"11065\",\"type\":\"UnionRenderers\"},{\"attributes\":{},\"id\":\"10984\",\"type\":\"LinearScale\"},{\"attributes\":{\"plot\":{\"id\":\"10978\",\"subtype\":\"Figure\",\"type\":\"Plot\"},\"ticker\":{\"id\":\"10989\",\"type\":\"BasicTicker\"}},\"id\":\"10992\",\"type\":\"Grid\"}],\"root_ids\":[\"10978\"]},\"title\":\"Bokeh Application\",\"version\":\"1.0.4\"}};\n",
       "  var render_items = [{\"docid\":\"22d3149a-16f2-44ea-a8ea-66be0235d60d\",\"roots\":{\"10978\":\"9fe2279a-1621-4f7f-9785-d71b8b63832d\"}}];\n",
       "  root.Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "\n",
       "  }\n",
       "  if (root.Bokeh !== undefined) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (root.Bokeh !== undefined) {\n",
       "        embed_document(root);\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "      attempts++;\n",
       "      if (attempts > 100) {\n",
       "        console.log(\"Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing\");\n",
       "        clearInterval(timer);\n",
       "      }\n",
       "    }, 10, root)\n",
       "  }\n",
       "})(window);"
      ],
      "application/vnd.bokehjs_exec.v0+json": ""
     },
     "metadata": {
      "application/vnd.bokehjs_exec.v0+json": {
       "id": "10978"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "output_notebook();\n",
    "output_file(\"by_data_type.html\")\n",
    "\n",
    "p = figure(plot_height=500, plot_width=800,title=\"Genomic Data on Server (in GB)\", toolbar_location=None,\n",
    "        tools=\"hover\", tooltips=\"@sub_database: @value GB\")\n",
    "\n",
    "p.wedge(x=0, y=1, radius=0.4,\n",
    "        start_angle=cumsum('angle', include_zero=True), end_angle=cumsum('angle'),\n",
    "        line_color=\"white\", fill_color='color', legend='sub_database', source=by_sub_plot)\n",
    "\n",
    "show(p);"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Bar Graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3020: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "data = pd.read_csv('fasta_index_file_oa.txt', sep=\"\\t\", header=None, names = [\"id\", \"filename_id\", \"filename\", \"database\",\"sub_database\",\"filename_2\",\"7\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=size_data.groupby([\"database\",\"sub_database\"])[\"size\"].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 392,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=by_db_type.reset_index(name='value').rename(columns={'index':'database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 393,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type[\"value\"] = by_db_type[\"value\"].apply(lambda x: round(x/(10**9),4)) # Gigabyte"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 394,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>1959.3144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>629.6026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea</td>\n",
       "      <td>1.4988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_archaea_v2</td>\n",
       "      <td>0.3575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria</td>\n",
       "      <td>254.7375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_bacteria_v2</td>\n",
       "      <td>2.2356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes</td>\n",
       "      <td>709.9082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>JGI</td>\n",
       "      <td>JGI_metagenomes_v2</td>\n",
       "      <td>23.0328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>11.3817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>680.2256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>6.9064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>1200.5480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>938.3580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenome</td>\n",
       "      <td>1461.6472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>339.4527</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database          sub_database      value\n",
       "0               ENA                   env   1959.3144\n",
       "1               ENA                   pro    629.6026\n",
       "2               JGI           JGI_archaea      1.4988\n",
       "3               JGI        JGI_archaea_v2      0.3575\n",
       "4               JGI          JGI_bacteria    254.7375\n",
       "5               JGI       JGI_bacteria_v2      2.2356\n",
       "6               JGI       JGI_metagenomes    709.9082\n",
       "7               JGI    JGI_metagenomes_v2     23.0328\n",
       "8            MGnify              bacteria     11.3817\n",
       "9           Metasub              bacteria    680.2256\n",
       "10             NCBI               archaea      6.9064\n",
       "11             NCBI              bacteria   1200.5480\n",
       "12             NCBI            metagenome    938.3580\n",
       "13   Segata-Pasolli            metagenome   1461.6472\n",
       "14          ensembl              bacteria    339.4527"
      ]
     },
     "execution_count": 394,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_db_type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 395,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\users\\denni\\appdata\\local\\programs\\python\\python36\\lib\\site-packages\\ipykernel_launcher.py:16: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['env', 'pro', 'archaea', 'bacteria', 'metagenomes']"
      ]
     },
     "execution_count": 395,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_dbs_dir = {'metagenome': 'metagenomes',\n",
    " 'JGI_archaea':'archaea',\n",
    " 'JGI_bacteria':'bacteria',\n",
    " 'JGI_metagenomes':'metagenomes',\n",
    " 'JGI_metagenomes_v2':'metagenomes',\n",
    " 'JGI_bacteria_v2':\"bacteria\",\n",
    " 'JGI_archaea_v2':\"archaea\"}\n",
    "\n",
    "\n",
    "for index, row in by_db_type.iterrows():\n",
    "    #print (color_dict)\n",
    "    \n",
    "    db_name=row['sub_database']\n",
    "    \n",
    "    new_db = sub_dbs_dir.get(db_name.strip(),db_name.strip())\n",
    "    by_db_type.set_value(index,\"sub_database\",new_db)\n",
    "\n",
    "    \n",
    "sub_db = by_db_type.sub_database.unique()\n",
    "list(sub_db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=by_db_type.groupby([\"database\",\"sub_database\"])[\"value\"].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "metadata": {},
   "outputs": [],
   "source": [
    "by_db_type=by_db_type.reset_index(name='value').rename(columns={'index':'database'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database</th>\n",
       "      <th>sub_database</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENA</td>\n",
       "      <td>env</td>\n",
       "      <td>1959.3144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENA</td>\n",
       "      <td>pro</td>\n",
       "      <td>629.6026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JGI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>1.8563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>JGI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>256.9731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JGI</td>\n",
       "      <td>metagenomes</td>\n",
       "      <td>732.9410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MGnify</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>11.3817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Metasub</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>680.2256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>archaea</td>\n",
       "      <td>6.9064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>1200.5480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NCBI</td>\n",
       "      <td>metagenomes</td>\n",
       "      <td>938.3580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Segata-Pasolli</td>\n",
       "      <td>metagenomes</td>\n",
       "      <td>1461.6472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ensembl</td>\n",
       "      <td>bacteria</td>\n",
       "      <td>339.4527</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            database sub_database      value\n",
       "0               ENA           env  1959.3144\n",
       "1               ENA           pro   629.6026\n",
       "2               JGI       archaea     1.8563\n",
       "3               JGI      bacteria   256.9731\n",
       "4               JGI   metagenomes   732.9410\n",
       "5            MGnify      bacteria    11.3817\n",
       "6           Metasub      bacteria   680.2256\n",
       "7              NCBI       archaea     6.9064\n",
       "8              NCBI      bacteria  1200.5480\n",
       "9              NCBI   metagenomes   938.3580\n",
       "10   Segata-Pasolli   metagenomes  1461.6472\n",
       "11          ensembl      bacteria   339.4527"
      ]
     },
     "execution_count": 398,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "by_db_type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>sub_database</th>\n",
       "      <th>archaea</th>\n",
       "      <th>bacteria</th>\n",
       "      <th>env</th>\n",
       "      <th>metagenomes</th>\n",
       "      <th>pro</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>database</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENA</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1959.3144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>629.6026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JGI</th>\n",
       "      <td>1.8563</td>\n",
       "      <td>256.9731</td>\n",
       "      <td>NaN</td>\n",
       "      <td>732.9410</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGnify</th>\n",
       "      <td>NaN</td>\n",
       "      <td>11.3817</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metasub</th>\n",
       "      <td>NaN</td>\n",
       "      <td>680.2256</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI</th>\n",
       "      <td>6.9064</td>\n",
       "      <td>1200.5480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>938.3580</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Segata-Pasolli</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1461.6472</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl</th>\n",
       "      <td>NaN</td>\n",
       "      <td>339.4527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "sub_database      archaea   bacteria        env  metagenomes       pro\n",
       "database                                                              \n",
       " ENA                  NaN        NaN  1959.3144          NaN  629.6026\n",
       " JGI               1.8563   256.9731        NaN     732.9410       NaN\n",
       " MGnify               NaN    11.3817        NaN          NaN       NaN\n",
       " Metasub              NaN   680.2256        NaN          NaN       NaN\n",
       " NCBI              6.9064  1200.5480        NaN     938.3580       NaN\n",
       " Segata-Pasolli       NaN        NaN        NaN    1461.6472       NaN\n",
       " ensembl              NaN   339.4527        NaN          NaN       NaN"
      ]
     },
     "execution_count": 399,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pivot_df = by_db_type.pivot(index='database', columns='sub_database', values='value')\n",
    "pivot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmAAAAHwCAYAAAAB5dMOAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzt3XuclWW9///XB0TBE1FikrAD26gIDKicRFPRndhZTcNTYlqYZam/0tLysE3Nym073alfTdOdqBhqWtquNA1RVEDHASQUFRUhJU08IMrh8/tjLaYBh2GAmXvNgtfz8ZjHrPu6D+uzbmB4z31f93VFZiJJkqTitKt0AZIkSRsbA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsE2qXQBTdlmm22yZ8+elS5DkiRpjaZOnfqPzOzanG3bdADr2bMnU6ZMqXQZkiRJaxQRzzd3W29BSpIkFcwAJkmSVDADmCRJUsHadB8wSZI2JkuWLGHu3LksXry40qWoCR07dqR79+506NBhnY9hAJMkqY2YO3cuW221FT179iQiKl2OGpGZvPrqq8ydO5devXqt83G8BSlJUhuxePFiPvShDxm+2rCI4EMf+tB6X6U0gEmS1IYYvtq+lvgzMoBJkiQVzAAmSZI499xzufjii9dp32OPPZbx48c3uc11113HvHnz1nisfffdd6MYhN0AJkmSWl1zA9jGwgAmSdIG6u233+bTn/40AwYMoF+/fowbN46ePXvyj3/8A4ApU6aw77771m//xBNPsN9++9G7d2+uvvrq1R43MznppJPYZZdd+PSnP80rr7xSv+68885j8ODB9OvXjzFjxpCZjB8/nilTpnDUUUcxcOBA3nnnnUa3W+GGG25g+PDh9OvXj0cffRSARx99lOHDh7PrrrsyfPhwZs2aBcCMGTMYMmQIAwcOpKamhqeffrr+GCvaTzjhBJYtW9Zi57UlGMAkSdpA/d///R8f+chHeOKJJ5g+fToHHnhgk9vX1dVx1113MWnSJM4777zVXrG6/fbbmTVrFtOmTePqq6/moYceql930kknMXnyZKZPn84777zD73//ew499FAGDRrE2LFjqa2tpVOnTo1ut8Lbb7/NQw89xOWXX85xxx0HwM4778yECRN4/PHHOe+88zjzzDMBuPLKKzn55JOpra1lypQpdO/enZkzZzJu3DgefPBBamtrad++PWPHjl3f09miDGCSJG2g+vfvzz333MN3v/tdHnjgATp37tzk9p///Ofp1KkT22yzDSNGjKi/+rSqCRMmcMQRR9C+fXs+8pGPsN9++9Wvu++++xg6dCj9+/fnL3/5CzNmzGj0GE1td8QRRwCw995788Ybb/D666+zcOFCDjvsMPr168epp55av/0ee+zBhRdeyI9//GOef/55OnXqxL333svUqVMZPHgwAwcO5N577+XZZ59dq3PX2hyIVZKkDdSOO+7I1KlTufvuuznjjDM44IAD2GSTTVi+fDnA+8ayWnV4haaGW2hs3eLFi/n617/OlClT6NGjB+eee26j42WtabvG6jjrrLMYMWIEt99+O3PmzKm/dXrkkUcydOhQ7rrrLkaOHMkvf/lLMpPRo0fzox/9qOkTVEFeAZMkaQM1b948Nt98c44++mi+853v8Nhjj9GzZ0+mTp0KwK233rrS9nfccQeLFy/m1Vdf5f7772fw4MGNHnfvvffm5ptvZtmyZcyfP5/77rsP+Feg22abbXjrrbdWejJyq6224s0331zjdgDjxo0DYOLEiXTu3JnOnTuzcOFCtt9+e6DUoX+FZ599lh122IFvfetbfO5zn6Ouro7999+f8ePH1/dNe+2113j++efX/gS2Iq+ASZK0gZo2bRqnnXYa7dq1o0OHDlxxxRW88847HH/88Vx44YUMHTp0pe2HDBnCpz/9aV544QXOOussPvKRjzR63IMPPpi//OUv9O/fnx133JF99tkHgA984AN89atfpX///vTs2XOlAHfsscfyta99jU6dOjFp0qTVbgfQpUsXhg8fzhtvvMG1114LwOmnn87o0aO55JJLVrrlOW7cOG644QY6dOjAdtttx9lnn80HP/hBzj//fA444ACWL19Ohw4d+MUvfsFHP/rRFjmvLSEaPnXQ1gwaNChbciyQvQ75aYsdqzVMvO20SpcgSaqgmTNn0qdPn0qXoWZo7M8qIqZm5qDm7O8tSEmSpIJ5C1KSJDVq2rRpfOlLX1qpbbPNNuORRx6pUEUbDgOYJElqVP/+/amtra10GRskb0FKkiQVzAAmSZJUMAOYJElSwewDJklSG/WJdoe16PH+vPw3LXo8rbs1XgGLiB4RcV9EzIyIGRFxcrn93Ih4KSJqy1+farDPGRExOyJmRcTIBu0HlttmR8T3WucjSZIktW3NuQW5FPh2ZvYBhgHfiIhdyut+lpkDy193A5TXHQ70BQ4ELo+I9hHRHvgF8ElgF+CIBseRJEltwA033MCQIUMYOHAgJ5xwAsuWLWPLLbfk+9//PgMGDGDYsGG8/PLLLFy4kJ49e9bPK7lo0SJ69OjBkiVLKvwJqsMaA1hmzs/Mx8qv3wRmAts3scvngZsz893MfA6YDQwpf83OzGcz8z3g5vK2kiSpDZg5cybjxo3jwQcfpLa2lvbt2zN27Fjefvtthg0bxhNPPMHee+/N1VdfTefOnRkwYAB//etfAfjd737HyJEj6dChQ4U/RXVYq074EdET2BVYMQLbSRFRFxHXRkSXctv2wIsNdptbbltduyRJagPuvfdepk6dyuDBgxk4cCD33nsvzz77LJtuuimf+cxnANh9992ZM2cOAKNGjaqfOPvmm29m1KhRlSq96jQ7gEXElsCtwCmZ+QZwBfAxYCAwH/ivFZs2sns20b7q+4yJiCkRMWXBggXNLU+SJK2nzGT06NHU1tZSW1vLrFmzOPfcc+nQoQMRpf/G27dvz9KlSwH43Oc+xx/+8Adee+01pk6dutIk2WpaswJYRHSgFL7GZuZtAJn5cmYuy8zlwNWUbjFC6cpWjwa7dwfmNdG+ksy8KjMHZeagrl27ru3nkSRJ62j//fdn/PjxvPLKKwC89tprPP/886vdfsstt2TIkCGcfPLJfOYzn6F9+/ZFlVr11jgMRZQi7zXAzMy8pEF7t8ycX148GJhefn0ncGNEXAJ8BOgNPErpCljviOgFvESpo/6RLfVBJEna0BQ9bMQuu+zC+eefzwEHHMDy5cvp0KEDv/jFL5rcZ9SoURx22GHcf//9xRS5gWjOOGB7Al8CpkXEigmhzqT0FONASrcR5wAnAGTmjIi4BXiS0hOU38jMZQARcRLwR6A9cG1mzmjBzyJJktbTqFGj3teX66233qp/feihh3LooYeutJz5vh5FWoM1BrDMnEjj/bfubmKfC4ALGmm/u6n9JEmSNgZORSRJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUsOYMQyFJkipgr0N+2qLHm3jbaS16PK07r4BJkqS1smzZskqXUPUMYJIkqd6cOXPYeeedGT16NDU1NRx66KEsWrSInj17ct5557HXXnvxm9/8htraWoYNG0ZNTQ0HH3ww//znPytdelUxgEmSpJXMmjWLMWPGUFdXx9Zbb83ll18OQMeOHZk4cSKHH344xxxzDD/+8Y+pq6ujf//+/Od//meFq64uBjBJkrSSHj16sOeeewJw9NFHM3HiRID6KYoWLlzI66+/zj777APA6NGjmTBhQmWKrVIGMEmStJKIaHR5iy22qEQ5GyQDmCRJWskLL7zApEmTALjpppvYa6+9VlrfuXNnunTpwgMPPADAr3/96/qrYWoeh6GQJKmNqtSwEX369OH666/nhBNOoHfv3px44olcdtllK21z/fXX87WvfY1Fixaxww478Ktf/aoitVYrA5gkSVpJu3btuPLKK1dqmzNnzkrLAwcO5OGHHy6wqg2LtyAlSZIKZgCTJEn1evbsyfTp0ytdxgbPACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzGEoJElqo3r993+16PGeO+XbLXq8hq677jqmTJnC//zP/7Tae2xIvAImSZKabdmyZZUuYYNgAJMkSfUOOuggdt99d/r27ctVV10FwJZbbsnZZ5/N0KFDmTRpEpMnT2b48OEMGDCAIUOG8OabbwIwb948DjzwQHr37s3pp59ef8wTTzyRQYMG0bdvX84555z69qlTp7LPPvuw++67M3LkSObPnw/A1VdfzeDBgxkwYABf+MIXWLRoUYFnoBgGMEmSVO/aa69l6tSpTJkyhUsvvZRXX32Vt99+m379+vHII48wZMgQRo0axc9//nOeeOIJ7rnnHjp16gRAbW0t48aNY9q0aYwbN44XX3wRgAsuuIApU6ZQV1fHX//6V+rq6liyZAnf/OY3GT9+PFOnTuW4447j+9//PgCHHHIIkydP5oknnqBPnz5cc801FTsfrcU+YJIkqd6ll17K7bffDsCLL77I008/Tfv27fnCF74AwKxZs+jWrRuDBw8GYOutt67fd//996dz584A7LLLLjz//PP06NGDW265hauuuoqlS5cyf/58nnzySdq1a8f06dP5xCc+AZRubXbr1g2A6dOn84Mf/IDXX3+dt956i5EjRxb2+YtiAJMkSQDcf//93HPPPUyaNInNN9+cfffdl8WLF9OxY0fat28PQGYSEY3uv9lmm9W/bt++PUuXLuW5557j4osvZvLkyXTp0oVjjz2WxYsXk5n07duXSZMmve84xx57LL/97W8ZMGAA1113Hffff3+rfN5K8hakJEkCYOHChXTp0oXNN9+cv/3tb41Otr3zzjszb948Jk+eDMCbb77J0qVLV3vMN954gy222ILOnTvz8ssv84c//AGAnXbaiQULFtQHsCVLljBjxoz6Y3br1o0lS5YwduzYlv6YbYJXwCRJaqNac9iIxhx44IFceeWV1NTUsNNOOzFs2LD3bbPpppsybtw4vvnNb/LOO+/QqVMn7rnnntUec8CAAey666707duXHXbYgT333LP+OOPHj+db3/oWCxcuZOnSpZxyyin07duXH/7whwwdOpSPfvSj9O/fv76T/4YkMrPSNazWoEGDcsqUKS12vL0O+WmLHas1TLzttEqXIEmqoJkzZ9KnT59Kl6FmaOzPKiKmZuag5uzvLUhJkqSCGcAkSZIKZgCTJEkqmAFMkiSpYAYwSZKkghnAJEmSCuY4YJIktVFHP3J8ix7vhqFrnlNxzpw5fOYzn2H69Onr/D5z5szhoYce4sgjj1zrfYcPH85DDz20zu9dLbwCJkmSWtScOXO48cYb12qfZcuWAWwU4QsMYJIkaRVLly5l9OjR1NTUcOihh7Jo0SLOO+88Bg8eTL9+/RgzZgwrBnKfPXs2//Ef/8GAAQPYbbfdeOaZZ/je977HAw88wMCBA/nZz37GsmXLOO200xg8eDA1NTX8v//3/4DS3JMjRozgyCOPpH///gBsueWWALz11lvsv//+7LbbbvTv35877rijMiejlRjAJEnSSmbNmsWYMWOoq6tj66235vLLL+ekk05i8uTJTJ8+nXfeeYff//73ABx11FF84xvf4IknnuChhx6iW7duXHTRRXz84x+ntraWU089lWuuuYbOnTszefJkJk+ezNVXX81zzz0HwKOPPsoFF1zAk08+uVINHTt25Pbbb+exxx7jvvvu49vf/jZtefaetWUfMEmStJIePXrUz9l49NFHc+mll9KrVy9+8pOfsGjRIl577TX69u3Lvvvuy0svvcTBBx8MlEJTY/70pz9RV1fH+PHjgdKk308//TSbbropQ4YMoVevXu/bJzM588wzmTBhAu3ateOll17i5ZdfZrvttmulT10sA5gkSVpJRLxv+etf/zpTpkyhR48enHvuuSxevLjZV6Qyk8suu4yRI0eu1H7//fezxRZbNLrP2LFjWbBgAVOnTqVDhw707NmTxYsXr9sHaoO8BSlJklbywgsvMGnSJABuuukm9tprLwC22WYb3nrrrforWVtvvTXdu3fnt7/9LQDvvvsuixYtYquttuLNN9+sP97IkSO54oorWLJkCQBPPfUUb7/9dpM1LFy4kG233ZYOHTpw33338fzzz7f456wkr4BJktRGNWfYiNbQp08frr/+ek444QR69+7NiSeeyD//+U/69+9Pz549GTx4cP22v/71rznhhBM4++yz6dChA7/5zW+oqalhk002YcCAARx77LGcfPLJzJkzh912243MpGvXrvWhbXWOOuooPvvZzzJo0CAGDhzIzjvv3Nofu1DRlju0DRo0KKdMmdJix9vrkJ+22LFaw8TbTqt0CZKkCpo5cyZ9+vSpdBlqhsb+rCJiamYOas7+3oKUJEkqmLcgJWkDt/zvvStdQpPabfd0pUuQCucVMEmSpIIZwCRJkgpmAJMkSSqYAUySJKlgdsKXJKmNaukHKFr7gYfa2lrmzZvHpz71qVZ9nw2BV8AkSVKLqK2t5e677650GVXBACZJkurNmTOHnXfema985Sv069ePo446invuuYc999yT3r178+ijj/L2229z3HHHMXjwYHbddVfuuOMO3nvvPc4++2zGjRvHwIEDGTduHI8++ijDhw9n1113Zfjw4cyaNQuARYsW8cUvfpGamhpGjRrF0KFDWTHw+p/+9Cf22GMPdtttNw477DDeeustAHr27Mk555zDbrvtRv/+/fnb3/4GwGuvvcZBBx1ETU0Nw4YNo66uDoBzzz2X0aNHc8ABB9CzZ09uu+02Tj/9dPr378+BBx5YPy3S1KlT2Weffdh9990ZOXIk8+fPB+DSSy9ll112oaamhsMPP7zFz7MBTJIkrWT27NmcfPLJ1NXV8be//Y0bb7yRiRMncvHFF3PhhRdywQUXsN9++zF58mTuu+8+TjvtNJYsWcJ5553HqFGjqK2tZdSoUey8885MmDCBxx9/nPPOO48zzzwTgMsvv5wuXbpQV1fHWWedxdSpUwH4xz/+wfnnn88999zDY489xqBBg7jkkkvq69pmm2147LHHOPHEE7n44osBOOecc9h1112pq6vjwgsv5Jhjjqnf/plnnuGuu+7ijjvu4Oijj2bEiBFMmzaNTp06cdddd7FkyRK++c1vMn78eKZOncpxxx3H97//fQAuuugiHn/8cerq6rjyyitb/BzbB0ySJK2kV69e9O/fH4C+ffuy//77ExH079+fOXPmMHfuXO688876ELR48WJeeOGF9x1n4cKFjB49mqeffpqIqL/qNHHiRE4++WQA+vXrR01NDQAPP/wwTz75JHvuuScA7733HnvssUf98Q455BAAdt99d2677bb6Y916660A7Lfffrz66qssXLgQgE9+8pN06NCB/v37s2zZMg488ECA+s8xa9Yspk+fzic+8QkAli1bRrdu3QCoqanhqKOO4qCDDuKggw5qkfPakAFMkiStZLPNNqt/3a5du/rldu3asXTpUtq3b8+tt97KTjvttNJ+jzzyyErLZ511FiNGjOD2229nzpw57LvvvgCsbh7qzOQTn/gEN910U5N1tW/fnqVLl672WBGx0vbt2rWjQ4cO9e0rPkdm0rdvXyZNmvS+Y9x1111MmDCBO++8kx/+8IfMmDGDTTZpudjkLUhJkrRWRo4cyWWXXVYffh5//HEAttpqK95888367RYuXMj2228PwHXXXVffvtdee3HLLbcA8OSTTzJt2jQAhg0bxoMPPsjs2bOBUl+xp556qsla9t57b8aOHQvA/fffzzbbbMPWW2/drM+x0047sWDBgvoAtmTJEmbMmMHy5ct58cUXGTFiBD/5yU94/fXX6/uitRSvgEmS1Ea11XkyzzrrLE455RRqamrITHr27Mnvf/97RowYwUUXXcTAgQM544wzOP300xk9ejSXXHIJ++23X/3+X//61xk9ejQ1NTXsuuuu1NTU0LlzZ7p27cp1113HEUccwbvvvgvA+eefz4477rjaWs4991y+/OUvU1NTw+abb87111/f7M+x6aabMn78eL71rW+xcOFCli5dyimnnMKOO+7I0UcfzcKFC8lMTj31VD7wgQ+s+wlrRKzuMmBbMGjQoFzxVERL2OuQn7bYsVrDxNtOq3QJkjZATsZdPWbOnEmfPn0qXUarW7ZsGUuWLKFjx44888wz7L///jz11FNsuummlS6t2Rr7s4qIqZk5qDn7ewVMkiQVatGiRYwYMYIlS5aQmVxxxRVVFb5aggFMkiQVaquttqIl73BVozV2wo+IHhFxX0TMjIgZEXFyuf2DEfHniHi6/L1LuT0i4tKImB0RdRGxW4NjjS5v/3REjG69jyVJUnVqy12DVNISf0bNeQpyKfDtzOwDDAO+ERG7AN8D7s3M3sC95WWATwK9y19jgCugFNiAc4ChwBDgnBWhTZIkQceOHXn11VcNYW1YZvLqq6/SsWPH9TrOGm9BZuZ8YH759ZsRMRPYHvg8sG95s+uB+4Hvltv/N0t/ex6OiA9ERLfytn/OzNcAIuLPwIFA44N9SJK0kenevTtz585lwYIFlS5FTejYsSPdu3dfr2OsVR+wiOgJ7Ao8Any4HM7IzPkRsW15s+2BFxvsNrfctrp2SZIEdOjQgV69elW6DBWg2QOxRsSWwK3AKZn5RlObNtKWTbSv+j5jImJKREzxNwBJkrQhalYAi4gOlMLX2My8rdz8cvnWIuXvr5Tb5wI9GuzeHZjXRPtKMvOqzByUmYO6du26Np9FkiSpKjTnKcgArgFmZuYlDVbdCax4knE0cEeD9mPKT0MOAxaWb1X+ETggIrqUO98fUG6TJEnaqDSnD9iewJeAaRFRW247E7gIuCUijgdeAA4rr7sb+BQwG1gEfBkgM1+LiB8Ck8vbnbeiQ74kSdLGpDlPQU6k8f5bAPs3sn0C31jNsa4Frl2bAiVJkjY0ze6EL0mSpJaxUU1F1Om3j1a6BEmSJK+ASZIkFc0AJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwdYYwCLi2oh4JSKmN2g7NyJeioja8tenGqw7IyJmR8SsiBjZoP3ActvsiPhey38USZKk6tCcK2DXAQc20v6zzBxY/robICJ2AQ4H+pb3uTwi2kdEe+AXwCeBXYAjyttKkiRtdDZZ0waZOSEiejbzeJ8Hbs7Md4HnImI2MKS8bnZmPgsQETeXt31yrSuWJEmqcuvTB+ykiKgr36LsUm7bHnixwTZzy22ra5ckSdrorGsAuwL4GDAQmA/8V7k9Gtk2m2h/n4gYExFTImLKggUL1rE8SZKktmudAlhmvpyZyzJzOXA1/7rNOBfo0WDT7sC8JtobO/ZVmTkoMwd17dp1XcqTJElq09YpgEVEtwaLBwMrnpC8Ezg8IjaLiF5Ab+BRYDLQOyJ6RcSmlDrq37nuZUuSJFWvNXbCj4ibgH2BbSJiLnAOsG9EDKR0G3EOcAJAZs6IiFsoda5fCnwjM5eVj3MS8EegPXBtZs5o8U8jSZJUBZrzFOQRjTRf08T2FwAXNNJ+N3D3WlUnSZK0AVpjAJMkaWO2/O+9K11Ck9pt93SlS9A6cCoiSZKkghnAJEmSCmYAkyRJKpgBTJIkqWAGMEmSpIIZwCRJkgpmAJMkSSqYAUySJKlgBjBJkqSCGcAkSZIKZgCTJEkqmAFMkiSpYAYwSZKkghnAJEmSCmYAkyRJKpgBTJIkqWAGMEmSpIIZwCRJkgpmAJMkSSqYAUySJKlgBjBJkqSCGcAkSZIKZgCTJEkqmAFMkiSpYAYwSZKkghnAJEmSCmYAkyRJKpgBTJIkqWAGMEmSpIIZwCRJkgpmAJMkSSqYAUySJKlgBjBJkqSCGcAkSZIKZgCTJEkq2CaVLkCS1LqOeX7vSpfQpBu2q3QFUvG8AiZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBVsjQEsIq6NiFciYnqDtg9GxJ8j4uny9y7l9oiISyNidkTURcRuDfYZXd7+6YgY3TofR5Ikqe1rzhWw64ADV2n7HnBvZvYG7i0vA3wS6F3+GgNcAaXABpwDDAWGAOesCG2SJEkbmzUGsMycALy2SvPngevLr68HDmrQ/r9Z8jDwgYjoBowE/pyZr2XmP4E/8/5QJ0mStFFY1z5gH87M+QDl79uW27cHXmyw3dxy2+raJUmSNjot3Qk/GmnLJtrff4CIMRExJSKmLFiwoEWLkyRJags2Wcf9Xo6Ibpk5v3yL8ZVy+1ygR4PtugPzyu37rtJ+f2MHzsyrgKsABg0a1GhIU2Us/3vvSpfQpHbbPV3pEiRJapZ1vQJ2J7DiScbRwB0N2o8pPw05DFhYvkX5R+CAiOhS7nx/QLlNkiRpo7PGK2ARcROlq1fbRMRcSk8zXgTcEhHHAy8Ah5U3vxv4FDAbWAR8GSAzX4uIHwKTy9vsknbNAAAX5klEQVSdl5mrduyXJEnaKKwxgGXmEatZtX8j2ybwjdUc51rg2rWqTpIkaQPkSPiSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwTapdAGS1BxHP3J8pUtYrRuGXlPpEiRVGa+ASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFW68AFhFzImJaRNRGxJRy2wcj4s8R8XT5e5dye0TEpRExOyLqImK3lvgAkiRJ1aYlroCNyMyBmTmovPw94N7M7A3cW14G+CTQu/w1BriiBd5bkiSp6rTGLcjPA9eXX18PHNSg/X+z5GHgAxHRrRXeX5IkqU1b3wCWwJ8iYmpEjCm3fTgz5wOUv29bbt8eeLHBvnPLbZIkSRuVTdZz/z0zc15EbAv8OSL+1sS20Uhbvm+jUpAbA/Bv//Zv61meJElS27NeV8Ayc175+yvA7cAQ4OUVtxbL318pbz4X6NFg9+7AvEaOeVVmDsrMQV27dl2f8iRJktqkdQ5gEbFFRGy14jVwADAduBMYXd5sNHBH+fWdwDHlpyGHAQtX3KqUJEnamKzPLcgPA7dHxIrj3JiZ/xcRk4FbIuJ44AXgsPL2dwOfAmYDi4Avr8d7S5IkVa11DmCZ+SwwoJH2V4H9G2lP4Bvr+n6SJEkbCkfClyRJKpgBTJIkqWAGMEmSpIIZwCRJkgpmAJMkSSqYAUySJKlgBjBJkqSCGcAkSZIKZgCTJEkqmAFMkiSpYAYwSZKkghnAJEmSCmYAkyRJKpgBTJIkqWAGMEmSpIIZwCRJkgpmAJMkSSqYAUySJKlgm1S6AEmStOE6+pHjK11Ck24Yek1F3tcrYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcypiNRsxzy/d6VLaNIN21W6AkmSmscrYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUMAOYJElSwQxgkiRJBTOASZIkFcwAJkmSVDADmCRJUsEMYJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJkiQVzAAmSZJUsMIDWEQcGBGzImJ2RHyv6PeXJEmqtEIDWES0B34BfBLYBTgiInYpsgZJkqRK26Tg9xsCzM7MZwEi4mbg88CTBdchFe7oR46vdAlNumHoNZUuQZI2GkUHsO2BFxsszwWGFlyDJG1UHpzUxm80+L+ANkKRmcW9WcRhwMjM/Ep5+UvAkMz8ZoNtxgBjyos7AbMKK3DtbQP8o9JFVDHP3/rx/K07z9368fytH8/f+mnL5++jmdm1ORsWfQVsLtCjwXJ3YF7DDTLzKuCqIotaVxExJTMHVbqOauX5Wz+ev3XnuVs/nr/14/lbPxvK+Sv6KcjJQO+I6BURmwKHA3cWXIMkSVJFFXoFLDOXRsRJwB+B9sC1mTmjyBokSZIqrehbkGTm3cDdRb9vK6mKW6VtmOdv/Xj+1p3nbv14/taP52/9bBDnr9BO+JIkSXIqIkmSpMIZwCRJkgpmAJMkSSqYAUySJKlgBjBJkqSCGcDWQ0T0iIjTKl2HpDWLiE0joiYi+pcHgpakiil8HLBqFxHbAIcBR1CaXPz2ylbU9kXE74DVjneSmZ8rsJyqFRFTgF8BN2bmPytdTzWJiE8DVwLPAAH0iogTMvMPla2sbYuIQ5pan5m3FVVLNYqI/87MU1b3M9CffU2LiGk0/n9HAJmZNQWX1KIMYM0QEVsBBwNHAjtSCl07ZGb3ihZWPS6udAEbiMOBLwOTG4SxP6WD+TXHfwEjMnM2QER8DLgLMIA17bNNrEvAANa0X5e/+zNw3Xym0gW0JgdibYaIeAd4FPgBMDEzMyKezcwdKlyaNkIR0Y7SD6YrgOXAtcDPM/O1ihbWhkXEhMzcu8FyAH9t2Cap7YqI7YAhlIL/5Mz8e4VLWm9eAWueMyldfbgCuDEixlW4nqoSEZ8HumfmL8rLjwBdy6tPz8zxFSuuykREDaWrYJ8CbgXGAnsBfwEGVrC0NqnBLbQZEXE3cAulH+CHAZMrVlgViYh9gH9mZl1EfBHYm9Kt3Msz893KVte2NXELDYBqv4VWlIj4CnA2pZ9zAVwWEedl5rWVrWz9eAVsLUTEDpT6fh0O9AbOAW7PzKcqWlgbFxEPAodn5ovl5Vpgf2AL4FeZuX8l66sWETEVeB24Bri14X9+EXFbZjbZX2djFBG/amJ1ZuZxhRVThSLiF0AN0BGYBWwJ/B8wHGifmUdVsLw2LyI+2tT6zHy+qFqqWUTMAoZn5qvl5Q8BD2XmTpWtbP0YwNZRRPSn1Cfsi5n5sUrX05ZFxOTMHNxg+X8y86Ty64czc1jlqmv7IuLkzPx5ROydmRMqXY82HhHxZGbuEhEdgZeAbTNzWfkWbl1m9q9widoIRMS9wCcz873y8qbA3Zn5H5WtbP0YwNTqImJ2Zv77atY9Y4BtWkTUZubAiHgsM3erdD3VqHwlrLGn0LwC1oSGf+dW/fvn38c1i4g3WfnvXZSXVzzFt3VFCqsSEfH/lV8OBPoDd1A6f58HHs3Mr1WqtpZgH7BmiIjnaPwfEZT+ERkgmvZIRHw1M69u2BgRJ1B6uEFNmxkRc4CuEVHXoH2DeBS7IL9v8Lojpaea51Wolmqybfk/wWjwmvJy19XvJoDM3KrSNVS5FefvmfLXCndUoJYW5xWwZijfb26oHfBF4DvAY5n5heKrqh4RsS3wW+Bd4LFy8+7AZsBBmflypWqrFuUngP4IvG/cIPuRrL3yk6T3ZOZ+la6lLYuIc5pan5n/WVQt1S4iBgAfLy9OyMy6prbXhs8AthbKP7S/BJwG1AIXZuaTla2qekTEfkDf8uKMzPxLJevRxisidgLuWt2tcaklRcTJwFf517hpBwNXZeZllauqekTEIOD7wEdpcOeu2q/+G8CaISI6AMcBpwITgR9l5jNN76UVIuKDTax+NzPfLqyYKhQRt2TmFxt5pN1bkM3USF+cvwNnZOatFSqpKkTET4BnM/PKVdpPBbbLzO9WprLqUu46sMeKn3URsQUwyX+7zVN+CvI0YBqlsQ+B6r/6bwBrhoiYCywF/ht4YdX1TsfRtAZ96Br2nYvy9xW/zXwvM8cWXVs1iIhumTl/dY+0V/sPIbVdEfEk0C8zl6/S3o7SU5D9KlNZdSn/8jQ4MxeXlztSGkzUp0ibISImZuZela6jpdkJv3nuoRQcBpS/GnI6jjXIzF5NrY+IrsBfKQ0qqlVk5vzyd4PWOoqIPYHazHw7Io4GdqM0e4DntGm5avgqNy4vD0Wh5vkVpYeRbqf0y+fnKY3np+Y5JyJ+CdxLqS8xUP0XP7wCpjYhIj6bmb+rdB1tWXlU9x8D21L6Ie6j7M1UvgU0gNKgor+m9J/fIZm5T0ULa+MiYjJwZGY+vUp7b+CmzBxUmcqqT0TsRmnWiqDUCf/xCpdUNSLiBmBnYAb/ugVZ9QMpG8CaYcWM9uXXJ2fmzxusuy4zj61YcdpoRMRs4LOZObPStVSbFWNWRcTZwEuZeY3jWK1ZRHwSuAw4H5habh4EnAGckpl3V6q2alKe/H1uZr4bEftS+kXgfzPz9cpWVh0iYtqGeLu2XaULqBINJ+wdvco6O1GqKC8bvtbZmxFxBnA0cFdEtAc6VLimNi8z/wAcBIwArit/jQC+YPhaK7cCyyLi34FfAr2AGytbUlV5OCJ2qXQRLc0+YM0Tq3ktFWlKeSL4FWOqAdXfD6IgoyhNHXZ8Zv49Iv4N+GmFa6oKmTmd9//iqbWzPDOXlrsR/DwzL4sIb0E2317A6PIDXe+ygTwBbgBrnnYR0YXSFcMVr1cEsfaVK0sbma2BRcABDdp8CKQZMvPvwCUNll8A/rdyFVWH1U3hVJaZeXyR9VSxJRFxBHAM8Nlym1dgm+/AShfQGuwD1gzlaWCW0/jVr8zMHYqtSNLaiIhhlPoy9QE2pfSL01uZ2bmihbVxEdHYLB//BpwCtM/M7gWXVJXKt8++Rmnsr5siohcwKjMvqnBpVSMi9gJ6Z+avyk/Ob5mZz1W6rvVhAJPauIjoC3wsM+8sL/8MWBEc/iczH1vtzgIgIqYAhwO/odSJ/BhKP8zPrGhhVSQidgDOpNQn9mfANZn5XmWr0sagPCXWIGCnzNwxIj4C/CYz96xwaevFTvhS23cR8I8GyyOBu4D7gLMrUlEVyszZlK7aLMvMXwH7VrikqhARfcrDAPyO0kwgu2TmFYav5ouI3hExPiKejIhnV3xVuq4qcjCleXDfBsjMefxrou6qZR8wqe3rlpkPNVh+Y8UUOhFxQoVqqjaLImJToLY8vc58YIsK19TmRcSKK4YXU5qKbRmw9YoxWDPztcpVV1V+BZxD6crhCODL+EDX2ngvMzMiEuqncqp63oKU2riImJWZO61m3VOZuWPRNVWb8jROL1Pq/3UqpVu4v3BO16aV+7+u+E9i1WnE7P/aTBExNTN3bzieVUQ8kJkfr3Rt1SAivgP0Bj4B/IjS3Mw3Vvtk5l4Bk9q+eRExNDMfadhY7lg+r0I1VZuDygMoLwb+E0qDKgM/b3KvjVxm9qx0DRuIxeX5M5+OiJOAlyjNaKFmyMyLI+ITwBvATsDZmfnnCpe13rwCJrVxETEEGEdpEMwVHe53pzQ206jMfLRCpVWNxka9j4jHM3PXStWkjUdEDAZmAh8AfkhpSJmfZubDFS2sSpRvOS7OzGURsROlEPaHzFxS4dLWiwFMqgIR8WHgG0DfctMMSrfQXq5cVW1feeylIykN5PhAg1VbA0sz8z8qUpg2GuUhEz4KzHbqoXUTEVOBjwNdgIeBKcCizDyqooWtJwOYpA1Wue9XL0r9Rr7XYNWbQF1mLq1IYdooRMRXgAuBZyj9PRyzYjgZNV+DuVy/CXTKzJ9sCFew7QMmtXERUbe6VWwA03G0psx8Hnge2KMcxnpn5j0R0QnoRCmISa3lFKBvZi4oj6M2FjCArb2IiD2Ao4AVsy9UfX6p+g8gbQSWU3oC7UZKYzG9U9lyqk9EfBUYA3wQ+BjQHbgS2L+SdWmD915mLgDIzGcjYrNKF1SlTgbOAG7PzBnlMHtfhWtab96ClKpAROwMHEFpHrknKYWxP3kLrXkiohYYAjyy4rZFwyEBpNYQEa8ANzdoOrzhcmZ+q/Ci1GZ4BUyqApn5N0oDOZ4TEaMoTST9Y+CnFS2serybme+tGEA0IjZh9ZNMSy3ltFWWp1akiioXETsC3wF60iC3ZOZ+laqpJRjApCoQEdtT+u35YOCflAYTvb2iRVWXv0bEmUCn8nhCX6d0O1dqNZl5/aptEbFdZv69EvVUsd9Q6jLwS0qzMWwQvAUptXER8VdK857dAowHVpr+xelg1qw8CObxwAGUHl74I/DL9AegCtbYmHRq2oqZBCpdR0szgElt3Gqmg4F/PQXpdDBSldgQhk8oWkScC7xC6ar/uyvaq/2XTwOYpA1WE0N4AOAQHipaRHw9My+vdB3VJCKea6S56n/5NIBJ2mCVn35c7RAe5XHCpEJExLZAxxXLmflCBctRhbWrdAGS1FoycyCl4Tu2pBTCLqA0ndNLhi8VJSI+FxFPA88BfwXmAH+oaFFVJCI2j4gfRMRV5eXeEfGZSte1vgxgkjZomfm3zDyn3PH5d5SG8Di1wmVp4/JDYBjwVGb2ojQA8IOVLamq/Ap4DxheXp4LnF+5clqGAUzSBi0ito+Ib0fEROBoSuHrigqXpY3Lksx8FWgXEe0y8z5gYKWLqiIfy8yfAEsAMvMdSg8hVTXHAZO0wVplCI9j+dcQHptGxAer/SkqVY3XI2JLYAIwtjxCvrNYNN975flbEyAiPkaDpyGrlZ3wJW2wHMJDbUFEbEHpAZB2lCaU7gzc4C8AzVMePPkHwC7An4A9gWMz8/5K1rW+DGCSJLWiiPhxZn53TW1avYj4EKV+dAE8nJn/qHBJ680AJklSK2ps9PuIqHMcuo2bfcAkSWoFEXEipXlHd1hlUOCt8CnIjZ5XwCRJagUR0RnoAvwI+F6DVW/a/0sGMEmSCuBI+GrIccAkSWpFEfFZR8LXqgxgkiS1rvNxJHytwgAmSVLrciR8vY9PQUqS1LocCV/vYyd8SZJaUXkk/MWUBhFdMRL+2PJVMW2kDGCSJEkF8xakJEmtKCLeZOW5SAEWAlOAb2fms8VXpUozgEmS1LouAeYBN1K6DXk4sB0wC7gW2LdilalivAUpSVIriohHMnPoKm0PZ+awiHgiMwdUqjZVjsNQSJLUupZHxBcjol3564sN1nkVZCPlFTBJklpRROwA/BzYg1Lgehg4FXgJ2D0zJ1awPFWIAUySJKlg3oKUJKkVRcSOEXFvREwvL9dExA8qXZcqywAmSVLruho4A1gCkJl1lJ6E1EbMACZJUuvaPDMfXaXNqYg2cgYwSZJa1z8i4mOUn3iMiEOB+ZUtSZVmJ3xJklpR+SnIq4DhwD+B54CjMvP5ihamijKASZJUgPKk3O0y881K16LK8xakJEmtICI+GxEfbdD0bWBiRNwZEb0qVZfaBgOYJEmt4wJgAUBEfAY4GjgOuBO4soJ1qQ0wgEmS1DoyMxeVXx8CXJOZUzPzl0DXCtalNsAAJklS64iI2DIi2gH7A/c2WNexQjWpjdik0gVIkrSB+m+gFngDmJmZUwAiYlcchmKj51OQkiS1kojYHtgWeCIzl5fbugEdMvOFihanijKASZIkFcw+YJIkSQUzgEmSJBXMACZJklQwA5gkSVLBDGCSJEkFM4BJqhoRcW5EfKeJ9QdFxC7NOM51EXFoy1YnSc1nAJO0ITkIWGMAk6RKM4BJatMi4vsRMSsi7gF2Krd9NSImR8QTEXFrRGweEcOBzwE/jYjaiPhYY9s1OPR/RMQDEfFUeaJkIqJnue2x8tfwcnu3iJhQPu70iPh4uf2AiJhU3vY3EbFlsWdHUrVyIFZJbVZE7A5cBwylNHXaY8CVwK8y89XyNucDL2fmZRFxHfD7zBxfXvehJrbbDvgU8DHgPuDfKf1SujwzF0dEb+CmzBwUEd8GOmbmBRHRHtgc2Ay4DfhkZr4dEd8FNsvM81r/zEiqds4FKakt+zhwe2YuAoiIO8vt/cqB6gPAlsAfV7N/U9vdUp4a5umIeBbYGXgO+J+IGAgsA3YsbzsZuDYiOgC/zczaiNiH0u3OByMCYFNgUkt8aEkbPgOYpLauscv01wEHZeYTEXEssO9q9m1qu1WPm8CpwMvAAEpXwxYDZOaEiNgb+DTw64j4KfBP4M+ZecRafyJJGz37gElqyyYAB0dEp4jYCvhsuX0rYH75itRRDbZ/s7yONWwHcFhEtIuIjwE7ALOAzsD88pWxLwHtASLio8ArmXk1cA2wG/AwsGdE/Ht5m80jYkckqRm8AiapzcrMxyJiHFALPA88UF51FvBIuW0a/wpdNwNXR8S3gEOb2A5KgeuvwIeBr5X7fV0O3BoRh1HqF/Z2edt9gdMiYgnwFnBMZi4oX1W7KSI2K2/3A+CpljsDkjZUdsKXJEkqmLcgJUmSCmYAkyRJKpgBTJIkqWAGMEmSpIIZwCRJkgpmAJMkSSqYAUySJKlgBjBJkqSC/f8sJuWqnUUFGQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 720x504 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "pivot_df.loc[:,sub_dbs].plot.bar(stacked=True, color=viridis(5), figsize=(10,7));"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>sub_database</th>\n",
       "      <th>archaea</th>\n",
       "      <th>bacteria</th>\n",
       "      <th>env</th>\n",
       "      <th>metagenomes</th>\n",
       "      <th>pro</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>database</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENA</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1959.3144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>629.6026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JGI</th>\n",
       "      <td>1.8563</td>\n",
       "      <td>256.9731</td>\n",
       "      <td>NaN</td>\n",
       "      <td>732.9410</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGnify</th>\n",
       "      <td>NaN</td>\n",
       "      <td>11.3817</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metasub</th>\n",
       "      <td>NaN</td>\n",
       "      <td>680.2256</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI</th>\n",
       "      <td>6.9064</td>\n",
       "      <td>1200.5480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>938.3580</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Segata-Pasolli</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1461.6472</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl</th>\n",
       "      <td>NaN</td>\n",
       "      <td>339.4527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "sub_database      archaea   bacteria        env  metagenomes       pro\n",
       "database                                                              \n",
       " ENA                  NaN        NaN  1959.3144          NaN  629.6026\n",
       " JGI               1.8563   256.9731        NaN     732.9410       NaN\n",
       " MGnify               NaN    11.3817        NaN          NaN       NaN\n",
       " Metasub              NaN   680.2256        NaN          NaN       NaN\n",
       " NCBI              6.9064  1200.5480        NaN     938.3580       NaN\n",
       " Segata-Pasolli       NaN        NaN        NaN    1461.6472       NaN\n",
       " ensembl              NaN   339.4527        NaN          NaN       NaN"
      ]
     },
     "execution_count": 301,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pivot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "metadata": {},
   "outputs": [],
   "source": [
    "databases = list(by_db_type.database.unique())\n",
    "\n",
    "sub_dbs = list(by_db_type.sub_database.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i,v in enumerate(databases):\n",
    "    databases[i]=v.strip()\n",
    "\n",
    "for i,v in enumerate(sub_dbs):\n",
    "    sub_dbs[i]=v.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_dir={\n",
    "    \"databases\": databases,\n",
    "}\n",
    "for i in sub_db:\n",
    "    data_dir[i] = pivot_df[i].fillna(0).tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'databases': ['ENA',\n",
       "  'JGI',\n",
       "  'MGnify',\n",
       "  'Metasub',\n",
       "  'NCBI',\n",
       "  'Segata-Pasolli',\n",
       "  'ensembl'],\n",
       " 'env': [1959.3144, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],\n",
       " 'pro': [629.6026, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],\n",
       " 'archaea': [0.0, 1.8562999999999998, 0.0, 0.0, 6.9064, 0.0, 0.0],\n",
       " 'bacteria': [0.0, 256.9731, 11.3817, 680.2256, 1200.548, 0.0, 339.4527],\n",
       " 'metagenomes': [0.0, 732.9409999999999, 0.0, 0.0, 938.358, 1461.6472, 0.0]}"
      ]
     },
     "execution_count": 403,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"data_accu\n",
    "\n",
    "{'databases': ['NCBI',\n",
    "  'ENA',\n",
    "  'JGI',\n",
    "  'ensembl',\n",
    "  'MGnify',\n",
    "  'Metasub',\n",
    "  'Segata-Pasolli'],\n",
    " 'metagenomes': [10402, 0, 5072, 0, 0, 0, 9395],\n",
    " 'archaea': [3541, 0, 958, 0, 0, 0, 0],\n",
    " 'bacteria': [315772, 0, 42908, 93221, 4644, 3840, 0],\n",
    " 'pro': [0, 146559, 0, 0, 0, 0, 0],\n",
    " 'env': [0, 51876, 0, 0, 0, 0, 0]}\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "metadata": {},
   "outputs": [],
   "source": [
    "from bokeh.io import show, output_file, reset_output\n",
    "from bokeh.plotting import figure\n",
    "from bokeh.models import Legend\n",
    "from bokeh.palettes import Viridis256, viridis\n",
    "\n",
    "reset_output()\n",
    "\n",
    "#output_file(\"stacked.html\")\n",
    "output_file(\"stack_bar.html\")\n",
    "\n",
    "#output_notebook();\n",
    "\n",
    "\n",
    "colors =  viridis(5)\n",
    "\n",
    "p = figure(x_range=databases, plot_height=700, plot_width=800,title=\"Amount of data (in GB) by Database and Datatype\",\n",
    "           toolbar_location=None, tools=\"hover\", tooltips=\"@databases $name: @$name\")\n",
    "\n",
    "rs = p.vbar_stack(sub_dbs, x='databases', width=0.9, color=colors, source=data_dir)\n",
    "\n",
    "\n",
    "p.y_range.start = 0\n",
    "p.x_range.range_padding = 0.1\n",
    "p.xgrid.grid_line_color = None\n",
    "p.axis.minor_tick_line_color = None\n",
    "p.outline_line_color = None\n",
    "#p.legend.location = \"top_left\"\n",
    "#p.legend.orientation = \"horizontal\"\n",
    "\n",
    "\n",
    "legend = Legend(items=[(sub, [r]) for (sub, r) in zip(sub_dbs, rs)], location=(0, 30))\n",
    "p.add_layout(legend, 'right')\n",
    "\n",
    "show(p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
